diff --git a/package.json b/package.json
index 4c31d65c6a6..52bb4dabdea 100644
--- a/package.json
+++ b/package.json
@@ -67,9 +67,10 @@
     "puppeteer-extra-plugin-stealth": "^2.11.2",
     "queue": "^7.0.0",
     "systeminformation": "^5.27.3",
-    "tar-fs": "^3.0.10"
+    "tar-fs": "^3.0.10",
+    "user-agents": "latest"
   },
-  "optionalDependencies": {
+  "devDependencies": {
     "@types/chai": "^5.2.2",
     "@types/debug": "^4.1.12",
     "@types/gradient-string": "^1.1.6",
@@ -78,6 +79,7 @@
     "@types/mocha": "^10.0.10",
     "@types/node": "^24.0.3",
     "@types/sinon": "^17.0.4",
+    "@types/user-agents": "latest",
     "@typescript-eslint/eslint-plugin": "^8.34.1",
     "@typescript-eslint/parser": "^8.34.1",
     "assert": "^2.0.0",
diff --git a/src/browsers/browsers.cdp.ts b/src/browsers/browsers.cdp.ts
index 42ab7c3809c..d6e4682cce4 100644
--- a/src/browsers/browsers.cdp.ts
+++ b/src/browsers/browsers.cdp.ts
@@ -21,6 +21,9 @@ import path from 'path';
 import playwright from 'playwright-core';
 import puppeteerStealth from 'puppeteer-extra';
 
+// 引入随机 UA 生成库
+import UserAgent from 'user-agents';
+
 puppeteerStealth.use(StealthPlugin());
 
 export class ChromiumCDP extends EventEmitter {
@@ -190,7 +193,11 @@ export class ChromiumCDP extends EventEmitter {
     );
 
     const extensions = [
+      /*
+      忽略 blockAds 并强制启用 uBlock
       this.blockAds ? ublockLitePath : null,
+      */
+      ublockLitePath,
       extensionLaunchArgs ? extensionLaunchArgs.split('=')[1] : null,
     ].filter((_) => !!_);
 
@@ -214,11 +221,19 @@ export class ChromiumCDP extends EventEmitter {
       }
     }
 
+    // 生成随机桌面 UA
+    const userAgent = new UserAgent([
+      {
+        deviceCategory: "desktop",
+      },
+    ]);
+
     const finalOptions = {
       ...options,
       args: [
         `--remote-debugging-port=${this.port}`,
         `--no-sandbox`,
+        `--user-agent=${userAgent.data.userAgent}`, // 注入 UA
         ...(options.args || []),
         this.userDataDir ? `--user-data-dir=${this.userDataDir}` : '',
       ].filter((_) => !!_),
@@ -232,9 +247,12 @@ export class ChromiumCDP extends EventEmitter {
       );
     }
 
+    /* 忽略 stealth，强制开启 stealth 模式
     const launch = stealth
       ? puppeteerStealth.launch.bind(puppeteerStealth)
       : puppeteer.launch.bind(puppeteer);
+    */
+    const launch = puppeteerStealth.launch.bind(puppeteerStealth);
 
     this.logger.info(
       finalOptions,
