diff --git a/package.json b/package.json
index f79f0affc61..d15652cafe1 100644
--- a/package.json
+++ b/package.json
@@ -25,7 +25,7 @@
     "dev": "npm run build:dev && env-cmd -f .env node build",
     "install:adblock": "node scripts/install-adblock.js",
     "install:debugger": "node scripts/install-debugger.js",
-    "install:browsers": "node node_modules/playwright-core/cli.js install chromium firefox webkit msedge",
+    "install:browsers": "node node_modules/rebrowser-playwright-core/cli.js install chromium firefox webkit msedge",
     "install:dev": "npm run install:browsers && npm run install:debugger",
     "lint": "cross-env ESLINT_USE_FLAT_CONFIG=false eslint . --ext .ts --fix",
     "prepack": "npm run build:dev",
@@ -57,12 +57,8 @@
     "http-proxy": "^1.18.1",
     "lighthouse": "^12.6.1",
     "micromatch": "^4.0.8",
-    "playwright-1.49": "npm:playwright-core@1.49.1",
-    "playwright-1.50": "npm:playwright-core@1.50.1",
-    "playwright-1.51": "npm:playwright-core@1.51.1",
-    "playwright-1.52": "npm:playwright-core@1.52.0",
-    "playwright-core": "1.53.0",
-    "puppeteer-core": "24.10.1",
+    "rebrowser-playwright-core": "latest",
+    "rebrowser-puppeteer-core": "latest",
     "puppeteer-extra": "^3.3.6",
     "puppeteer-extra-plugin-stealth": "^2.11.2",
     "queue": "^7.0.0",
@@ -98,14 +94,6 @@
     "typescript": "^5.8.3",
     "typescript-json-schema": "^0.65.1"
   },
-  "playwrightVersions": {
-    "default": "playwright-core",
-    "1.53": "playwright-core",
-    "1.52": "playwright-1.52",
-    "1.51": "playwright-1.51",
-    "1.50": "playwright-1.50",
-    "1.49": "playwright-1.49"
-  },
   "eslintConfig": {
     "root": true,
     "parser": "@typescript-eslint/parser",
diff --git a/src/browsers/browsers.cdp.ts b/src/browsers/browsers.cdp.ts
index 42ab7c3809c..3c4901f1d4f 100644
--- a/src/browsers/browsers.cdp.ts
+++ b/src/browsers/browsers.cdp.ts
@@ -18,7 +18,7 @@ import StealthPlugin from 'puppeteer-extra-plugin-stealth';
 import getPort from 'get-port';
 import httpProxy from 'http-proxy';
 import path from 'path';
-import playwright from 'playwright-core';
+import playwright from 'rebrowser-playwright-core';
 import puppeteerStealth from 'puppeteer-extra';
 
 puppeteerStealth.use(StealthPlugin());
diff --git a/src/browsers/browsers.playwright.ts b/src/browsers/browsers.playwright.ts
index f14b9440eee..5ba5aee3bc4 100644
--- a/src/browsers/browsers.playwright.ts
+++ b/src/browsers/browsers.playwright.ts
@@ -8,7 +8,7 @@ import {
   chromeExecutablePath,
   edgeExecutablePath,
 } from '@browserless.io/browserless';
-import playwright, { Page } from 'playwright-core';
+import playwright, { Page } from 'rebrowser-playwright-core';
 import { Duplex } from 'stream';
 import { EventEmitter } from 'events';
 import httpProxy from 'http-proxy';
diff --git a/src/config.ts b/src/config.ts
index 62866286812..5a625348dbb 100644
--- a/src/config.ts
+++ b/src/config.ts
@@ -9,7 +9,7 @@ import debug from 'debug';
 import { fileURLToPath } from 'url';
 import { mkdir } from 'fs/promises';
 import path from 'path';
-import playwright from 'playwright-core';
+import playwright from 'rebrowser-playwright-core';
 import { tmpdir } from 'os';
 
 const __dirname = path.dirname(fileURLToPath(import.meta.url));
diff --git a/src/routes/chromium/tests/websocket.spec.ts b/src/routes/chromium/tests/websocket.spec.ts
index 9332b968a82..51d21cc32e0 100644
--- a/src/routes/chromium/tests/websocket.spec.ts
+++ b/src/routes/chromium/tests/websocket.spec.ts
@@ -7,7 +7,7 @@ import {
   fetchJson,
   sleep,
 } from '@browserless.io/browserless';
-import { chromium } from 'playwright-core';
+import { chromium } from 'rebrowser-playwright-core';
 import { deleteAsync } from 'del';
 import { expect } from 'chai';
 import puppeteer from 'puppeteer-core';
diff --git a/src/utils.ts b/src/utils.ts
index a8146cd32de..512672114af 100644
--- a/src/utils.ts
+++ b/src/utils.ts
@@ -19,9 +19,9 @@ import {
   encryptionAlgo,
   encryptionSep,
 } from '@browserless.io/browserless';
-import playwright, { CDPSession } from 'playwright-core';
+import playwright, { CDPSession } from 'rebrowser-playwright-core';
 import { Duplex } from 'stream';
-import { Page } from 'puppeteer-core';
+import { Page } from 'rebrowser-puppeteer-core';
 import { ServerResponse } from 'http';
 import crypto from 'crypto';
 import debug from 'debug';
