diff --git a/package.json b/package.json
index 4c31d65c6a6..7331980b3e4 100644
--- a/package.json
+++ b/package.json
@@ -69,7 +69,7 @@
     "systeminformation": "^5.27.3",
     "tar-fs": "^3.0.10"
   },
-  "optionalDependencies": {
+  "devDependencies": {
     "@types/chai": "^5.2.2",
     "@types/debug": "^4.1.12",
     "@types/gradient-string": "^1.1.6",
diff --git a/scripts/install-adblock.js b/scripts/install-adblock.js
index 65129bd9ef5..94cb0c99b91 100644
--- a/scripts/install-adblock.js
+++ b/scripts/install-adblock.js
@@ -39,7 +39,11 @@ import unzip from 'extract-zip';
     await deleteAsync(uBlockLiteDir);
   }
   const data = await fetch(
+    /*
+    更换去广告插件为 AdGuard
     'https://api.github.com/repos/uBlockOrigin/uBOL-home/releases/latest',
+    */
+    'https://api.github.com/repos/AdguardTeam/AdguardBrowserExtension/releases/latest',
   );
   const json = await data.json();
 
diff --git a/src/browsers/browsers.cdp.ts b/src/browsers/browsers.cdp.ts
index 42ab7c3809c..ea7d3697c4c 100644
--- a/src/browsers/browsers.cdp.ts
+++ b/src/browsers/browsers.cdp.ts
@@ -11,7 +11,11 @@ import {
   once,
   ublockLitePath,
 } from '@browserless.io/browserless';
+/*
+移除 puppeteer，修复 TS6133
 import puppeteer, { Browser, Page, Target } from 'puppeteer-core';
+*/
+import { Browser, Page, Target } from 'puppeteer-core';
 import { Duplex } from 'stream';
 import { EventEmitter } from 'events';
 import StealthPlugin from 'puppeteer-extra-plugin-stealth';
@@ -173,7 +177,10 @@ export class ChromiumCDP extends EventEmitter {
 
   public async launch({
     options,
+    /*
+    移除 stealth，修复 TS6133
     stealth,
+    */
   }: BrowserLauncherOptions): Promise<Browser> {
     this.port = await getPort();
     this.logger.info(`${this.constructor.name} got open port ${this.port}`);
@@ -190,7 +197,11 @@ export class ChromiumCDP extends EventEmitter {
     );
 
     const extensions = [
+      /*
+      忽略 blockAds 并强制启用 uBlock
       this.blockAds ? ublockLitePath : null,
+      */
+      ublockLitePath,
       extensionLaunchArgs ? extensionLaunchArgs.split('=')[1] : null,
     ].filter((_) => !!_);
 
@@ -232,9 +243,12 @@ export class ChromiumCDP extends EventEmitter {
       );
     }
 
+    /* 忽略 stealth，强制开启 stealth 模式
     const launch = stealth
       ? puppeteerStealth.launch.bind(puppeteerStealth)
       : puppeteer.launch.bind(puppeteer);
+    */
+    const launch = puppeteerStealth.launch.bind(puppeteerStealth);
 
     this.logger.info(
       finalOptions,
