diff --git a/src/browsers/browsers.cdp.ts b/src/browsers/browsers.cdp.ts
index 2135d92d0ca..cc0cb3308b0 100644
--- a/src/browsers/browsers.cdp.ts
+++ b/src/browsers/browsers.cdp.ts
@@ -15,6 +15,7 @@ import {
 import puppeteer, { Browser, Page, Target } from 'puppeteer-core';
 import { Duplex } from 'stream';
 import { EventEmitter } from 'events';
+import AdblockPlugin from 'puppeteer-extra-plugin-adblocker';
 import StealthPlugin from 'puppeteer-extra-plugin-stealth';
 import getPort from 'get-port';
 import httpProxy from 'http-proxy';
@@ -23,6 +24,9 @@ import playwright from 'playwright-core';
 import puppeteerStealth from 'puppeteer-extra';
 
 puppeteerStealth.use(StealthPlugin());
+puppeteerStealth.use(AdblockPlugin({
+  blockTrackers: true
+}));
 
 export class ChromiumCDP extends EventEmitter {
   protected config: Config;
