diff --git a/src/browsers/browsers.cdp.ts b/src/browsers/browsers.cdp.ts
index 2135d92d0ca..d4bc8e27573 100644
--- a/src/browsers/browsers.cdp.ts
+++ b/src/browsers/browsers.cdp.ts
@@ -15,6 +15,9 @@ import {
 import puppeteer, { Browser, Page, Target } from 'puppeteer-core';
 import { Duplex } from 'stream';
 import { EventEmitter } from 'events';
+// 引入 adblocker 及 recaptcha 插件
+import AdblockPlugin from 'puppeteer-extra-plugin-adblocker';
+import RecaptchaPlugin from 'puppeteer-extra-plugin-recaptcha';
 import StealthPlugin from 'puppeteer-extra-plugin-stealth';
 import getPort from 'get-port';
 import httpProxy from 'http-proxy';
@@ -22,6 +25,18 @@ import path from 'path';
 import playwright from 'playwright-core';
 import puppeteerStealth from 'puppeteer-extra';
 
+// 启用 AdblockPlugin 插件，启用 blockTrackersAndAnnoyances 功能
+puppeteerStealth.use(AdblockPlugin({
+  blockTrackersAndAnnoyances: true,
+}));
+// 启用 RecaptchaPlugin 插件
+puppeteerStealth.use(RecaptchaPlugin({
+  provider: {
+    id: '2captcha',
+    token: process.env.TWORECAPTCHA_API_KEY || '',
+  },
+  visualFeedback: true,
+}));
 puppeteerStealth.use(StealthPlugin());
 
 export class ChromiumCDP extends EventEmitter {
