diff --git a/src/browsers/browsers.cdp.ts b/src/browsers/browsers.cdp.ts
index ce0a980a19b..786ee19cd8c 100644
--- a/src/browsers/browsers.cdp.ts
+++ b/src/browsers/browsers.cdp.ts
@@ -10,6 +10,7 @@ import {
   noop,
   once,
   adGuardPath,
+  privacyBadgerPath,
   ublockLitePath,
 } from '@browserless.io/browserless';
 /*
@@ -203,6 +204,7 @@ export class ChromiumCDP extends EventEmitter {
       this.blockAds ? ublockLitePath : null,
       */
       adGuardPath,
+      privacyBadgerPath,
       ublockLitePath,
       extensionLaunchArgs ? extensionLaunchArgs.split('=')[1] : null,
     ].filter((_) => !!_);
diff --git a/src/utils.ts b/src/utils.ts
index 607c0347b3e..a5bee693d72 100644
--- a/src/utils.ts
+++ b/src/utils.ts
@@ -912,6 +912,13 @@ export const adGuardPath = path.join(
   'adguard',
 );
 
+export const privacyBadgerPath = path.join(
+  __dirname,
+  '..',
+  'extensions',
+  'privacy_badger',
+);
+
 export const ublockLitePath = path.join(
   __dirname,
   '..',
