diff --git a/package.json b/package.json
index 7f49f4ce766..922c84caf44 100644
--- a/package.json
+++ b/package.json
@@ -64,6 +64,8 @@
     "playwright-core": "1.53.0",
     "puppeteer-core": "24.10.2",
     "puppeteer-extra": "^3.3.6",
+    "puppeteer-extra-plugin-adblocker": "^2.13.6",
+		"puppeteer-extra-plugin-recaptcha": "^3.6.8",
     "puppeteer-extra-plugin-stealth": "^2.11.2",
     "queue": "^7.0.0",
     "systeminformation": "^5.27.3",
diff --git a/src/browsers/browsers.cdp.ts b/src/browsers/browsers.cdp.ts
index c2b36e41769..1129843076f 100644
--- a/src/browsers/browsers.cdp.ts
+++ b/src/browsers/browsers.cdp.ts
@@ -24,9 +24,24 @@ import getPort from 'get-port';
 import httpProxy from 'http-proxy';
 import path from 'path';
 import playwright from 'playwright-core';
+/*
+移除 puppeteerStealth
 import puppeteerStealth from 'puppeteer-extra';
 
 puppeteerStealth.use(StealthPlugin());
+*/
+
+import puppeteer from 'puppeteer-extra';
+
+puppeteer.use(AdblockerPlugin({ blockTrackers: true }));
+puppeteer.use(RecaptchaPlugin({
+  provider: {
+    id: '2captcha',
+    token: process.env.TWOCAPTCHA_API_KEY || '',
+  },
+  visualFeedback: true
+}));
+puppeteer.use(StealthPlugin());
 
 export class ChromiumCDP extends EventEmitter {
   protected config: Config;
@@ -250,7 +265,7 @@ export class ChromiumCDP extends EventEmitter {
       ? puppeteerStealth.launch.bind(puppeteerStealth)
       : puppeteer.launch.bind(puppeteer);
     */
-    const launch = puppeteerStealth.launch.bind(puppeteerStealth);
+    const launch = puppeteer.launch.bind(puppeteer);
 
     this.logger.info(
       finalOptions,
