diff --git a/package.json b/package.json
index 3bfdd906970..a1698c3dcd1 100644
--- a/package.json
+++ b/package.json
@@ -58,11 +58,11 @@
     "http-proxy": "^1.18.1",
     "lighthouse": "^12.6.1",
     "micromatch": "^4.0.8",
-    "playwright-1.49": "npm:playwright-core@1.49.1",
-    "playwright-1.50": "npm:playwright-core@1.50.1",
-    "playwright-1.51": "npm:playwright-core@1.51.1",
-    "playwright-1.52": "npm:playwright-core@1.52.0",
-    "playwright-core": "1.53.0",
+    "playwright-1.49": "npm:patchright-core@1.49.1",
+    "playwright-1.50": "npm:patchright-core@1.50.1",
+    "playwright-1.51": "npm:patchright-core@1.51.1",
+    "playwright-1.52": "npm:patchright-core@1.52.0",
+    "playwright-core": "npm:patchright-core@latest",
     "puppeteer-core": "24.10.2",
     "puppeteer-extra": "^3.3.6",
     "puppeteer-extra-plugin-stealth": "^2.11.2",
diff --git a/src/browsers/browsers.cdp.ts b/src/browsers/browsers.cdp.ts
index 786ee19cd8c..be8500e5b70 100644
--- a/src/browsers/browsers.cdp.ts
+++ b/src/browsers/browsers.cdp.ts
@@ -13,11 +13,7 @@ import {
   privacyBadgerPath,
   ublockLitePath,
 } from '@browserless.io/browserless';
-/*
-移除 puppeteer，修复 TS6133
 import puppeteer, { Browser, Page, Target } from 'puppeteer-core';
-*/
-import { Browser, Page, Target } from 'puppeteer-core';
 import { Duplex } from 'stream';
 import { EventEmitter } from 'events';
 import StealthPlugin from 'puppeteer-extra-plugin-stealth';
@@ -179,10 +175,7 @@ export class ChromiumCDP extends EventEmitter {
 
   public async launch({
     options,
-    /*
-    移除 stealth，修复 TS6133
     stealth,
-    */
   }: BrowserLauncherOptions): Promise<Browser> {
     this.port = await getPort();
     this.logger.info(`${this.constructor.name} got open port ${this.port}`);
@@ -247,12 +240,9 @@ export class ChromiumCDP extends EventEmitter {
       );
     }
 
-    /* 忽略 stealth，强制开启 stealth 模式
     const launch = stealth
       ? puppeteerStealth.launch.bind(puppeteerStealth)
       : puppeteer.launch.bind(puppeteer);
-    */
-    const launch = puppeteerStealth.launch.bind(puppeteerStealth);
 
     this.logger.info(
       finalOptions,
