diff --git a/src/browsers/browsers.cdp.ts b/src/browsers/browsers.cdp.ts
index c2b36e41769..e05e9f22227 100644
--- a/src/browsers/browsers.cdp.ts
+++ b/src/browsers/browsers.cdp.ts
@@ -28,6 +28,17 @@ import puppeteerStealth from 'puppeteer-extra';
 
 puppeteerStealth.use(StealthPlugin());
 
+// 引入 Adblock 插件
+puppeteerStealth.use(AdblockerPlugin({ blockTrackers: true }));
+// 引入 reChaptcha 插件
+puppeteerStealth.use(RecaptchaPlugin({
+  provider: {
+    id: '2captcha',
+    token: process.env.TWOCAPTCHA_API_KEY || '',
+  },
+  visualFeedback: true
+}));
+
 export class ChromiumCDP extends EventEmitter {
   protected config: Config;
   protected userDataDir: string | null;
