diff --git a/package.json b/package.json
index 7f49f4ce766..123ee80ba09 100644
--- a/package.json
+++ b/package.json
@@ -62,7 +62,7 @@
     "playwright-1.51": "npm:playwright-core@1.51.1",
     "playwright-1.52": "npm:playwright-core@1.52.0",
     "playwright-core": "1.53.0",
-    "puppeteer-core": "24.10.2",
+    "puppeteer-core": "npm:rebrowser-puppeteer-core@latest",
     "puppeteer-extra": "^3.3.6",
     "puppeteer-extra-plugin-stealth": "^2.11.2",
     "queue": "^7.0.0",
diff --git a/src/browsers/browsers.cdp.ts b/src/browsers/browsers.cdp.ts
index c2b36e41769..1bc6d72feb4 100644
--- a/src/browsers/browsers.cdp.ts
+++ b/src/browsers/browsers.cdp.ts
@@ -12,10 +12,7 @@ import {
   privacyBadgerPath,
   ublockLitePath,
 } from '@browserless.io/browserless';
-/*
-移除 puppeteer，修复 TS6133
 import puppeteer, { Browser, Page, Target } from 'puppeteer-core';
-*/
 import { Browser, Page, Target } from 'puppeteer-core';
 import { Duplex } from 'stream';
 import { EventEmitter } from 'events';
@@ -178,10 +175,7 @@ export class ChromiumCDP extends EventEmitter {
 
   public async launch({
     options,
-    /*
-    移除 stealth，修复 TS6133
     stealth,
-    */
   }: BrowserLauncherOptions): Promise<Browser> {
     this.port = await getPort();
     this.logger.info(`${this.constructor.name} got open port ${this.port}`);
@@ -245,12 +239,9 @@ export class ChromiumCDP extends EventEmitter {
       );
     }
 
-    /* 忽略 stealth，强制开启 stealth 模式
     const launch = stealth
       ? puppeteerStealth.launch.bind(puppeteerStealth)
       : puppeteer.launch.bind(puppeteer);
-    */
-    const launch = puppeteerStealth.launch.bind(puppeteerStealth);
 
     this.logger.info(
       finalOptions,
