diff --git a/src/browsers/browsers.cdp.ts b/src/browsers/browsers.cdp.ts
index bb22eaa272b..e3abf7072fa 100644
--- a/src/browsers/browsers.cdp.ts
+++ b/src/browsers/browsers.cdp.ts
@@ -234,6 +234,7 @@ export class ChromiumCDP extends EventEmitter {
       '--disable-crashpad-for-testing',
       '--disable-crashpad-forwarding',
       '--disable-in-process-stack-traces',
+      '--disable-web-security',
       '--no-default-browser-check',
 
       // 反检测增强
@@ -251,13 +252,28 @@ export class ChromiumCDP extends EventEmitter {
       '--single-process',
     ];
 
+    const mobileDeviceArgs = [
+      // 移动设备模式
+      '--enable-viewport',
+      '--touch-events=enabled',
+      '--use-mobile-user-agent',
+      // iPad Mini 设备参数
+      '--device-scale-factor=2',
+      '--screen-height=1024',
+      '--screen-width=768',
+      '--window-position=0,0',
+      '--window-size=768,1024',
+    ];
+
     const finalOptions = {
       ...options,
       args: [
         `--remote-debugging-port=${this.port}`,
         `--no-sandbox`,
-        // 注入补充 Patch 参数
+        // 注入 patchOptions 参数
         ...patchOptions,
+        // 注入 mobileDeviceArgs 参数
+        ...mobileDeviceArgs,
         ...(options.args || []),
         this.userDataDir ? `--user-data-dir=${this.userDataDir}` : '',
       ].filter((_) => !!_),
