diff --git a/configure b/configure
index a41e3e1eb..3f1def34b 100755
--- a/configure
+++ b/configure
@@ -22438,18 +22438,8 @@ then :
 fi
 
 
-  for ac_func in SSL_is_quic
-do :
-  ac_fn_c_check_func "$LINENO" "SSL_is_quic" "ac_cv_func_SSL_is_quic"
-if test "x$ac_cv_func_SSL_is_quic" = xyes
-then :
-  printf "%s\n" "#define HAVE_SSL_IS_QUIC 1" >>confdefs.h
-
-else $as_nop
-  as_fn_error $? "No QUIC support detected in OpenSSL. Need OpenSSL version with QUIC support to enable DNS over QUIC with libngtcp2." "$LINENO" 5
-fi
+  printf "%s\n" "#define HAVE_SSL_PROVIDE_QUIC_DATA 1" >>confdefs.h
 
-done
     ac_fn_c_check_type "$LINENO" "struct ngtcp2_version_cid" "ac_cv_type_struct_ngtcp2_version_cid" "$ac_includes_default
     #include <ngtcp2/ngtcp2.h>
 
