diff --git a/config.h.in b/config.h.in
index dc03e82dd..293d9f91a 100644
--- a/config.h.in
+++ b/config.h.in
@@ -654,8 +654,8 @@
 /* Define to 1 if you have the `SSL_get1_peer_certificate' function. */
 #undef HAVE_SSL_GET1_PEER_CERTIFICATE
 
-/* Define to 1 if you have the `SSL_is_quic' function. */
-#undef HAVE_SSL_IS_QUIC
+/* Define to 1 if you have the `SSL_provide_quic_data' function. */
+#undef HAVE_SSL_PROVIDE_QUIC_DATA
 
 /* Define to 1 if you have the `SSL_set1_host' function. */
 #undef HAVE_SSL_SET1_HOST
diff --git a/configure b/configure
index a41e3e1eb..3f1def34b 100755
--- a/configure
+++ b/configure
@@ -22438,18 +22438,8 @@ then :
 fi
 
 
-  for ac_func in SSL_is_quic
-do :
-  ac_fn_c_check_func "$LINENO" "SSL_is_quic" "ac_cv_func_SSL_is_quic"
-if test "x$ac_cv_func_SSL_is_quic" = xyes
-then :
-  printf "%s\n" "#define HAVE_SSL_IS_QUIC 1" >>confdefs.h
-
-else $as_nop
-  as_fn_error $? "No QUIC support detected in OpenSSL. Need OpenSSL version with QUIC support to enable DNS over QUIC with libngtcp2." "$LINENO" 5
-fi
+  printf "%s\n" "#define HAVE_SSL_PROVIDE_QUIC_DATA 1" >>confdefs.h
 
-done
     ac_fn_c_check_type "$LINENO" "struct ngtcp2_version_cid" "ac_cv_type_struct_ngtcp2_version_cid" "$ac_includes_default
     #include <ngtcp2/ngtcp2.h>
 
diff --git a/configure.ac b/configure.ac
index a5a856fc1..2b9b22773 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1616,7 +1616,7 @@ if test x_$withval = x_yes -o x_$withval != x_no; then
     AC_CHECK_LIB([ngtcp2_crypto_openssl], [ngtcp2_crypto_encrypt_cb], [ LIBS="$LIBS -lngtcp2_crypto_openssl" ])
     AC_CHECK_LIB([ngtcp2_crypto_quictls], [ngtcp2_crypto_encrypt_cb], [ LIBS="$LIBS -lngtcp2_crypto_quictls" ])
     AC_CHECK_FUNCS([ngtcp2_crypto_encrypt_cb ngtcp2_ccerr_default ngtcp2_conn_in_closing_period ngtcp2_conn_in_draining_period ngtcp2_conn_get_max_local_streams_uni ngtcp2_crypto_quictls_from_ossl_encryption_level ngtcp2_crypto_quictls_configure_server_context ngtcp2_crypto_quictls_configure_client_context ngtcp2_conn_get_num_scid ngtcp2_conn_tls_early_data_rejected ngtcp2_conn_encode_0rtt_transport_params])
-    AC_CHECK_FUNCS([SSL_is_quic], [], [AC_MSG_ERROR([No QUIC support detected in OpenSSL. Need OpenSSL version with QUIC support to enable DNS over QUIC with libngtcp2.])])
+    AC_CHECK_FUNCS([SSL_provide_quic_data], [], [AC_MSG_ERROR([No QUIC support detected in OpenSSL. Need OpenSSL version with QUIC support to enable DNS over QUIC with libngtcp2.])])
     AC_CHECK_TYPES([struct ngtcp2_version_cid, ngtcp2_encryption_level],,,[AC_INCLUDES_DEFAULT
     #include <ngtcp2/ngtcp2.h>
     ])
