diff --git a/packages/model-runtime/src/core/openaiCompatibleFactory/index.ts b/packages/model-runtime/src/core/openaiCompatibleFactory/index.ts
index 0c5c5547df862..0b374e4bb5b23 100644
--- a/packages/model-runtime/src/core/openaiCompatibleFactory/index.ts
+++ b/packages/model-runtime/src/core/openaiCompatibleFactory/index.ts
@@ -184,12 +184,23 @@ export const createOpenAICompatibleRuntime = <T extends Record<string, any> = an
         apiKey: options.apiKey?.trim() || DEFAULT_API_KEY,
         baseURL: options.baseURL?.trim() || DEFAULT_BASE_URL,
       };
-      const { apiKey, baseURL = DEFAULT_BASE_URL, ...res } = _options;
+      const { apiKey, baseURL = DEFAULT_BASE_URL, defaultHeaders, ...res } = _options;
       this._options = _options as ConstructorOptions<T>;
 
       if (!apiKey) throw AgentRuntimeError.createError(ErrorType?.invalidAPIKey);
 
-      const initOptions = { apiKey, baseURL, ...constructorOptions, ...res };
+      const initOptions = {
+        apiKey,
+        baseURL,
+        defaultHeaders: {
+          ...defaultHeaders,
+          ...(baseURL === 'https://aihubmix.com/v1' && {
+            'APP-Code': 'LobeHub',
+          }),
+        },
+        ...constructorOptions,
+        ...res,
+      };
 
       // if the custom client is provided, use it as client
       if (customClient?.createClient) {
diff --git a/packages/model-runtime/src/providers/anthropic/index.ts b/packages/model-runtime/src/providers/anthropic/index.ts
index 857dcadfcce4a..2f82b09920cc3 100644
--- a/packages/model-runtime/src/providers/anthropic/index.ts
+++ b/packages/model-runtime/src/providers/anthropic/index.ts
@@ -68,7 +68,13 @@ export class LobeAnthropicAI implements LobeRuntimeAI {
     this.client = new Anthropic({
       apiKey,
       baseURL,
-      defaultHeaders: { ...defaultHeaders, 'anthropic-beta': betaHeaders },
+      defaultHeaders: {
+        ...defaultHeaders,
+        'anthropic-beta': betaHeaders,
+        ...(baseURL === 'https://aihubmix.com' && {
+          'APP-Code': 'LobeHub',
+        }),
+      },
       ...res,
     });
     this.baseURL = this.client.baseURL;
diff --git a/packages/model-runtime/src/providers/google/index.ts b/packages/model-runtime/src/providers/google/index.ts
index b7811800bda2b..e0e505ee5283e 100644
--- a/packages/model-runtime/src/providers/google/index.ts
+++ b/packages/model-runtime/src/providers/google/index.ts
@@ -126,7 +126,15 @@ export class LobeGoogleAI implements LobeRuntimeAI {
     if (!apiKey) throw AgentRuntimeError.createError(AgentRuntimeErrorType.InvalidProviderAPIKey);
 
     const httpOptions = baseURL
-      ? ({ baseUrl: baseURL, headers: defaultHeaders } as HttpOptions)
+      ? ({
+          baseUrl: baseURL,
+          headers: {
+            ...defaultHeaders,
+            ...(baseURL === 'https://aihubmix.com/gemini' && {
+              'APP-Code': 'LobeHub',
+            }),
+          },
+        } as HttpOptions)
       : undefined;
 
     this.apiKey = apiKey;
