diff --git a/packages/business/const/src/index.ts b/packages/business/const/src/index.ts
index bafd6f087998d..388cff873c3c3 100644
--- a/packages/business/const/src/index.ts
+++ b/packages/business/const/src/index.ts
@@ -3,4 +3,4 @@ export * from './branding';
 export * from './llm';
 export * from './url';
 
-export const ENABLE_BUSINESS_FEATURES = false;
+export const ENABLE_BUSINESS_FEATURES = true;
