diff --git a/packages/model-bank/src/modelProviders/lmstudio.ts b/packages/model-bank/src/modelProviders/lmstudio.ts
index 2d433027954..786154dc9d1 100644
--- a/packages/model-bank/src/modelProviders/lmstudio.ts
+++ b/packages/model-bank/src/modelProviders/lmstudio.ts
@@ -17,7 +17,6 @@ const LMStudio: ModelProviderCard = {
       speed: 2,
       text: 'smooth',
     },
-    showApiKey: false,
     showModelFetcher: true,
   },
   url: 'https://lmstudio.ai',
diff --git a/src/envs/llm.ts b/src/envs/llm.ts
index 754b1d6f6f8..c49af7f12bd 100644
--- a/src/envs/llm.ts
+++ b/src/envs/llm.ts
@@ -81,6 +81,9 @@ export const getLLMConfig = () => {
       ENABLED_XINFERENCE: z.boolean(),
       XINFERENCE_API_KEY: z.string().optional(),
 
+      ENABLED_LMSTUDIO: z.boolean(),
+      LMSTUDIO_API_KEY: z.string().optional(),
+
       ENABLED_QINIU: z.boolean(),
       QINIU_API_KEY: z.string().optional(),
 
@@ -303,6 +306,9 @@ export const getLLMConfig = () => {
       ENABLED_XINFERENCE: !!process.env.XINFERENCE_API_KEY,
       XINFERENCE_API_KEY: process.env.XINFERENCE_API_KEY,
 
+      ENABLED_LMSTUDIO: !!process.env.LMSTUDIO_API_KEY,
+      LMSTUDIO_API_KEY: process.env.LMSTUDIO_API_KEY,
+
       ENABLED_QINIU: !!process.env.QINIU_API_KEY,
       QINIU_API_KEY: process.env.QINIU_API_KEY,
 
