diff --git a/packages/business/const/src/index.ts b/packages/business/const/src/index.ts
index 255ba3a04f52a..0e49091496924 100644
--- a/packages/business/const/src/index.ts
+++ b/packages/business/const/src/index.ts
@@ -3,5 +3,5 @@ export * from './branding';
 export * from './llm';
 export * from './url';
 
-export const ENABLE_BUSINESS_FEATURES = false;
+export const ENABLE_BUSINESS_FEATURES = true;
 export const APPLY_OSS_FEATURES_PATCH = true;
