diff --git a/next.config.ts b/next.config.ts
index da85eccb2e1..95b3b76d2a8 100644
--- a/next.config.ts
+++ b/next.config.ts
@@ -18,11 +18,6 @@ const nextConfig = defineConfig({
         ],
       }
     : undefined,
-  // Include ffmpeg binary for video webhook processing
-  // refs: https://github.com/vercel-labs/ffmpeg-on-vercel
-  outputFileTracingIncludes: {
-    '/api/webhooks/video/*': ['./node_modules/ffmpeg-static/ffmpeg'],
-  },
   webpack: (webpackConfig, context) => {
     const { dev } = context;
     if (!dev) {
