diff --git a/scripts/lint-ts.sh b/scripts/lint-ts.sh
index fe29d613986..ddeed16a638 100755
--- a/scripts/lint-ts.sh
+++ b/scripts/lint-ts.sh
@@ -1,5 +1,5 @@
 #!/bin/bash
 set -o pipefail
 
-eslint "{src,tests}/**/*.{js,jsx,ts,tsx}" --fix --concurrency=auto --prune-suppressions
-eslint "{src,tests}/**/*.{js,jsx,ts,tsx}" --concurrency=auto
\ No newline at end of file
+eslint "src/**/*.{js,jsx,ts,tsx}" "tests/**/*.{js,jsx,ts,tsx}" --fix --concurrency=auto --prune-suppressions
+eslint "src/**/*.{js,jsx,ts,tsx}" "tests/**/*.{js,jsx,ts,tsx}" --concurrency=auto
