diff --git a/app/src/config/about.ts b/app/src/config/about.ts
index 767b0b6c3df..e8634a2df67 100644
--- a/app/src/config/about.ts
+++ b/app/src/config/about.ts
@@ -26,11 +26,10 @@ export const about = {
     <div class="fn__flex-1">
         ${window.siyuan.languages.currentVer} v${Constants.SIYUAN_VERSION}
         <span id="isInsider"></span>
-        <div class="b3-label__text">${window.siyuan.languages.downloadLatestVer}</div>
     </div>
     <div class="fn__space"></div>
     <div class="fn__flex-center fn__size200 config__item-line">
-        <button id="checkUpdateBtn" class="b3-button b3-button--outline fn__block">
+        <button id="checkUpdateBtn" class="b3-button b3-button--outline fn__block" disabled>
             <svg><use xlink:href="#iconRefresh"></use></svg>${window.siyuan.languages.checkUpdate}
         </button>
     </div>
@@ -61,7 +60,7 @@ export const about = {
         <div class="b3-label__text">${window.siyuan.languages.googleAnalyticsTip}</div>
     </div>
     <div class="fn__space"></div>
-    <input class="b3-switch fn__flex-center" id="googleAnalytics" type="checkbox"${window.siyuan.config.system.disableGoogleAnalytics ? "" : " checked"}>
+    <input class="b3-switch fn__flex-center" id="googleAnalytics" type="checkbox"${window.siyuan.config.system.disableGoogleAnalytics ? " checked" : ""}>
 </label>
 <label class="b3-label fn__flex">
     <div class="fn__flex-1">
diff --git a/app/src/types/protyle.d.ts b/app/src/types/protyle.d.ts
index a000df00963..8f4a3eadfd6 100644
--- a/app/src/types/protyle.d.ts
+++ b/app/src/types/protyle.d.ts
@@ -512,7 +512,7 @@ interface IProtyle {
     gutter?: import("../protyle/gutter").Gutter,
     breadcrumb?: import("../protyle/breadcrumb").Breadcrumb,
     title?: import("../protyle/header/Title").Title,
-    background?: import("../protyle/header/background").Background,
+    background?: import("../protyle/header/Background").Background,
     contentElement?: HTMLElement,
     options: IProtyleOptions;
     lute?: Lute;
diff --git a/app/src/util/needSubscribe.ts b/app/src/util/needSubscribe.ts
index 6ba7848a6b2..7d635f5b891 100644
--- a/app/src/util/needSubscribe.ts
+++ b/app/src/util/needSubscribe.ts
@@ -19,5 +19,5 @@ export const needSubscribe = (tip = window.siyuan.languages._kernel[29]) => {
 };
 
 export const isPaidUser = () => {
-    return window.siyuan.user && (0 === window.siyuan.user.userSiYuanSubscriptionStatus || 1 === window.siyuan.user.userSiYuanOneTimePayStatus);
+    return true;
 };
diff --git a/kernel/conf/account.go b/kernel/conf/account.go
index 5e3b51fb8f9..d4f4e00dea0 100644
--- a/kernel/conf/account.go
+++ b/kernel/conf/account.go
@@ -23,7 +23,7 @@ type Account struct {
 
 func NewAccount() *Account {
 	return &Account{
-		DisplayTitle: true,
-		DisplayVIP:   true,
+		DisplayTitle: false,
+		DisplayVIP:   false,
 	}
 }
diff --git a/kernel/conf/appearance.go b/kernel/conf/appearance.go
index ebd1002acf9..1254085d8a3 100644
--- a/kernel/conf/appearance.go
+++ b/kernel/conf/appearance.go
@@ -43,7 +43,7 @@ func NewAppearance() *Appearance {
 		ThemeLight:          "daylight",
 		Icon:                "material",
 		CodeBlockThemeLight: "github",
-		CodeBlockThemeDark:  "base16/dracula",
+		CodeBlockThemeDark:  "github-dark-dimmed",
 		Lang:                "en_US",
 		CloseButtonBehavior: 0,
 		HideStatusBar:       false,
diff --git a/kernel/conf/sync.go b/kernel/conf/sync.go
index ac37900cce7..16f35d58af2 100644
--- a/kernel/conf/sync.go
+++ b/kernel/conf/sync.go
@@ -35,8 +35,8 @@ func NewSync() *Sync {
 		Enabled:             false,
 		Perception:          false,
 		Mode:                1,
-		GenerateConflictDoc: false,
-		Provider:            ProviderSiYuan,
+		GenerateConflictDoc: true,
+		Provider:            ProviderS3,
 	}
 }
 
diff --git a/kernel/conf/system.go b/kernel/conf/system.go
index 6d51b4ce102..e43d3184a73 100644
--- a/kernel/conf/system.go
+++ b/kernel/conf/system.go
@@ -52,7 +52,7 @@ func NewSystem() *System {
 		Name:               util.GetDeviceName(),
 		KernelVersion:      util.Ver,
 		NetworkProxy:       &NetworkProxy{},
-		DownloadInstallPkg: true,
+		DownloadInstallPkg: false,
 	}
 }
 
diff --git a/kernel/model/conf.go b/kernel/model/conf.go
index 36bad7c97a3..0453d907dd7 100644
--- a/kernel/model/conf.go
+++ b/kernel/model/conf.go
@@ -847,17 +847,7 @@ func IsSubscriber() bool {
 }
 
 func IsPaidUser() bool {
-	// S3/WebDAV data sync and backup are available for a fee https://github.com/siyuan-note/siyuan/issues/8780
-
-	if IsSubscriber() {
-		return true
-	}
-
-	u := Conf.GetUser()
-	if nil == u {
-		return false
-	}
-	return 1 == u.UserSiYuanOneTimePayStatus
+	return true
 }
 
 const (
diff --git a/kernel/model/mount.go b/kernel/model/mount.go
index 367c5d4dcc0..dde0000ce6b 100644
--- a/kernel/model/mount.go
+++ b/kernel/model/mount.go
@@ -220,11 +220,6 @@ func Mount(boxID string) (alreadyMount bool, err error) {
 		}
 
 		task.AppendAsyncTaskWithDelay(task.PushMsg, 3*time.Second, util.PushErrMsg, Conf.Language(52), 7000)
-		go func() {
-			// 每次打开帮助文档时自动检查版本更新并提醒 https://github.com/siyuan-note/siyuan/issues/5057
-			time.Sleep(time.Second * 10)
-			CheckUpdate(true)
-		}()
 	}
 
 	if !gulu.File.IsDir(localPath) {
diff --git a/kernel/util/working.go b/kernel/util/working.go
index 60c57d8fbd9..5eef08e1aab 100644
--- a/kernel/util/working.go
+++ b/kernel/util/working.go
@@ -195,7 +195,8 @@ func SetBooted() {
 
 var (
 	HomeDir, _    = gulu.OS.Home()
-	WorkingDir, _ = os.Getwd()
+	ExecPath, _   = os.Executable()
+	WorkingDir    = filepath.Dir(ExecPath)
 
 	WorkspaceDir       string        // 工作空间目录路径
 	WorkspaceName      string        // 工作空间名称
