diff --git a/app/src/config/tabs/aboutTab.ts b/app/src/config/tabs/aboutTab.ts
index aee4faf3c78..64e8e750f98 100644
--- a/app/src/config/tabs/aboutTab.ts
+++ b/app/src/config/tabs/aboutTab.ts
@@ -60,7 +60,7 @@ const genAboutVersionHtml = (): string => {
     </div>
     <div class="fn__space"></div>
     <div class="fn__flex-center fn__size200">
-        <button id="checkUpdateBtn" class="b3-button b3-button--outline fn__block">
+        <button id="checkUpdateBtn" class="b3-button b3-button--outline fn__block" disabled>
             <svg><use xlink:href="#iconRefresh"></use></svg>${window.siyuan.languages.checkUpdate}
         </button>
     </div>
diff --git a/app/src/util/needSubscribe.ts b/app/src/util/needSubscribe.ts
index 8681b9e848c..83d30f8ed09 100644
--- a/app/src/util/needSubscribe.ts
+++ b/app/src/util/needSubscribe.ts
@@ -20,5 +20,5 @@ export const needSubscribe = (tip = window.siyuan.languages._kernel[29]) => {
  * 判断是否可以使用第三方同步
  */
 export const isPaidUser = () => {
-    return window.siyuan.user && (0 === window.siyuan.user.userSiYuanSubscriptionStatus || 1 === window.siyuan.user.userSiYuanOneTimePayStatus);
+    return true;
 };
diff --git a/kernel/conf/account.go b/kernel/conf/account.go
index 5e3b51fb8f9..d4f4e00dea0 100644
--- a/kernel/conf/account.go
+++ b/kernel/conf/account.go
@@ -23,7 +23,7 @@ type Account struct {
 
 func NewAccount() *Account {
 	return &Account{
-		DisplayTitle: true,
-		DisplayVIP:   true,
+		DisplayTitle: false,
+		DisplayVIP:   false,
 	}
 }
diff --git a/kernel/conf/system.go b/kernel/conf/system.go
index d01f5885fb0..b5e5c44fdba 100644
--- a/kernel/conf/system.go
+++ b/kernel/conf/system.go
@@ -77,7 +77,7 @@ func NewSystem() *System {
 		Name:               util.GetDeviceName(),
 		KernelVersion:      util.Ver,
 		NetworkProxy:       &NetworkProxy{},
-		DownloadInstallPkg: true,
+		DownloadInstallPkg: false,
 		UpdateChannel:      UpdateChannelStable,
 	}
 }
diff --git a/kernel/model/conf.go b/kernel/model/conf.go
index 76f4283768d..54315419141 100644
--- a/kernel/model/conf.go
+++ b/kernel/model/conf.go
@@ -1210,15 +1210,7 @@ func IsSubscriber() bool {
 }
 
 func IsPaidUser() bool {
-	if IsSubscriber() {
-		return true
-	}
-
-	u := Conf.GetUser()
-	if nil == u {
-		return false
-	}
-	return 1 == u.UserSiYuanOneTimePayStatus
+	return true
 }
 
 const (
diff --git a/kernel/model/updater.go b/kernel/model/updater.go
index 8b9cfe0023c..a594dd162d6 100644
--- a/kernel/model/updater.go
+++ b/kernel/model/updater.go
@@ -33,7 +33,6 @@ import (
 	"github.com/imroc/req/v3"
 	"github.com/siyuan-note/logging"
 	"github.com/siyuan-note/siyuan/kernel/util"
-	"golang.org/x/mod/semver"
 )
 
 func getNewVerInstallPkgPath() string {
@@ -260,7 +259,7 @@ func pushNewVersionNotification(release *updateRelease) {
 }
 
 func isVersionUpToDate(releaseVer string) bool {
-	return semver.Compare("v"+releaseVer, "v"+util.Ver) <= 0
+	return true
 }
 
 // skipInstallPkgPlatformCached 缓存平台相关判断，-1 未初始化，0 表示不跳过，1 表示跳过
diff --git a/kernel/util/working.go b/kernel/util/working.go
index 1875011b263..38708595081 100644
--- a/kernel/util/working.go
+++ b/kernel/util/working.go
@@ -264,7 +264,8 @@ func SetBooted() {
 
 var (
 	HomeDir, _    = gulu.OS.Home()
-	WorkingDir, _ = os.Getwd()
+	ExecPath, _   = os.Executable()
+	WorkingDir    = filepath.Dir(ExecPath)
 
 	WorkspaceDir       string        // 工作空间目录路径
 	WorkspaceName      string        // 工作空间名称
