diff --git a/app/src/config/about.ts b/app/src/config/about.ts
index 767b0b6c3df..e8634a2df67 100644
--- a/app/src/config/about.ts
+++ b/app/src/config/about.ts
@@ -61,7 +60,7 @@ export const about = {
         <div class="b3-label__text">${window.siyuan.languages.googleAnalyticsTip}</div>
     </div>
     <div class="fn__space"></div>
-    <input class="b3-switch fn__flex-center" id="googleAnalytics" type="checkbox"${window.siyuan.config.system.disableGoogleAnalytics ? "" : " checked"}>
+    <input class="b3-switch fn__flex-center" id="googleAnalytics" type="checkbox"${window.siyuan.config.system.disableGoogleAnalytics ? " checked" : ""}>
 </label>
 <label class="b3-label fn__flex">
     <div class="fn__flex-1">
diff --git a/kernel/conf/appearance.go b/kernel/conf/appearance.go
index ebd1002acf9..1254085d8a3 100644
--- a/kernel/conf/appearance.go
+++ b/kernel/conf/appearance.go
@@ -43,7 +43,7 @@ func NewAppearance() *Appearance {
 		ThemeLight:          "daylight",
 		Icon:                "material",
 		CodeBlockThemeLight: "github",
-		CodeBlockThemeDark:  "base16/dracula",
+		CodeBlockThemeDark:  "github-dark-dimmed",
 		Lang:                "en_US",
 		CloseButtonBehavior: 0,
 		HideStatusBar:       false,
