diff --git a/kernel/util/working.go b/kernel/util/working.go
index c79278c1cfd..38270d712cf 100644
--- a/kernel/util/working.go
+++ b/kernel/util/working.go
@@ -195,7 +195,8 @@ func SetBooted() {
 
 var (
 	HomeDir, _    = gulu.OS.Home()
-	WorkingDir, _ = os.Getwd()
+	ExecPath, _   = os.Executable()
+	WorkingDir    = filepath.Dir(ExecPath)
 
 	WorkspaceDir       string        // 工作空间目录路径
 	WorkspaceName      string        // 工作空间名称
