diff --git a/app/src/constants.ts b/app/src/constants.ts
index 4064b32bf45..fba8a54e6d7 100644
--- a/app/src/constants.ts
+++ b/app/src/constants.ts
@@ -559,12 +559,6 @@ export abstract class Constants {
                     show: false,
                     icon: "iconAlignCenter",
                     hotkeyLangId: "outline",
-                }, {
-                    type: "inbox",
-                    size: {width: 320, height: 0},
-                    show: false,
-                    icon: "iconInbox",
-                    hotkeyLangId: "inbox",
                 }], [{
                     type: "bookmark",
                     size: {width: 232, height: 0},
diff --git a/app/src/layout/dock/index.ts b/app/src/layout/dock/index.ts
index f4918a69baf..4bc11ae3e16 100644
--- a/app/src/layout/dock/index.ts
+++ b/app/src/layout/dock/index.ts
@@ -11,7 +11,6 @@ import {Graph} from "./Graph";
 import {Model} from "../Model";
 import {saveLayout, setPanelFocus} from "../util";
 import {getDockByType, resizeTabs} from "../tabUtil";
-import {Inbox} from "./Inbox";
 import {Protyle} from "../../protyle";
 import {Backlink} from "./Backlink";
 import {resetFloatDockSize} from "./util";
@@ -335,7 +334,7 @@ export class Dock {
                 let minSize = 227;
                 Array.from(this.layout.element.querySelectorAll(".file-tree")).find((item) => {
                     if (item.classList.contains("sy__backlink") || item.classList.contains("sy__graph")
-                        || item.classList.contains("sy__globalGraph") || item.classList.contains("sy__inbox")) {
+                        || item.classList.contains("sy__globalGraph")) {
                         if (!item.classList.contains("fn__none") && !hasClosestByClassName(item, "fn__none")) {
                             minSize = 320;
                             return true;
@@ -638,13 +637,6 @@ export class Dock {
                             }
                         });
                         break;
-                    case "inbox":
-                        tab = new Tab({
-                            callback: (tab: Tab) => {
-                                tab.addModel(new Inbox(this.app, tab));
-                            }
-                        });
-                        break;
                     default:
                         tab = new Tab({
                             callback: (tab: Tab) => {
