diff --git a/kernel/job/cron.go b/kernel/job/cron.go
index 7255631f2a2..5ee6884c02c 100644
--- a/kernel/job/cron.go
+++ b/kernel/job/cron.go
@@ -43,7 +43,7 @@ func StartCron() {
 	go every(30*time.Second, model.FlushAssetsTextsJob)
 	go every(30*time.Second, model.HookDesktopUIProcJob)
 	go every(24*time.Hour, model.AutoPurgeRepoJob)
-	go every(30*time.Minute, model.AutoCheckMicrosoftDefender)
+	//go every(30*time.Minute, model.AutoCheckMicrosoftDefender)
 }
 
 func every(interval time.Duration, f func()) {
