diff --git a/app/src/config/about.ts b/app/src/config/about.ts
index f876207c489..59762952d45 100644
--- a/app/src/config/about.ts
+++ b/app/src/config/about.ts
@@ -27,11 +27,10 @@ export const about = {
     <div class="fn__flex-1">
         ${window.siyuan.languages.currentVer} v${Constants.SIYUAN_VERSION}
         <span id="isInsider"></span>
-        <div class="b3-label__text">${window.siyuan.languages.downloadLatestVer}</div>
     </div>
     <div class="fn__space"></div>
     <div class="fn__flex-center fn__size200 config__item-line">
-        <button id="checkUpdateBtn" class="b3-button b3-button--outline fn__block">
+        <button id="checkUpdateBtn" class="b3-button b3-button--outline fn__block" disabled>
             <svg><use xlink:href="#iconRefresh"></use></svg>${window.siyuan.languages.checkUpdate}
         </button>
     </div>
diff --git a/kernel/conf/appearance.go b/kernel/conf/appearance.go
index 8be9e428a08..aae7a69580c 100644
--- a/kernel/conf/appearance.go
+++ b/kernel/conf/appearance.go
@@ -46,7 +46,7 @@ func NewAppearance() *Appearance {
 		ThemeLight:          "daylight",
 		Icon:                "material",
 		CodeBlockThemeLight: "github",
-		CodeBlockThemeDark:  "base16/dracula",
+		CodeBlockThemeDark:  "github-dark-dimmed",
 		Lang:                "en_US",
 		CloseButtonBehavior: 0,
 		HideStatusBar:       false,
diff --git a/kernel/conf/system.go b/kernel/conf/system.go
index 58b0475a23d..614602a8df3 100644
--- a/kernel/conf/system.go
+++ b/kernel/conf/system.go
@@ -54,7 +54,7 @@ func NewSystem() *System {
 		Name:               util.GetDeviceName(),
 		KernelVersion:      util.Ver,
 		NetworkProxy:       &NetworkProxy{},
-		DownloadInstallPkg: true,
+		DownloadInstallPkg: false,
 	}
 }
 
diff --git a/kernel/model/updater.go b/kernel/model/updater.go
index e29a756371e..260ff9836c0 100644
--- a/kernel/model/updater.go
+++ b/kernel/model/updater.go
@@ -301,7 +301,7 @@ func CheckUpdate(showMsg bool) {
 }
 
 func isVersionUpToDate(releaseVer string) bool {
-	return semver.Compare("v"+releaseVer, "v"+util.Ver) <= 0
+	return true
 }
 
 // skipInstallPkgPlatformCached 缓存平台相关判断，-1 未初始化，0 表示不跳过，1 表示跳过
diff --git a/kernel/util/working.go b/kernel/util/working.go
index 5244e9c4bec..03c12da8f9c 100644
--- a/kernel/util/working.go
+++ b/kernel/util/working.go
@@ -228,7 +228,8 @@ func SetBooted() {
 
 var (
 	HomeDir, _    = gulu.OS.Home()
-	WorkingDir, _ = os.Getwd()
+	ExecPath, _   = os.Executable()
+	WorkingDir    = filepath.Dir(ExecPath)
 
 	WorkspaceDir       string        // 工作空间目录路径
 	WorkspaceName      string        // 工作空间名称
