diff --git a/app/src/config/about.ts b/app/src/config/about.ts
index 767b0b6c3df..e8634a2df67 100644
--- a/app/src/config/about.ts
+++ b/app/src/config/about.ts
@@ -26,11 +26,10 @@ export const about = {
     <div class="fn__flex-1">
         ${window.siyuan.languages.currentVer} v${Constants.SIYUAN_VERSION}
         <span id="isInsider"></span>
-        <div class="b3-label__text">${window.siyuan.languages.downloadLatestVer}</div>
     </div>
     <div class="fn__space"></div>
     <div class="fn__flex-center fn__size200 config__item-line">
-        <button id="checkUpdateBtn" class="b3-button b3-button--outline fn__block">
+        <button id="checkUpdateBtn" class="b3-button b3-button--outline fn__block" disabled>
             <svg><use xlink:href="#iconRefresh"></use></svg>${window.siyuan.languages.checkUpdate}
         </button>
     </div>
diff --git a/kernel/conf/system.go b/kernel/conf/system.go
index 6d51b4ce102..e43d3184a73 100644
--- a/kernel/conf/system.go
+++ b/kernel/conf/system.go
@@ -52,7 +52,7 @@ func NewSystem() *System {
 		Name:               util.GetDeviceName(),
 		KernelVersion:      util.Ver,
 		NetworkProxy:       &NetworkProxy{},
-		DownloadInstallPkg: true,
+		DownloadInstallPkg: false,
 	}
 }
diff --git a/kernel/model/mount.go b/kernel/model/mount.go
index cc55d454313..1bb7df45e1d 100644
--- a/kernel/model/mount.go
+++ b/kernel/model/mount.go
@@ -220,11 +220,6 @@ func Mount(boxID string) (alreadyMount bool, err error) {
 		}
 
 		task.AppendAsyncTaskWithDelay(task.PushMsg, 3*time.Second, util.PushErrMsg, Conf.Language(52), 7000)
-		go func() {
-			// 每次打开帮助文档时自动检查版本更新并提醒 https://github.com/siyuan-note/siyuan/issues/5057
-			time.Sleep(time.Second * 10)
-			CheckUpdate(true)
-		}()
 	}
 
 	if !gulu.File.IsDir(localPath) {
