diff --git a/app/src/layout/status.ts b/app/src/layout/status.ts
index 759a4220642..ae71c9830f3 100644
--- a/app/src/layout/status.ts
+++ b/app/src/layout/status.ts
@@ -92,20 +92,6 @@ export const initStatus = (isWindow = false) => {
                     }
                 }).element);
                 /// #endif
-                window.siyuan.menus.menu.append(new MenuItem({
-                    label: window.siyuan.languages["_trayMenu"].officialWebsite,
-                    icon: "iconSiYuan",
-                    click: () => {
-                        window.open("https://b3log.org/siyuan");
-                    }
-                }).element);
-                window.siyuan.menus.menu.append(new MenuItem({
-                    label: window.siyuan.languages["_trayMenu"].openSource,
-                    icon: "iconGithub",
-                    click: () => {
-                        window.open("https://github.com/siyuan-note/siyuan");
-                    }
-                }).element);
                 const rect = target.getBoundingClientRect();
                 window.siyuan.menus.menu.popup({x: rect.right, y: rect.top, isLeft: true});
                 event.stopPropagation();
