diff --git a/dockerfiles/uwsgi.customized.ini b/dockerfiles/uwsgi.customized.ini
new file mode 100644
index 00000000000..10bfe1fc1ef
--- /dev/null
+++ b/dockerfiles/uwsgi.customized.ini
@@ -0,0 +1,33 @@
+[uwsgi]
+add-header = Connection: close
+auto-procname = true
+buffer-size = 8192
+chdir = /usr/local/searxng/searx/
+chmod-socket = 666
+die-on-term = true
+disable-logging = true
+enable-threads = true
+gid = root
+harakiri = 60
+lazy-apps = true
+log-4xx = true
+log-5xx = true
+master = true
+module = searx.webapp
+need-app = true
+offload-threads = %k
+plugin = python3
+py-callos-afterfork = true
+pythonpath = /usr/local/searxng/
+reload-on-rss = 4096
+single-interpreter = true
+static-expires = /* 86400
+static-gzip-all = true
+static-map = /static=/usr/local/searxng/searx/static
+strict = true
+threads = %k
+thunder-lock = true
+uid = root
+vacuum = true
+worker-reload-mercy = 60
+workers = %k
