diff --git a/dockerfiles/uwsgi.ini b/dockerfiles/uwsgi.ini
index 812716853ac..d79d51d1d71 100644
--- a/dockerfiles/uwsgi.ini
+++ b/dockerfiles/uwsgi.ini
@@ -1,15 +1,15 @@
 [uwsgi]
 # Who will run the code
-uid = searxng
-gid = searxng
+uid = root
+gid = root
 
 # Number of workers (usually CPU count)
 # default value: %k (= number of CPU core, see Dockerfile)
-workers = $(UWSGI_WORKERS)
+workers = %k
 
 # Number of threads per worker
 # default value: 4 (see Dockerfile)
-threads = $(UWSGI_THREADS)
+threads = 4
 
 # The right granted on the created socket
 chmod-socket = 666
