diff --git a/dockerfiles/uwsgi.customized.ini b/dockerfiles/uwsgi.customized.ini
new file mode 100644
index 00000000000..676b57f6507
--- /dev/null
+++ b/dockerfiles/uwsgi.customized.ini
@@ -0,0 +1,31 @@
+[uwsgi]
+auto-procname = true
+buffer-size = 16384
+chdir = /usr/local/searxng/searx/
+chmod-socket = 666
+die-on-term = true
+disable-logging = true
+enable-threads = true
+gid = root
+harakiri = 60
+lazy-apps = true
+master = true
+max-worker-lifetime = 3600
+max-worker-lifetime-delta = 60
+module = searx.webapp
+need-app = true
+offload-threads = %k
+plugin = python3
+py-callos-afterfork = true
+pythonpath = /usr/local/searxng/
+reload-on-rss = 8192
+single-interpreter = true
+static-gzip-all = true
+static-map = /static=/usr/local/searxng/searx/static
+strict = false
+threads = %k
+thunder-lock = true
+uid = root
+vacuum = true
+worker-reload-mercy = 60
+workers = %k
