diff --git a/searx/version.py b/searx/version.py
index 8f9ef3a84ab..3048e0dde66 100644
--- a/searx/version.py
+++ b/searx/version.py
@@ -12,8 +12,8 @@
 VERSION_STRING: str = "1.0.0"
 VERSION_TAG: str = "1.0.0"
 DOCKER_TAG: str = "1.0.0"
-GIT_URL: str = "unknown"
-GIT_BRANCH: str = "unknown"
+GIT_URL: str = "https://github.com/hezhijie0327/searxng"
+GIT_BRANCH: str = "master"
 
 logger = logging.getLogger("searx")
 
@@ -93,15 +93,6 @@ def get_information() -> tuple[str, str, str, str, str]:
     git_url: str = GIT_URL
     git_branch: str = GIT_BRANCH
 
-    try:
-        version_string, version_tag, docker_tag = get_git_version()
-    except subprocess.CalledProcessError as ex:
-        logger.error("Error while getting the version: %s", ex.stderr)
-    try:
-        git_url, git_branch = get_git_url_and_branch()
-    except subprocess.CalledProcessError as ex:
-        logger.error("Error while getting the git URL & branch: %s", ex.stderr)
-
     return version_string, version_tag, docker_tag, git_url, git_branch
 
 
