diff --git a/searx/engines/marginalia.py b/searx/engines/marginalia.py
index 76090d8cd03..47e4b1ac5bf 100644
--- a/searx/engines/marginalia.py
+++ b/searx/engines/marginalia.py
@@ -170,6 +170,13 @@ def init(_: dict[str, t.Any]):
         logger.error("missing api_key: see https://about.marginalia-search.com/article/api")
         return False
 
+    if filter_name:
+        filter_names: list[str] = _get_filter_names()
+        if filter_name not in filter_names:
+            logger.error(f"invalid value for filter_name: '{filter_name}'")
+            return False
+
+    """
     if api_key == "public":
         logger.error("invalid api_key (%s): see https://about.marginalia-search.com/article/api", api_key)
     elif filter_name:
@@ -177,5 +184,6 @@ def init(_: dict[str, t.Any]):
         if filter_name not in filter_names:
             logger.error(f"invalid value for filter_name: '{filter_name}'")
             return False
+    """
 
     return True
