diff --git a/searx/engines/marginalia.py b/searx/engines/marginalia.py
index 76090d8cd03..9d942812f8b 100644
--- a/searx/engines/marginalia.py
+++ b/searx/engines/marginalia.py
@@ -170,6 +170,7 @@ def init(_: dict[str, t.Any]):
         logger.error("missing api_key: see https://about.marginalia-search.com/article/api")
         return False
 
+    """
     if api_key == "public":
         logger.error("invalid api_key (%s): see https://about.marginalia-search.com/article/api", api_key)
     elif filter_name:
@@ -177,5 +178,6 @@ def init(_: dict[str, t.Any]):
         if filter_name not in filter_names:
             logger.error(f"invalid value for filter_name: '{filter_name}'")
             return False
+    """
 
     return True
