diff --git a/searx/engines/marginalia.py b/searx/engines/marginalia.py
index 63f411cf031..bccb25b03b0 100644
--- a/searx/engines/marginalia.py
+++ b/searx/engines/marginalia.py
@@ -119,7 +119,4 @@ def init(engine_settings: dict[str, t.Any]):
         logger.error("missing api_key: see https://about.marginalia-search.com/article/api")
         return False
 
-    if _api_key == "public":
-        logger.error("invalid api_key (%s): see https://about.marginalia-search.com/article/api", api_key)
-
     return True
