From 2998b459f20af1f9e6606cd869cda61ba0c8cae3 Mon Sep 17 00:00:00 2001
From: Zhijie He <hezhijie0327@hotmail.com>
Date: Tue, 26 May 2026 22:09:08 +0800
Subject: [PATCH] fix: fix sqlitedb init error

---
 searx/sqlitedb.py | 50 +++++++++++++++++++++++++----------------------
 1 file changed, 27 insertions(+), 23 deletions(-)

diff --git a/searx/sqlitedb.py b/searx/sqlitedb.py
index a4f4983..10e653d 100644
--- a/searx/sqlitedb.py
+++ b/searx/sqlitedb.py
@@ -195,6 +195,7 @@ class SQLiteAppl(abc.ABC):
         self.db_url: str = db_url
         self.properties: SQLiteProperties = SQLiteProperties(db_url)
         self._init_done: bool = False
+        self._init_lock: threading.Lock = threading.Lock()
         self._compatibility()
         # atexit.register(self.tear_down)
 
@@ -325,24 +326,25 @@ class SQLiteAppl(abc.ABC):
         already been carried out in the past, `False` is returned.
         """
 
-        if self._init_done:
-            return False
-        self._init_done = True
+        with self._init_lock:
+            if self._init_done:
+                return False
+            self._init_done = True
 
-        logger.debug("init DB: %s", self.db_url)
-        self.properties.init(conn)
+            logger.debug("init DB: %s", self.db_url)
+            self.properties.init(conn)
 
-        ver = self.properties("DB_SCHEMA")
-        if ver is None:
-            with conn:
-                self.create_schema(conn)
-        else:
-            ver = int(ver)
-            if ver != self.DB_SCHEMA:
-                raise sqlite3.DatabaseError("Expected DB schema v%s, DB schema is v%s" % (self.DB_SCHEMA, ver))
-            logger.debug("DB_SCHEMA = %s", ver)
+            ver = self.properties("DB_SCHEMA")
+            if ver is None:
+                with conn:
+                    self.create_schema(conn)
+            else:
+                ver = int(ver)
+                if ver != self.DB_SCHEMA:
+                    raise sqlite3.DatabaseError("Expected DB schema v%s, DB schema is v%s" % (self.DB_SCHEMA, ver))
+                logger.debug("DB_SCHEMA = %s", ver)
 
-        return True
+            return True
 
     def create_schema(self, conn: sqlite3.Connection):
 
@@ -399,19 +401,21 @@ CREATE TABLE IF NOT EXISTS properties (
 
         self.db_url: str = db_url
         self._init_done: bool = False
+        self._init_lock: threading.Lock = threading.Lock()
         self._compatibility()
 
     def init(self, conn: sqlite3.Connection) -> bool:
         """Initializes DB schema of the properties in the DB."""
 
-        if self._init_done:
-            return False
-        self._init_done = True
-        logger.debug("init properties of DB: %s", self.db_url)
-        res = conn.execute(self.SQL_TABLE_EXISTS)
-        if res.fetchone() is None:  # DB schema needs to be be created
-            self.create_schema(conn)
-        return True
+        with self._init_lock:
+            if self._init_done:
+                return False
+            self._init_done = True
+            logger.debug("init properties of DB: %s", self.db_url)
+            res = conn.execute(self.SQL_TABLE_EXISTS)
+            if res.fetchone() is None:  # DB schema needs to be be created
+                self.create_schema(conn)
+            return True
 
     def __call__(self, name: str, default: t.Any = None) -> t.Any:
         """Returns the value of the property ``name`` or ``default`` if property
-- 
2.54.0

