diff --git a/searx/engines/openalex.py b/searx/engines/openalex.py
index 7ff6ec58249..7d82f6e1f46 100644
--- a/searx/engines/openalex.py
+++ b/searx/engines/openalex.py
@@ -119,6 +119,7 @@
 # engines: - name: openalex; engine: openalex; mailto: "[email protected]"
 mailto = ""
 
+api_key = ""
 
 def request(query: str, params: "OnlineParams") -> None:
     # Build OpenAlex query using search parameter and paging
@@ -146,6 +147,9 @@ def request(query: str, params: "OnlineParams") -> None:
     if isinstance(mailto, str) and mailto != "":
         args["mailto"] = mailto
 
+    if isinstance(api_key, str) and api_key != "":
+        args["api_key"] = api_key
+
     params["url"] = f"{search_url}?{urlencode(args)}"
 
 
