diff --git a/searx/autocomplete.py b/searx/autocomplete.py
index 8a4f0a66a92..2aca30cb5a2 100644
--- a/searx/autocomplete.py
+++ b/searx/autocomplete.py
@@ -149,6 +149,21 @@ def mwmbl(query, _lang):
     return [result for result in results if not result.startswith("go: ") and not result.startswith("search: ")]
 
 
+def naver(query, _lang):
+    # Naver search autocompleter
+    url = f"https://ac.search.naver.com/nx/ac?{urlencode({'q': query, 'r_format': 'json', 'st': 0})}"
+    response = get(url)
+
+    results = []
+
+    if response.ok:
+        data = response.json()
+        if 'items' in data and data['items']:
+            for item in data['items'][0]:
+                results.append(item[0])
+    return results
+
+
 def qihu360search(query, _lang):
     # 360Search search autocompleter
     url = f"https://sug.so.360.cn/suggest?{urlencode({'format': 'json', 'word': query})}"
@@ -164,6 +179,21 @@ def qihu360search(query, _lang):
     return results
 
 
+def quark(query, _lang):
+    # Quark search autocompleter
+    url = f"https://sugs.m.sm.cn/web?{urlencode({'q': query})}"
+    response = get(url)
+
+    results = []
+
+    if response.ok:
+        data = response.json()
+        if 'r' in data:
+            for item in data['r']:
+                results.append(item['w'])
+    return results
+
+
 def seznam(query, _lang):
     # seznam search autocompleter
     url = 'https://suggest.seznam.cz/fulltext/cs?{query}'
@@ -286,6 +316,8 @@ def yandex(query, _lang):
     'duckduckgo': duckduckgo,
     'google': google_complete,
     'mwmbl': mwmbl,
+    'naver': naver,
+    'quark': quark,
     'qwant': qwant,
     'seznam': seznam,
     'sogou': sogou,
