diff --git a/searx/engines/chp.py b/searx/engines/chp.py
new file mode 100644
index 00000000000..52adcd4a0a1
--- /dev/null
+++ b/searx/engines/chp.py
@@ -0,0 +1,53 @@
+# SPDX-License-Identifier: AGPL-3.0-or-later
+"""ChP (Chinese Pharmacopoeia) engine for searxng"""
+
+from json import loads, dumps
+from datetime import datetime
+
+from searx.utils import html_to_text
+
+about = {
+    "website": 'https://ydz.chp.org.cn/',
+    "wikidata_id": 'Q111049218',
+    "use_official_api": False,
+    "require_api_key": False,
+    "results": 'JSON',
+    "language": "zh",
+}
+
+paging = True
+categories = ['dictionaries']
+results_per_page = 10
+
+base_url = "https://ydz.chp.org.cn"
+
+
+def request(query, params):
+
+    params['url'] = f'{base_url}/front-api/search'
+
+    params['headers']['Content-Type'] = 'application/json'
+    params['method'] = 'POST'
+
+    params['data'] = dumps({"keyword": query, "pageSize": results_per_page, "pageNum": params["pageno"]}).encode(
+        'utf-8'
+    )
+
+    return params
+
+
+def response(resp):
+    results = []
+    json = loads(resp.text)
+
+    for item in json.get('data', {}).get('list', []):
+        results.append(
+            {
+                'content': html_to_text(f"{item['bookName']} | {item['pageNum']} | {item['textContent']}"),
+                'title': html_to_text(f"{item['title']} | {item['etitle']} | {item['pinyinTitle']}"),
+                'url': f'{base_url}/#/item?bookId={item["bookId"]}&entryId={item["id"]}',
+                "publishedDate": datetime.strptime(item["createTime"], "%Y-%m-%d %H:%M:%S"),
+            }
+        )
+
+    return results
