diff --git a/searx/engines/pubmed.py b/searx/engines/pubmed.py
index be934cdc8d9..6a69c874bd2 100644
--- a/searx/engines/pubmed.py
+++ b/searx/engines/pubmed.py
@@ -29,8 +29,10 @@
 
 categories = ['science', 'scientific publications']
 
+api_key = ''
 base_url = (
-    'https://eutils.ncbi.nlm.nih.gov/entrez/eutils/esearch.fcgi' + '?db=pubmed&{query}&retstart={offset}&retmax={hits}'
+    'https://eutils.ncbi.nlm.nih.gov/entrez/eutils/esearch.fcgi'
+    + '?db=pubmed&{query}&retstart={offset}&retmax={hits}&api_key={api_key}'
 )
 
 # engine dependent config
@@ -43,6 +45,7 @@ def request(query, params):
     offset = (params['pageno'] - 1) * number_of_results
 
     string_args = {
+        'api_key': api_key,
         'query': urlencode({'term': query}),
         'offset': offset,
         'hits': number_of_results,
@@ -58,7 +61,8 @@ def response(resp):  # pylint: disable=too-many-locals
 
     # First retrieve notice of each result
     pubmed_retrieve_api_url = (
-        'https://eutils.ncbi.nlm.nih.gov/entrez/eutils/efetch.fcgi?' + 'db=pubmed&retmode=xml&id={pmids_string}'
+        'https://eutils.ncbi.nlm.nih.gov/entrez/eutils/efetch.fcgi?'
+        + 'db=pubmed&retmode=xml&id={pmids_string}&api_key={api_key}'
     )
 
     pmids_results = etree.XML(resp.content)
@@ -68,7 +72,10 @@ def response(resp):  # pylint: disable=too-many-locals
     for item in pmids:
         pmids_string += item.text + ','
 
-    retrieve_notice_args = {'pmids_string': pmids_string}
+    retrieve_notice_args = {
+        'api_key': api_key,
+        'pmids_string': pmids_string,
+    }
 
     retrieve_url_encoded = pubmed_retrieve_api_url.format(**retrieve_notice_args)
 
