diff --git a/searx/engines/pubmed.py b/searx/engines/pubmed.py
index 6fcfaa9a308..ae3a86e0c35 100644
--- a/searx/engines/pubmed.py
+++ b/searx/engines/pubmed.py
@@ -54,6 +54,7 @@
 
 categories = ["science", "scientific publications"]
 
+api_key = ''
 eutils_api = "https://eutils.ncbi.nlm.nih.gov/entrez/eutils"
 
 # engine dependent config
@@ -71,6 +72,8 @@ def request(query: str, params: "OnlineParams") -> None:
             "hits": number_of_results,
         }
     )
+    if api_key:
+       args['api_key'] = api_key
     esearch_url = f"{eutils_api}/esearch.fcgi?{args}"
     # DTD: https://eutils.ncbi.nlm.nih.gov/eutils/dtd/20060628/esearch.dtd
     esearch_resp: "SXNG_Response" = get(esearch_url)
@@ -85,6 +88,8 @@ def request(query: str, params: "OnlineParams") -> None:
             "id": ",".join(pmids),
         }
     )
+    if api_key:
+       args['api_key'] = api_key
     efetch_url = f"{eutils_api}/efetch.fcgi?{args}"
     params["url"] = efetch_url
 
@@ -148,4 +153,4 @@ def _field_txt(xml: ElementType, xpath_str: str) -> str:
                 publishedDate=pub_date,
             )
         )
-    return res
+    return res
\ No newline at end of file
