diff --git a/searx/engines/pubmed.py b/searx/engines/pubmed.py
index 6fcfaa9a308..a188eb8b977 100644
--- a/searx/engines/pubmed.py
+++ b/searx/engines/pubmed.py
@@ -22,7 +22,7 @@
 import typing as t
 
 from datetime import datetime
-from urllib.parse import urlencode
+from urllib.parse import urlencode, quote
 
 from lxml import etree
 
@@ -54,6 +54,7 @@
 
 categories = ["science", "scientific publications"]
 
+api_key = ''
 eutils_api = "https://eutils.ncbi.nlm.nih.gov/entrez/eutils"
 
 # engine dependent config
@@ -72,6 +73,8 @@ def request(query: str, params: "OnlineParams") -> None:
         }
     )
     esearch_url = f"{eutils_api}/esearch.fcgi?{args}"
+    if api_key:
+        esearch_url += f"&api_key={quote(api_key)}"
     # DTD: https://eutils.ncbi.nlm.nih.gov/eutils/dtd/20060628/esearch.dtd
     esearch_resp: "SXNG_Response" = get(esearch_url)
     pmids_results = etree.XML(esearch_resp.content)
@@ -86,6 +89,8 @@ def request(query: str, params: "OnlineParams") -> None:
         }
     )
     efetch_url = f"{eutils_api}/efetch.fcgi?{args}"
+    if api_key:
+        efetch_url += f"&api_key={quote(api_key)}"
     params["url"] = efetch_url
 
 
