diff --git a/searx/engines/pubmed.py b/searx/engines/pubmed.py
index be934cdc8d9..dd8d919b99c 100644
--- a/searx/engines/pubmed.py
+++ b/searx/engines/pubmed.py
@@ -29,8 +29,10 @@
 
 categories = ['science', 'scientific publications']
 
+api_key = ''
 base_url = (
-    'https://eutils.ncbi.nlm.nih.gov/entrez/eutils/esearch.fcgi' + '?db=pubmed&{query}&retstart={offset}&retmax={hits}'
+    'https://eutils.ncbi.nlm.nih.gov/entrez/eutils/esearch.fcgi'
+    +'?db=pubmed&{query}&retstart={offset}&retmax={hits}&api_key={api_key}'
 )
 
 # engine dependent config
@@ -58,7 +60,8 @@ def response(resp):  # pylint: disable=too-many-locals
 
     # First retrieve notice of each result
     pubmed_retrieve_api_url = (
-        'https://eutils.ncbi.nlm.nih.gov/entrez/eutils/efetch.fcgi?' + 'db=pubmed&retmode=xml&id={pmids_string}'
+        'https://eutils.ncbi.nlm.nih.gov/entrez/eutils/efetch.fcgi?'
+        + 'db=pubmed&retmode=xml&id={pmids_string}&api_key={api_key}'
     )
 
     pmids_results = etree.XML(resp.content)
