diff --git a/searx/engines/baidu.py b/searx/engines/baidu.py
index 57d1ed444b3..c7c87d2183e 100644
--- a/searx/engines/baidu.py
+++ b/searx/engines/baidu.py
@@ -96,7 +96,7 @@ def request(query, params):
 def response(resp):
     # Detect Baidu Captcha, it will redirect to wappass.baidu.com
     if 'wappass.baidu.com/static/captcha' in resp.headers.get('Location', ''):
-        raise SearxEngineCaptchaException()
+        raise SearxEngineCaptchaException(suspended_time=300, message="Baidu CAPTCHA detected. Please try again later.")
 
     text = resp.text
     if baidu_category == 'images':
