diff --git a/searx/engines/mojeek.py b/searx/engines/mojeek.py
index 1a454ca65f8..f77425f2254 100644
--- a/searx/engines/mojeek.py
+++ b/searx/engines/mojeek.py
@@ -75,6 +75,11 @@ def request(query, params):
 
     params['url'] = f"{base_url}/search?{urlencode(args)}"
 
+    if search_type == "images":
+        # At least in the impersonate mode we have to overwrite the "Accept" header
+        # from curl_cffi default headers:
+        params["headers"]["Accept"] = "*/*"
+
     return params
 
 
