diff --git a/searx/engines/tonline.py b/searx/engines/tonline.py
new file mode 100644
index 00000000000..369456741d2
--- /dev/null
+++ b/searx/engines/tonline.py
@@ -0,0 +1,149 @@
+# SPDX-License-Identifier: AGPL-3.0-or-later
+"""T-Online_ is a German news portal, which is powered by Ströer,
+a German advertising company, not by Deutsche Telekom (contrary to its name).
+
+It gets its web results from Google, image results from Flickr and videos results
+from YouTube.
+
+.. _T-Online: https://www.t-online.de/
+"""
+
+import typing as t
+from urllib.parse import urlencode
+
+from lxml import html
+
+from searx.utils import eval_xpath_list, eval_xpath, extract_text, get_embeded_stream_url
+from searx.result_types import EngineResults
+
+if t.TYPE_CHECKING:
+    from lxml.etree import ElementBase
+    from searx.extended_types import SXNG_Response
+    from searx.search.processors import OnlineParams
+
+about = {
+    "website": "https://www.t-online.de",
+    "wikidata_id": "Q590940",
+    "official_api_documentation": None,
+    "use_official_api": False,
+    "require_api_key": False,
+    "results": "HTML",
+    "language": "de",
+}
+
+paging = True
+time_range_support = True
+
+base_url = "https://suche.t-online.de"
+tonline_categ = "web"
+"""Supported categories are ``web``, ``videos``, ``news`` and ``images``."""
+
+time_range_map = {"day": "d", "week": "w", "month": "m", "year": "y"}
+
+# result provider has to be specified during pagination, pagination can alternatively
+# use "tonline" to only search for results from t-online news articles
+tonline_channel_map = {"images": "flickr", "videos": "yt"}
+
+
+def init(_):
+    if tonline_categ not in ("web", "images", "videos", "news"):
+        raise ValueError("invalid category: %s" % tonline_categ)
+
+
+def request(query: str, params: "OnlineParams") -> None:
+    # "mandant", "dia" and "ptl" are not needed, but this might reduce changes of captchas
+    args = {"q": query, "mandant": "toi", "dia": "suche", "ptl": "std"}
+    if params["time_range"]:
+        args["age"] = time_range_map[params["time_range"]]
+
+    if params["pageno"] > 1 and tonline_categ in tonline_channel_map:
+        ch = tonline_channel_map[tonline_categ]
+        args["ch"] = ch
+        args[f"{ch}_page"] = str(params["pageno"])
+    else:
+        args["page"] = str(params["pageno"])
+
+    params["url"] = f"{base_url}/{tonline_categ}?{urlencode(args)}"
+
+
+def _general_results(doc: "ElementBase") -> EngineResults:
+    res = EngineResults()
+    for result in eval_xpath_list(doc, "//div[@id='google_re']/div[contains(@class, 'doc')]"):
+        (
+            res.add(
+                res.types.MainResult(
+                    url=extract_text(eval_xpath(result, "./a/@href")) or "",
+                    title=extract_text(eval_xpath(result, ".//span[contains(@class, 'tMMReshl')]")) or "",
+                    content=extract_text(eval_xpath(result, ".//div[contains(@class, 'tMMRest')]")) or "",
+                ),
+            )
+        )
+    for suggestion in eval_xpath_list(doc, "//div[starts-with(@class, 'rsbl')]/a"):
+        res.add(res.types.LegacyResult({"suggestion": extract_text(suggestion)}))
+    return res
+
+
+def _image_results(doc: "ElementBase") -> EngineResults:
+    res = EngineResults()
+    for result in eval_xpath_list(doc, "//div[@class='doc']"):
+        (
+            res.add(
+                res.types.Image(
+                    url=extract_text(eval_xpath(result, "./a/@href")) or "",
+                    title=extract_text(eval_xpath(result, ".//div[contains(@class, 'doc_info')]")) or "",
+                    thumbnail_src=extract_text(eval_xpath(result, ".//img/@src")) or "",
+                ),
+            )
+        )
+    return res
+
+
+def _news_results(doc: "ElementBase") -> EngineResults:
+    res = EngineResults()
+    for result in eval_xpath_list(doc, "//div[@id='portal_re']/div[contains(@class, 'doc')]"):
+        title_parts = eval_xpath(result, ".//a[starts-with(@class, 'tMMReshl')]")
+        (
+            res.add(
+                res.types.MainResult(
+                    url=extract_text(eval_xpath(result, "(./a/@href)[1]")) or "",
+                    title=" - ".join(extract_text(part) or "" for part in title_parts),
+                    content=extract_text(eval_xpath(result, ".//div[contains(@class, 'tMMRest')]")) or "",
+                    thumbnail=extract_text(eval_xpath(result, ".//img[contains(@class, 'desk')]/@src")) or "",
+                ),
+            )
+        )
+    return res
+
+
+def _video_results(doc: "ElementBase") -> EngineResults:
+    res = EngineResults()
+    for result in eval_xpath_list(doc, "//div[@class='doc']"):
+        url = extract_text(eval_xpath(result, "./a/@href")) or ""
+        title_parts = eval_xpath(result, ".//a[starts-with(@class, 'tMMReshl')]")
+
+        res.add(
+            res.types.MainResult(
+                template="videos.html",
+                url=url,
+                title=" - ".join(extract_text(part) or "" for part in title_parts),
+                thumbnail=extract_text(eval_xpath(result, ".//img/@src")) or "",
+                iframe_src=get_embeded_stream_url(url) or "",
+            )
+        )
+
+    return res
+
+
+def response(resp: "SXNG_Response") -> EngineResults:
+    doc = html.fromstring(resp.text)
+    match tonline_categ:
+        case "web":
+            return _general_results(doc)
+        case "news":
+            return _news_results(doc)
+        case "images":
+            return _image_results(doc)
+        case "videos":
+            return _video_results(doc)
+        case _:
+            raise ValueError("invalid category: %s" % tonline_categ)
