diff --git a/searx/engines/baidu.py b/searx/engines/baidu.py
index 29c9c0e4d57..f5d55d7659f 100644
--- a/searx/engines/baidu.py
+++ b/searx/engines/baidu.py
@@ -13,7 +13,7 @@
 import time
 import json
 
-from searx.exceptions import SearxEngineAPIException
+from searx.exceptions import SearxEngineAPIException, SearxEngineCaptchaException
 from searx.utils import html_to_text
 
 about = {
@@ -93,6 +93,9 @@ def request(query, params):
 
 
 def response(resp):
+    # Detect Baidu Captcha, it will redirect to wappass.baidu.com
+    if 'wappass.baidu.com' in str(resp.url):
+        raise SearxEngineCaptchaException()
 
     text = resp.text
     if baidu_category == 'images':
