diff --git a/searx/engines/qwant.py b/searx/engines/qwant.py
index 829ee429840..81aeaf32d48 100644
--- a/searx/engines/qwant.py
+++ b/searx/engines/qwant.py
@@ -6,7 +6,6 @@
 For Qwant's *web-search* two alternatives are implemented:
 
 - ``web``: uses the :py:obj:`api_url` which returns a JSON structure
-- ``web-lite``: uses the :py:obj:`web_lite_url` which returns a HTML page
 
 
 Configuration
@@ -22,7 +21,7 @@
 .. code:: yaml
 
   - name: qwant
-    qwant_categ: web-lite  # alternatively use 'web'
+    qwant_categ: web
     ...
   - name: qwant news
     qwant_categ: news
@@ -39,6 +38,8 @@
 
 """
 
+import typing as t
+
 from datetime import (
     datetime,
     timedelta,
@@ -47,7 +48,6 @@
 from urllib.parse import urlencode
 
 import babel
-import lxml
 from flask_babel import gettext
 
 from searx.enginelib.traits import EngineTraits
@@ -59,12 +59,13 @@
 )
 from searx.network import raise_for_httperror
 from searx.utils import (
-    eval_xpath,
-    eval_xpath_list,
-    extract_text,
     get_embeded_stream_url,
 )
 
+if t.TYPE_CHECKING:
+    from searx.search.processors import OnlineParams
+    from searx.extended_types import SXNG_Response
+
 # about
 about = {
     "website": "https://www.qwant.com/",
@@ -83,7 +84,7 @@
 redirect"""
 
 qwant_categ = None
-"""One of ``web-lite`` (or ``web``), ``news``, ``images`` or ``videos``"""
+"""One of ``web``, ``news``, ``images`` or ``videos``"""
 
 safesearch = True
 # safe_search_map = {0: '&safesearch=0', 1: '&safesearch=1', 2: '&safesearch=2'}
@@ -98,85 +99,38 @@
 ]
 # fmt: on
 
-# search-url
-
 api_url = "https://api.qwant.com/v3/search/"
 """URL of Qwant's API (JSON)"""
 
-web_lite_url = "https://lite.qwant.com/"
-"""URL of Qwant-Lite (HTML)"""
 
-
-def request(query, params):
+def request(query: str, params: 'OnlineParams') -> None:
     """Qwant search request"""
 
     if not query:
-        return None
+        return
 
     q_locale = traits.get_region(params["searxng_locale"], default="en_US")
 
-    url = api_url + f"{qwant_categ}?"
-    args = {"q": query}
+    results_per_page = 10
+    if qwant_categ == "images":
+        results_per_page = 50
+    args = {
+        "q": query,
+        "count": results_per_page,
+        "locale": q_locale,
+        "offset": (params["pageno"] - 1) * results_per_page,
+        "device": "desktop",
+        "safesearch": params["safesearch"],
+        "tgp": 1,
+        "display": True,
+        "llm": True,
+    }
     params["raise_for_httperror"] = False
 
-    if qwant_categ == "web-lite":
-        url = web_lite_url + "?"
-        args["locale"] = q_locale.lower()
-        args["l"] = q_locale.split("_")[0]
-        args["s"] = params["safesearch"]
-        args["p"] = params["pageno"]
-
-        params["raise_for_httperror"] = True
-
-    elif qwant_categ == "images":
-        args["count"] = 50
-        args["locale"] = q_locale
-        args["safesearch"] = params["safesearch"]
-        args["tgp"] = 3
-        args["offset"] = (params["pageno"] - 1) * args["count"]
-
-    else:  # web, news, videos
-        args["count"] = 10
-        args["locale"] = q_locale
-        args["safesearch"] = params["safesearch"]
-        args["llm"] = "false"
-        args["tgp"] = 3
-        args["offset"] = (params["pageno"] - 1) * args["count"]
-
-    params["url"] = url + urlencode(args)
-
-    return params
-
-
-def response(resp):
-
-    if qwant_categ == "web-lite":
-        return parse_web_lite(resp)
-    return parse_web_api(resp)
-
-
-def parse_web_lite(resp):
-    """Parse results from Qwant-Lite"""
-
-    results = []
-    dom = lxml.html.fromstring(resp.text)
-
-    for item in eval_xpath_list(dom, "//section/article"):
-        if eval_xpath(item, "./span[contains(@class, 'tooltip')]"):
-            # ignore randomly interspersed advertising adds
-            continue
-        results.append(
-            {
-                "url": extract_text(eval_xpath(item, "./span[contains(@class, 'url partner')]")),
-                "title": extract_text(eval_xpath(item, "./h2/a")),
-                "content": extract_text(eval_xpath(item, "./p")),
-            }
-        )
-
-    return results
+    params["url"] = f"{api_url}{qwant_categ}?{urlencode(args)}"
 
 
-def parse_web_api(resp):
+def response(resp: 'SXNG_Response'):
     """Parse results from Qwant's API"""
     # pylint: disable=too-many-locals, too-many-branches, too-many-statements
 
@@ -184,7 +138,7 @@ def parse_web_api(resp):
 
     # Try to load JSON result
     try:
-        search_results = loads(resp.text)
+        search_results = resp.json()
     except ValueError:
         search_results = {}
 
