diff --git a/searx/engines/bilibili.py b/searx/engines/bilibili.py
index c6aa70dfd17..2bce66a8869 100644
--- a/searx/engines/bilibili.py
+++ b/searx/engines/bilibili.py
@@ -51,6 +51,7 @@ def request(query, params):
     }
 
     params["url"] = f"{base_url}?{urlencode(query_params)}"
+    params["headers"]["Referer"] = "https://www.bilibili.com"
     params["cookies"] = cookie
 
     return params
