diff --git a/searx/engines/yep.py b/searx/engines/yep.py
index 726f62c0dbc..2de18b729ef 100644
--- a/searx/engines/yep.py
+++ b/searx/engines/yep.py
@@ -19,6 +19,8 @@
 safesearch = True
 safesearch_map = {0: 'off', 1: 'moderate', 2: 'strict'}
 
+enable_http2 = False
+
 
 def request(query, params):
     args = {
@@ -30,6 +32,7 @@ def request(query, params):
     }
     params['url'] = f"{base_url}/fs/2/search?{urlencode(args)}"
     params['headers']['Referer'] = 'https://yep.com/'
+    params['headers']['Origin'] = 'https://yep.com'
     return params
 
 
