diff --git a/searx/autocomplete.py b/searx/autocomplete.py
index 8b7285376d2..4e66dff2298 100644
--- a/searx/autocomplete.py
+++ b/searx/autocomplete.py
@@ -179,6 +179,23 @@ def naver(query: str, _sxng_locale: str) -> list[str]:
     return results


+def privacywall(query: str, sxng_locale: str) -> list[str]:
+    # Privacywall search autocompleter
+    country = None
+    if "-" in sxng_locale:
+        country = sxng_locale.split("-")[1]
+    args = {'q': query, 'cc': country}
+
+    url = f"https://www.privacywall.org/search/secure/suggestions.php?{urlencode(args)}"
+    response = get(url)
+
+    if not response.ok:
+        return []
+
+    data: list[list[str]] = response.json()
+    return data[1]
+
+
 def qihu360search(query: str, _sxng_locale: str) -> list[str]:
     # 360Search search autocompleter
     url = f"https://sug.so.360.cn/suggest?{urlencode({'format': 'json', 'word': query})}"
@@ -361,6 +378,7 @@ def yandex(query: str, _sxng_locale: str) -> list[str]:
     'google': google_complete,
     'mwmbl': mwmbl,
     'naver': naver,
+    'privacywall': privacywall,
     'quark': quark,
     'qwant': qwant,
     'seznam': seznam,
diff --git a/searx/data/engine_traits.json b/searx/data/engine_traits.json
index 7c5b102b673..433b4b542b6 100644
--- a/searx/data/engine_traits.json
+++ b/searx/data/engine_traits.json
@@ -6814,6 +6814,255 @@
     },
     "regions": {}
   },
+  "privacywall": {
+    "all_locale": null,
+    "custom": {},
+    "data_type": "traits_v1",
+    "languages": {},
+    "regions": {
+      "bg-BG": "BG",
+      "cs-CZ": "CZ",
+      "da-DK": "DK",
+      "de-AT": "AT",
+      "de-BE": "BE",
+      "de-CH": "CH",
+      "de-DE": "DE",
+      "de-LI": "LI",
+      "de-LU": "LU",
+      "el-CY": "CY",
+      "el-GR": "GR",
+      "en-AU": "AU",
+      "en-CA": "CA",
+      "en-GB": "GB",
+      "en-HK": "HK",
+      "en-IE": "IE",
+      "en-IN": "IN",
+      "en-MT": "MT",
+      "en-NZ": "NZ",
+      "en-PH": "PH",
+      "en-SG": "SG",
+      "en-US": "US",
+      "es-AR": "AR",
+      "es-CL": "CL",
+      "es-CO": "CO",
+      "es-ES": "ES",
+      "es-MX": "MX",
+      "es-PE": "PE",
+      "es-VE": "VE",
+      "et-EE": "EE",
+      "fi-FI": "FI",
+      "fil-PH": "PH",
+      "fr-BE": "BE",
+      "fr-CA": "CA",
+      "fr-CH": "CH",
+      "fr-FR": "FR",
+      "fr-LU": "LU",
+      "ga-IE": "IE",
+      "gsw-CH": "CH",
+      "gsw-LI": "LI",
+      "hi-IN": "IN",
+      "hr-HR": "HR",
+      "hu-HU": "HU",
+      "id-ID": "ID",
+      "it-CH": "CH",
+      "it-IT": "IT",
+      "ja-JP": "JP",
+      "ko-KR": "KR",
+      "lb-LU": "LU",
+      "lt-LT": "LT",
+      "lv-LV": "LV",
+      "mi-NZ": "NZ",
+      "ms-MY": "MY",
+      "ms-SG": "SG",
+      "mt-MT": "MT",
+      "nb-NO": "NO",
+      "nl-BE": "BE",
+      "nl-NL": "NL",
+      "nn-NO": "NO",
+      "pl-PL": "PL",
+      "pt-BR": "BR",
+      "pt-PT": "PT",
+      "qu-PE": "PE",
+      "ro-RO": "RO",
+      "sk-SK": "SK",
+      "sl-SI": "SI",
+      "sv-FI": "FI",
+      "sv-SE": "SE",
+      "ta-SG": "SG",
+      "th-TH": "TH",
+      "tr-CY": "CY",
+      "vi-VN": "VN",
+      "zh-HK": "HK",
+      "zh-SG": "SG",
+      "zh-TW": "TW"
+    }
+  },
+  "privacywall images": {
+    "all_locale": null,
+    "custom": {},
+    "data_type": "traits_v1",
+    "languages": {},
+    "regions": {
+      "bg-BG": "BG",
+      "cs-CZ": "CZ",
+      "da-DK": "DK",
+      "de-AT": "AT",
+      "de-BE": "BE",
+      "de-CH": "CH",
+      "de-DE": "DE",
+      "de-LI": "LI",
+      "de-LU": "LU",
+      "el-CY": "CY",
+      "el-GR": "GR",
+      "en-AU": "AU",
+      "en-CA": "CA",
+      "en-GB": "GB",
+      "en-HK": "HK",
+      "en-IE": "IE",
+      "en-IN": "IN",
+      "en-MT": "MT",
+      "en-NZ": "NZ",
+      "en-PH": "PH",
+      "en-SG": "SG",
+      "en-US": "US",
+      "es-AR": "AR",
+      "es-CL": "CL",
+      "es-CO": "CO",
+      "es-ES": "ES",
+      "es-MX": "MX",
+      "es-PE": "PE",
+      "es-VE": "VE",
+      "et-EE": "EE",
+      "fi-FI": "FI",
+      "fil-PH": "PH",
+      "fr-BE": "BE",
+      "fr-CA": "CA",
+      "fr-CH": "CH",
+      "fr-FR": "FR",
+      "fr-LU": "LU",
+      "ga-IE": "IE",
+      "gsw-CH": "CH",
+      "gsw-LI": "LI",
+      "hi-IN": "IN",
+      "hr-HR": "HR",
+      "hu-HU": "HU",
+      "id-ID": "ID",
+      "it-CH": "CH",
+      "it-IT": "IT",
+      "ja-JP": "JP",
+      "ko-KR": "KR",
+      "lb-LU": "LU",
+      "lt-LT": "LT",
+      "lv-LV": "LV",
+      "mi-NZ": "NZ",
+      "ms-MY": "MY",
+      "ms-SG": "SG",
+      "mt-MT": "MT",
+      "nb-NO": "NO",
+      "nl-BE": "BE",
+      "nl-NL": "NL",
+      "nn-NO": "NO",
+      "pl-PL": "PL",
+      "pt-BR": "BR",
+      "pt-PT": "PT",
+      "qu-PE": "PE",
+      "ro-RO": "RO",
+      "sk-SK": "SK",
+      "sl-SI": "SI",
+      "sv-FI": "FI",
+      "sv-SE": "SE",
+      "ta-SG": "SG",
+      "th-TH": "TH",
+      "tr-CY": "CY",
+      "vi-VN": "VN",
+      "zh-HK": "HK",
+      "zh-SG": "SG",
+      "zh-TW": "TW"
+    }
+  },
+  "privacywall videos": {
+    "all_locale": null,
+    "custom": {},
+    "data_type": "traits_v1",
+    "languages": {},
+    "regions": {
+      "bg-BG": "BG",
+      "cs-CZ": "CZ",
+      "da-DK": "DK",
+      "de-AT": "AT",
+      "de-BE": "BE",
+      "de-CH": "CH",
+      "de-DE": "DE",
+      "de-LI": "LI",
+      "de-LU": "LU",
+      "el-CY": "CY",
+      "el-GR": "GR",
+      "en-AU": "AU",
+      "en-CA": "CA",
+      "en-GB": "GB",
+      "en-HK": "HK",
+      "en-IE": "IE",
+      "en-IN": "IN",
+      "en-MT": "MT",
+      "en-NZ": "NZ",
+      "en-PH": "PH",
+      "en-SG": "SG",
+      "en-US": "US",
+      "es-AR": "AR",
+      "es-CL": "CL",
+      "es-CO": "CO",
+      "es-ES": "ES",
+      "es-MX": "MX",
+      "es-PE": "PE",
+      "es-VE": "VE",
+      "et-EE": "EE",
+      "fi-FI": "FI",
+      "fil-PH": "PH",
+      "fr-BE": "BE",
+      "fr-CA": "CA",
+      "fr-CH": "CH",
+      "fr-FR": "FR",
+      "fr-LU": "LU",
+      "ga-IE": "IE",
+      "gsw-CH": "CH",
+      "gsw-LI": "LI",
+      "hi-IN": "IN",
+      "hr-HR": "HR",
+      "hu-HU": "HU",
+      "id-ID": "ID",
+      "it-CH": "CH",
+      "it-IT": "IT",
+      "ja-JP": "JP",
+      "ko-KR": "KR",
+      "lb-LU": "LU",
+      "lt-LT": "LT",
+      "lv-LV": "LV",
+      "mi-NZ": "NZ",
+      "ms-MY": "MY",
+      "ms-SG": "SG",
+      "mt-MT": "MT",
+      "nb-NO": "NO",
+      "nl-BE": "BE",
+      "nl-NL": "NL",
+      "nn-NO": "NO",
+      "pl-PL": "PL",
+      "pt-BR": "BR",
+      "pt-PT": "PT",
+      "qu-PE": "PE",
+      "ro-RO": "RO",
+      "sk-SK": "SK",
+      "sl-SI": "SI",
+      "sv-FI": "FI",
+      "sv-SE": "SE",
+      "ta-SG": "SG",
+      "th-TH": "TH",
+      "tr-CY": "CY",
+      "vi-VN": "VN",
+      "zh-HK": "HK",
+      "zh-SG": "SG",
+      "zh-TW": "TW"
+    }
+  },
   "qwant": {
     "all_locale": null,
     "custom": {},
diff --git a/searx/engines/privacywall.py b/searx/engines/privacywall.py
new file mode 100644
index 00000000000..9540e3a057e
--- /dev/null
+++ b/searx/engines/privacywall.py
@@ -0,0 +1,217 @@
+# SPDX-License-Identifier: AGPL-3.0-or-later
+"""Privacywall_ claims to be a "privacy-friendly" search engine,
+but according to a `Privacyguides discussion`_ it's sharing private
+user information with Microsoft and Amazon.
+
+.. _Privacywall : https://www.privacywall.org
+.. _`Privacyguides discussion` : https://discuss.privacyguides.net/t/how-is-privacy-wall-search-engine/29486
+"""
+
+import typing as t
+from urllib.parse import urlencode, unquote_plus
+
+from lxml import html
+import babel
+
+from searx.enginelib.traits import EngineTraits
+from searx.utils import eval_xpath_list, eval_xpath, extract_text, get_embeded_stream_url, extr
+from searx.locales import region_tag
+from searx.result_types import EngineResults
+
+
+if t.TYPE_CHECKING:
+    from lxml.etree import ElementBase
+    from searx.extended_types import SXNG_Response
+    from searx.search.processors import OnlineParams
+
+about = {
+    "website": "https://privacywall.org",
+    "wikidata_id": None,
+    "official_api_documentation": None,
+    "use_official_api": False,
+    "require_api_key": False,
+    "results": "HTML",
+}
+
+paging = True
+safesearch = True
+time_range_support = True
+
+base_url = "https://www.privacywall.org"
+privacywall_category = "general"
+"""Supported categories are ``general``, ``videos`` and ``images``."""
+
+
+# corresponds to the "k" query param
+safesearch_map = {0: "off", 1: "on", 2: "on"}
+
+# page number sent for videos (is independent of the query) - certainly there's
+# a pattern in this, but for our use case it's enough to just support the first
+# 10 pages by hardcoding the page "numbers"
+video_page_map = {
+    2: "CAoQAA",
+    3: "CBQQAA",
+    4: "CB4QAA",
+    5: "CCgQAA",
+    6: "CDIQAA",
+    7: "CDwQAA",
+    8: "CEYQAA",
+    9: "CFAQAA",
+    10: "CFoQAA",
+}
+
+
+def init(_):
+    if privacywall_category not in ("general", "images", "videos"):
+        raise ValueError("invalid category: %s" % privacywall_category)
+
+
+def request(query: str, params: "OnlineParams") -> None:
+    if params["pageno"] > 10:
+        params["url"] = None
+        return
+
+    args = {"q": query, "safesearch": safesearch_map[params["safesearch"]]}
+    if params["searxng_locale"] != "all":
+        args["cc"] = traits.get_region(params["searxng_locale"]) or "US"
+    if params["time_range"]:
+        # time range uses the same "day", "week", "month", "year" naming scheme as SearXNG
+        args["time"] = params["time_range"]
+
+    if params["pageno"] > 1:
+        if privacywall_category == "images":
+            args["page"] = str(params["pageno"])
+        elif privacywall_category == "videos":
+            args["page"] = video_page_map[params["pageno"]]
+        else:
+            raise ValueError("general engine does not support pagination")
+
+    if privacywall_category == "general":
+        params["url"] = f"{base_url}/search/secure/?{urlencode(args)}"
+    else:
+        params["url"] = f"{base_url}/{privacywall_category}/?{urlencode(args)}"
+
+
+def _general_results(doc: "ElementBase") -> EngineResults:
+    res = EngineResults()
+    for result in eval_xpath_list(doc, "//div[@id='pw-results-main']/div[contains(@class, 'result-card')]"):
+        (
+            res.add(
+                res.types.MainResult(
+                    url=extract_text(eval_xpath(result, ".//a[contains(@class, 'result-url-anchor')]/@href")) or "",
+                    title=extract_text(eval_xpath(result, ".//div[contains(@class, 'result_title')]")) or "",
+                    content=extract_text(eval_xpath(result, ".//div[contains(@class, 'result-description')]")) or "",
+                ),
+            )
+        )
+    return res
+
+
+def _extract_thumbnail_url(url: str) -> str:
+    """
+    Get the URL from strings like "/videos/video.php?id=<urlencoded-urlhere>".
+    """
+    url_start = url.find("?id=") + len("?id=")
+    thumbnail = unquote_plus(url[url_start:])
+    return thumbnail
+
+
+def _image_results(doc: "ElementBase") -> EngineResults:
+    res = EngineResults()
+    for result in eval_xpath_list(doc, "//div[@id='container']/div[contains(@class, 'imgcontainer')]"):
+        (
+            res.add(
+                res.types.Image(
+                    url=extract_text(eval_xpath(result, "./a/@href")) or "",
+                    content=extract_text(eval_xpath(result, "./a/@alt")) or "",
+                    thumbnail_src=_extract_thumbnail_url(extract_text(eval_xpath(result, ".//img/@src")) or ""),
+                    source=extract_text(eval_xpath(result, ".//div[contains(@class, 'image-source-badge')]")) or "",
+                ),
+            )
+        )
+    return res
+
+
+def _video_results(doc: "ElementBase") -> EngineResults:
+    res = EngineResults()
+    for result in eval_xpath_list(
+        doc, "//div[contains(@class, 'video-container')]/div[contains(@class, 'video-card')]"
+    ):
+        url = extract_text(eval_xpath(result, "./a/@href")) or ""
+        if not url:
+            continue
+
+        thumbnail = None
+        # looks like <div style="background-image:url(/videos/video.php?id=<urlencoded-urlhere>);position:relative">
+        thumbnail_style = extract_text(eval_xpath(result, ".//div[contains(@class, 'video-img')]/@style"))
+        if thumbnail_style:
+            thumbnail = _extract_thumbnail_url(extr(thumbnail_style, ":url(", ")"))
+
+        res.add(
+            res.types.MainResult(
+                template="videos.html",
+                url=url,
+                title=extract_text(eval_xpath(result, ".//h2[contains(@class, 'video-card-title')]")) or "",
+                content=extract_text(eval_xpath(result, ".//p")) or "",
+                thumbnail=thumbnail or "",
+                iframe_src=get_embeded_stream_url(url) or "",
+            )
+        )
+
+    return res
+
+
+def response(resp: "SXNG_Response") -> EngineResults:
+    doc = html.fromstring(resp.text)
+    match privacywall_category:
+        case "general":
+            return _general_results(doc)
+        case "images":
+            return _image_results(doc)
+        case "videos":
+            return _video_results(doc)
+        case _:
+            raise ValueError("invalid vuhuv category: %s" % privacywall_category)
+
+
+def fetch_traits(engine_traits: EngineTraits) -> None:
+    """Fetch regions from Bing-Web."""
+    # pylint: disable=import-outside-toplevel
+
+    from searx.network import get  # see https://github.com/searxng/searxng/issues/762
+    from searx.utils import gen_useragent
+
+    headers = {
+        "User-Agent": gen_useragent(),
+    }
+
+    resp = get(base_url, headers=headers)
+    if not resp.ok:
+        raise RuntimeError("Response from Privacywall is not OK.")
+
+    dom = html.fromstring(resp.text)
+
+    # <div class="dropdown-option" onclick="changeMenuLanguage(&quot;CZ&quot;)"></div>
+    for onclick_listener in eval_xpath(
+        dom, "//div[contains(@class, 'lang-menu')]//div[contains(@class, 'dropdown-option')]/@onclick"
+    ):
+        # this is either a normal lang-country tag (e.g. cs-cz) or only a country code (e.g. de, at, ...)
+        country_tag = extr(onclick_listener, "(\"", "\")")
+
+        # the locale tag is only a country tag, so we get languages the from the list of official languages
+        # of the country
+        lang_tag: str
+        for lang_tag in babel.languages.get_official_languages(country_tag, de_facto=True):  # pyright: ignore
+            try:
+                sxng_tag = region_tag(babel.Locale.parse(f"{lang_tag}_{country_tag.upper()}"))
+            except babel.UnknownLocaleError:
+                # silently ignore unknown languages
+                continue
+
+            conflict = engine_traits.regions.get(sxng_tag)
+            if conflict:
+                if conflict != sxng_tag:
+                    print("CONFLICT: babel %s --> %s" % (sxng_tag, conflict))
+                continue
+
+            engine_traits.regions[sxng_tag] = country_tag
diff --git a/searx/sxng_locales.py b/searx/sxng_locales.py
index 8d75bec9829..b0a71ded5a5 100644
--- a/searx/sxng_locales.py
+++ b/searx/sxng_locales.py
@@ -21,6 +21,7 @@
     ('da-DK', 'Dansk', 'Danmark', 'Danish', '\U0001f1e9\U0001f1f0'),
     ('de', 'Deutsch', '', 'German', '\U0001f310'),
     ('de-AT', 'Deutsch', 'Österreich', 'German', '\U0001f1e6\U0001f1f9'),
+    ('de-BE', 'Deutsch', 'Belgien', 'German', '\U0001f1e7\U0001f1ea'),
     ('de-CH', 'Deutsch', 'Schweiz', 'German', '\U0001f1e8\U0001f1ed'),
     ('de-DE', 'Deutsch', 'Deutschland', 'German', '\U0001f1e9\U0001f1ea'),
     ('el', 'Ελληνικά', '', 'Greek', '\U0001f310'),
@@ -29,6 +30,7 @@
     ('en-AU', 'English', 'Australia', 'English', '\U0001f1e6\U0001f1fa'),
     ('en-CA', 'English', 'Canada', 'English', '\U0001f1e8\U0001f1e6'),
     ('en-GB', 'English', 'United Kingdom', 'English', '\U0001f1ec\U0001f1e7'),
+    ('en-HK', 'English', 'Hong Kong SAR China', 'English', '\U0001f1ed\U0001f1f0'),
     ('en-IE', 'English', 'Ireland', 'English', '\U0001f1ee\U0001f1ea'),
     ('en-IN', 'English', 'India', 'English', '\U0001f1ee\U0001f1f3'),
     ('en-NZ', 'English', 'New Zealand', 'English', '\U0001f1f3\U0001f1ff'),
@@ -44,17 +46,23 @@
     ('es-ES', 'Español', 'España', 'Spanish', '\U0001f1ea\U0001f1f8'),
     ('es-MX', 'Español', 'México', 'Spanish', '\U0001f1f2\U0001f1fd'),
     ('es-PE', 'Español', 'Perú', 'Spanish', '\U0001f1f5\U0001f1ea'),
+    ('es-VE', 'Español', 'Venezuela', 'Spanish', '\U0001f1fb\U0001f1ea'),
     ('et', 'Eesti', '', 'Estonian', '\U0001f310'),
     ('et-EE', 'Eesti', 'Eesti', 'Estonian', '\U0001f1ea\U0001f1ea'),
     ('fi', 'Suomi', '', 'Finnish', '\U0001f310'),
     ('fi-FI', 'Suomi', 'Suomi', 'Finnish', '\U0001f1eb\U0001f1ee'),
+    ('fil', 'Filipino', '', 'Filipino', '\U0001f310'),
+    ('fil-PH', 'Filipino', 'Pilipinas', 'Filipino', '\U0001f1f5\U0001f1ed'),
     ('fr', 'Français', '', 'French', '\U0001f310'),
     ('fr-BE', 'Français', 'Belgique', 'French', '\U0001f1e7\U0001f1ea'),
     ('fr-CA', 'Français', 'Canada', 'French', '\U0001f1e8\U0001f1e6'),
     ('fr-CH', 'Français', 'Suisse', 'French', '\U0001f1e8\U0001f1ed'),
     ('fr-FR', 'Français', 'France', 'French', '\U0001f1eb\U0001f1f7'),
     ('gl', 'Galego', '', 'Galician', '\U0001f310'),
+    ('hi', 'हिन्दी', '', 'Hindi', '\U0001f310'),
+    ('hi-IN', 'हिन्दी', 'भारत', 'Hindi', '\U0001f1ee\U0001f1f3'),
     ('hr', 'Hrvatski', '', 'Croatian', '\U0001f310'),
+    ('hr-HR', 'Hrvatski', 'Hrvatska', 'Croatian', '\U0001f1ed\U0001f1f7'),
     ('hu', 'Magyar', '', 'Hungarian', '\U0001f310'),
     ('hu-HU', 'Magyar', 'Magyarország', 'Hungarian', '\U0001f1ed\U0001f1fa'),
     ('id', 'Indonesia', '', 'Indonesian', '\U0001f310'),
@@ -71,6 +79,8 @@
     ('nl', 'Nederlands', '', 'Dutch', '\U0001f310'),
     ('nl-BE', 'Nederlands', 'België', 'Dutch', '\U0001f1e7\U0001f1ea'),
     ('nl-NL', 'Nederlands', 'Nederland', 'Dutch', '\U0001f1f3\U0001f1f1'),
+    ('nn', 'Norsk Nynorsk', '', 'Norwegian Nynorsk', '\U0001f310'),
+    ('nn-NO', 'Norsk Nynorsk', 'Noreg', 'Norwegian Nynorsk', '\U0001f1f3\U0001f1f4'),
     ('pl', 'Polski', '', 'Polish', '\U0001f310'),
     ('pl-PL', 'Polski', 'Polska', 'Polish', '\U0001f1f5\U0001f1f1'),
     ('pt', 'Português', '', 'Portuguese', '\U0001f310'),
@@ -83,6 +93,7 @@
     ('sk', 'Slovenčina', '', 'Slovak', '\U0001f310'),
     ('sq', 'Shqip', '', 'Albanian', '\U0001f310'),
     ('sv', 'Svenska', '', 'Swedish', '\U0001f310'),
+    ('sv-FI', 'Svenska', 'Finland', 'Swedish', '\U0001f1eb\U0001f1ee'),
     ('sv-SE', 'Svenska', 'Sverige', 'Swedish', '\U0001f1f8\U0001f1ea'),
     ('th', 'ไทย', '', 'Thai', '\U0001f310'),
     ('th-TH', 'ไทย', 'ไทย', 'Thai', '\U0001f1f9\U0001f1ed'),
