diff --git a/searx/engines/bilibili.py b/searx/engines/bilibili.py
index 2bce66a8869..3fd613b9c1b 100644
--- a/searx/engines/bilibili.py
+++ b/searx/engines/bilibili.py
@@ -51,11 +51,10 @@ def request(query, params):
     }
 
     params["url"] = f"{base_url}?{urlencode(query_params)}"
-    params["headers"]["Referer"] = "https://www.bilibili.com"
+    params["headers"]["Referer"] = "https://www.bilibili.com/"
+    params["headers"]["Accept"] = "application/json, text/javascript, */*; q=0.01"
     params["cookies"] = cookie
 
-    return params
-
 
 def response(resp):
     search_res = resp.json()
