diff --git a/searx/engines/duckduckgo_extra.py b/searx/engines/duckduckgo_extra.py
index 4e32ba11897f..3cb12d0e3322 100644
--- a/searx/engines/duckduckgo_extra.py
+++ b/searx/engines/duckduckgo_extra.py
@@ -41,7 +41,9 @@
 safesearch_args = {0: "1", 1: None, 2: "1"}
 
 search_path_map = {"images": "i", "videos": "v", "news": "news"}
+
 _HTTP_User_Agent: str = gen_useragent()
+send_accept_language_header = False
 
 
 def init(engine_settings: dict[str, t.Any]):
diff --git a/searx/search/processors/online.py b/searx/search/processors/online.py
index 427aff6e09ea..c4967ef6a594 100644
--- a/searx/search/processors/online.py
+++ b/searx/search/processors/online.py
@@ -152,7 +152,6 @@ def get_params(self, search_query: "SearchQuery", engine_category: str) -> Onlin
         # add Accept-Language header
         # https://developer.mozilla.org/en-US/docs/Web/HTTP/Reference/Headers/Accept-Language
 
-        headers["Accept-Language"] = "en,en-US;q=0.7,en;q=0.3"
         if self.engine.send_accept_language_header and search_query.locale:
             _l = search_query.locale.language
             _t = search_query.locale.territory or _l
