diff --git a/searx/engines/quark.py b/searx/engines/quark.py
index 957ef0d97a7..1c2e844928c 100644
--- a/searx/engines/quark.py
+++ b/searx/engines/quark.py
@@ -11,7 +11,7 @@
 
 # Metadata
 about = {
-    "website": "https://m.quark.cn/",
+    "website": "https://quark.sm.cn/",
     "wikidata_id": "Q48816502",
     "use_official_api": False,
     "require_api_key": False,
@@ -53,7 +53,7 @@ def request(query, params):
 
     category_config = {
         'general': {
-            'endpoint': 'https://m.quark.cn/s',
+            'endpoint': 'https://quark.sm.cn/s',
             'params': {
                 "q": query,
                 "layout": "html",
