diff --git a/searx/engines/resulthunter.py b/searx/engines/resulthunter.py
index 8de08502d1d..5ebda3e7e58 100644
--- a/searx/engines/resulthunter.py
+++ b/searx/engines/resulthunter.py
@@ -52,6 +52,7 @@ def request(query: str, params: "OnlineParams") -> None:
         "q": query,
         "search_type": resulthunter_categ,
         "offset": params["pageno"] - 1,
+        "search_source": "other",
     }
 
     # uses Brave's engine traits
