From 28931e2ef19af911b0171cf4569a039960525c37 Mon Sep 17 00:00:00 2001
From: Zhijie He <hezhijie0327@hotmail.com>
Date: Thu, 7 Sep 2023 18:20:35 +0800
Subject: [PATCH] Change default lazycache ttl

---
 plugin/executable/cache/cache.go | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/plugin/executable/cache/cache.go b/plugin/executable/cache/cache.go
index f67d740..be2b772 100644
--- a/plugin/executable/cache/cache.go
+++ b/plugin/executable/cache/cache.go
@@ -57,7 +57,7 @@ func init() {

 const (
 	defaultLazyUpdateTimeout = time.Second * 5
-	expiredMsgTtl            = 5
+	expiredMsgTtl            = 30

 	minimumChangesToDump   = 1024
 	dumpHeader             = "mosdns_cache_v2"
--
2.42.0
