From c139684d3437acaec9116b1886a843406a746757 Mon Sep 17 00:00:00 2001
From: Zhijie He <hezhijie0327@hotmail.com>
Date: Mon, 26 Jul 2021 08:08:45 +0800
Subject: [PATCH] A possible nullptr dereference problem.

---
 src/UriListParser.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/UriListParser.cc b/src/UriListParser.cc
index ca021cb3..775197cd 100644
--- a/src/UriListParser.cc
+++ b/src/UriListParser.cc
@@ -112,7 +112,7 @@ void UriListParser::parseNext(std::vector<std::string>& uris, Option& op)
 bool UriListParser::hasNext()
 {
   bool rv = !line_.empty() || (fp_ && *fp_ && !fp_->eof());
-  if (!rv) {
+  if (!rv && fp_) {
     fp_->close();
   }
   return rv;
-- 
2.32.0

