From de12a6d2245297152f45303a91ce584082a62c03 Mon Sep 17 00:00:00 2001
From: Zhijie He <hezhijie0327@hotmail.com>
Date: Thu, 19 Aug 2021 07:42:04 +0000
Subject: [PATCH] Fixed an error when building static with latest libuv.

---
 src/LibuvEventPoll.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/LibuvEventPoll.h b/src/LibuvEventPoll.h
index 42641d77..9b9fc904 100644
--- a/src/LibuvEventPoll.h
+++ b/src/LibuvEventPoll.h
@@ -93,7 +93,7 @@ private:
     inline KPoll(LibuvEventPoll* eventer, KSocketEntry* entry, sock_t sock)
         : eventer_(eventer), entry_(entry)
     {
-      uv_poll_init_socket(eventer->loop_, &handle_, sock);
+      uv_poll_init(eventer->loop_, &handle_, sock);
       handle_.data = this;
     }
     inline void start()
--
2.32.0
