From f23d4b91a9fcdf4fa3d252d2ae6f86e2de3b8b72 Mon Sep 17 00:00:00 2001
From: Zhijie He <hezhijie0327@hotmail.com>
Date: Sat, 7 Jan 2023 08:19:30 +0800
Subject: [PATCH] Retry on HTTP 500

---
 src/HttpSkipResponseCommand.cc | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/HttpSkipResponseCommand.cc b/src/HttpSkipResponseCommand.cc
index de4ad6c..6da3dc3 100644
--- a/src/HttpSkipResponseCommand.cc
+++ b/src/HttpSkipResponseCommand.cc
@@ -243,6 +243,7 @@ bool HttpSkipResponseCommand::processResponse()
       break;
     case 408: // Request Timeout
     case 429: // Too Many Requests
+    case 500: // https://github.com/aria2/aria2/pull/2004
     case 502: // Bad Gateway
     case 503: // Service Unavailable
     case 507: // Insufficient Storage
-- 
2.39.0

