From 16ad27c2c60eb6b9e63271ef60d198387effefbe Mon Sep 17 00:00:00 2001
From: Zhijie He <hezhijie0327@hotmail.com>
Date: Wed, 12 Jun 2024 05:53:15 +0800
Subject: [PATCH] 123

---
 .../Settings/Dns/Upstream/Examples.tsx        | 24 +++++++++----------
 1 file changed, 12 insertions(+), 12 deletions(-)

diff --git a/client/src/components/Settings/Dns/Upstream/Examples.tsx b/client/src/components/Settings/Dns/Upstream/Examples.tsx
index 5a63e856..ad32b12c 100644
--- a/client/src/components/Settings/Dns/Upstream/Examples.tsx
+++ b/client/src/components/Settings/Dns/Upstream/Examples.tsx
@@ -11,34 +11,34 @@ const Examples = (props: ExamplesProps) => (
         <Trans>examples_title</Trans>:
         <ol className="leading-loose">
             <li>
-                <code>94.140.14.140</code>, <code>2a10:50c0::1:ff</code>: {props.t('example_upstream_regular')}
+                <code>223.5.5.5</code>, <code>2400:3200::1</code>: {props.t('example_upstream_regular')}
             </li>

             <li>
-                <code>94.140.14.140:53</code>, <code>[2a10:50c0::1:ff]:53</code>:{' '}
+                <code>223.6.6.6:53</code>, <code>[2400:3200:baba::1]:53</code>:{' '}
                 {props.t('example_upstream_regular_port')}
             </li>

             <li>
-                <code>udp://unfiltered.adguard-dns.com</code>: <Trans>example_upstream_udp</Trans>
+                <code>udp://dns.alidns.com:53</code>: <Trans>example_upstream_udp</Trans>
             </li>

             <li>
-                <code>tcp://94.140.14.140</code>, <code>tcp://[2a10:50c0::1:ff]</code>:{' '}
+                <code>tcp://223.5.5.5</code>, <code>tcp://[2400:3200::1]</code>:{' '}
                 <Trans>example_upstream_tcp</Trans>
             </li>

             <li>
-                <code>tcp://94.140.14.140:53</code>, <code>tcp://[2a10:50c0::1:ff]:53</code>:{' '}
+                <code>tcp://223.6.6.6:53</code>, <code>tcp://[2400:3200:baba::1]:53</code>:{' '}
                 <Trans>example_upstream_tcp_port</Trans>
             </li>

             <li>
-                <code>tcp://unfiltered.adguard-dns.com</code>: <Trans>example_upstream_tcp_hostname</Trans>
+                <code>tcp://dns.alidns.com:53</code>: <Trans>example_upstream_tcp_hostname</Trans>
             </li>

             <li>
-                <code>tls://unfiltered.adguard-dns.com</code>:{' '}
+                <code>tls://dns.alidns.com:853</code>:{' '}
                 <Trans
                     components={[
                         <a
@@ -54,7 +54,7 @@ const Examples = (props: ExamplesProps) => (
             </li>

             <li>
-                <code>https://unfiltered.adguard-dns.com/dns-query</code>:{' '}
+                <code>https://dns.alidns.com:443/dns-query</code>:{' '}
                 <Trans
                     components={[
                         <a
@@ -70,7 +70,7 @@ const Examples = (props: ExamplesProps) => (
             </li>

             <li>
-                <code>h3://unfiltered.adguard-dns.com/dns-query</code>:{' '}
+                <code>h3://dns.alidns.com:443/dns-query</code>:{' '}
                 <Trans
                     components={[
                         <a
@@ -86,7 +86,7 @@ const Examples = (props: ExamplesProps) => (
             </li>

             <li>
-                <code>quic://unfiltered.adguard-dns.com</code>:{' '}
+                <code>quic://dns.alidns.com:853</code>:{' '}
                 <Trans
                     components={[
                         <a
@@ -126,7 +126,7 @@ const Examples = (props: ExamplesProps) => (
             </li>

             <li>
-                <code>[/example.local/]94.140.14.140</code>:{' '}
+                <code>[/example.local/]https://dns.alidns.com:443/dns-query</code>:{' '}
                 <Trans
                     components={[
                         <a
@@ -142,7 +142,7 @@ const Examples = (props: ExamplesProps) => (
             </li>

             <li>
-                <code>[/example.local/]94.140.14.140 2a10:50c0::1:ff</code>:{' '}
+                <code>[/example.local/]https://dns.alidns.com:443/dns-query tls://dns.alidns.com:853</code>:{' '}
                 <Trans
                     components={[
                         <a
--
2.45.2.windows.1
