From 28259aefb4e2e3b73cbae1d45765b2b65928eae3 Mon Sep 17 00:00:00 2001
From: Zhijie He <hezhijie0327@hotmail.com>
Date: Sat, 4 Nov 2023 20:30:04 +0800
Subject: [PATCH] Change the default examples of upstream DNS

---
 .../Settings/Dns/Upstream/Examples.js         | 24 +++++++++----------
 1 file changed, 12 insertions(+), 12 deletions(-)

diff --git a/client/src/components/Settings/Dns/Upstream/Examples.js b/client/src/components/Settings/Dns/Upstream/Examples.js
index c35d65c..95e2694 100644
--- a/client/src/components/Settings/Dns/Upstream/Examples.js
+++ b/client/src/components/Settings/Dns/Upstream/Examples.js
@@ -8,25 +8,25 @@ const Examples = (props) => (
         <Trans>examples_title</Trans>:
         <ol className="leading-loose">
             <li>
-                <code>94.140.14.140</code>, <code>2a10:50c0::1:ff</code>: {props.t('example_upstream_regular')}
+                <code>223.5.5.5</code>, <code>2400:3200::1</code>: {props.t('example_upstream_regular')}
             </li>
             <li>
-                <code>94.140.14.140:53</code>, <code>[2a10:50c0::1:ff]:53</code>: {props.t('example_upstream_regular_port')}
+                <code>223.6.6.6:53</code>, <code>[2400:3200:baba::1]:53</code>: {props.t('example_upstream_regular_port')}
             </li>
             <li>
-                <code>udp://unfiltered.adguard-dns.com</code>: <Trans>example_upstream_udp</Trans>
+                <code>udp://dns.alidns.com:53</code>: <Trans>example_upstream_udp</Trans>
             </li>
             <li>
-                <code>tcp://94.140.14.140</code>, <code>tcp://[2a10:50c0::1:ff]</code>: <Trans>example_upstream_tcp</Trans>
+                <code>tcp://223.5.5.5</code>, <code>tcp://[2400:3200::1]</code>: <Trans>example_upstream_tcp</Trans>
             </li>
             <li>
-                <code>tcp://94.140.14.140:53</code>, <code>tcp://[2a10:50c0::1:ff]:53</code>: <Trans>example_upstream_tcp_port</Trans>
+                <code>tcp://223.6.6.6:53</code>, <code>tcp://[2400:3200:baba::1]:53</code>: <Trans>example_upstream_tcp_port</Trans>
             </li>
             <li>
-                <code>tcp://unfiltered.adguard-dns.com</code>: <Trans>example_upstream_tcp_hostname</Trans>
+                <code>tcp://dns.alidns.com:53</code>: <Trans>example_upstream_tcp_hostname</Trans>
             </li>
             <li>
-                <code>tls://unfiltered.adguard-dns.com</code>: <Trans
+                <code>tls://dns.alidns.com:853</code>: <Trans
                     components={[
                         <a
                             href="https://en.wikipedia.org/wiki/DNS_over_TLS"
@@ -42,7 +42,7 @@ const Examples = (props) => (
                 </Trans>
             </li>
             <li>
-                <code>https://unfiltered.adguard-dns.com/dns-query</code>: <Trans
+                <code>https://dns.alidns.com:443/dns-query</code>: <Trans
                     components={[
                         <a
                             href="https://en.wikipedia.org/wiki/DNS_over_HTTPS"
@@ -58,7 +58,7 @@ const Examples = (props) => (
                 </Trans>
             </li>
             <li>
-                <code>h3://unfiltered.adguard-dns.com/dns-query</code>: <Trans
+                <code>h3://dns.alidns.com:443/dns-query</code>: <Trans
                     components={[
                         <a
                             href="https://en.wikipedia.org/wiki/HTTP/3"
@@ -74,7 +74,7 @@ const Examples = (props) => (
                 </Trans>
             </li>
             <li>
-                <code>quic://unfiltered.adguard-dns.com</code>: <Trans
+                <code>quic://dns.alidns.com:853</code>: <Trans
                     components={[
                         <a
                             href="https://datatracker.ietf.org/doc/html/rfc9250"
@@ -122,7 +122,7 @@ const Examples = (props) => (
                 </Trans>
             </li>
             <li>
-                <code>[/example.local/]94.140.14.140</code>: <Trans
+                <code>[/example.local/]https://dns.alidns.com:443/dns-query</code>: <Trans
                     components={[
                         <a
                             href="https://github.com/AdguardTeam/AdGuardHome/wiki/Configuration#upstreams-for-domains"
@@ -138,7 +138,7 @@ const Examples = (props) => (
                 </Trans>
             </li>
             <li>
-                <code>[/example.local/]94.140.14.140 2a10:50c0::1:ff</code>: <Trans
+                <code>[/example.local/]https://dns.alidns.com:443/dns-query tls://dns.alidns.com:853</code>: <Trans
                     components={[
                         <a
                             href="https://github.com/AdguardTeam/AdGuardHome/wiki/Configuration#upstreams-for-domains"
--
2.42.0.windows.2
