From 39e89c40ab84ea16b0405be99540e0414227199b Mon Sep 17 00:00:00 2001
From: Zhijie He <hezhijie0327@hotmail.com>
Date: Sun, 20 Aug 2023 16:59:52 +0800
Subject: [PATCH] options: unlock DNS_MAX_SERVERS limit

---
 src/dns_conf.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/dns_conf.h b/src/dns_conf.h
index 8167814..57b2ec7 100644
--- a/src/dns_conf.h
+++ b/src/dns_conf.h
@@ -35,7 +35,7 @@ extern "C" {
 #endif

 #define DNS_MAX_BIND_IP 32
-#define DNS_MAX_SERVERS 64
+#define DNS_MAX_SERVERS 65536
 #define DNS_MAX_SERVER_NAME_LEN 128
 #define DNS_MAX_PTR_LEN 128
 #define DNS_MAX_IPSET_NAMELEN 32
--
2.41.0
