diff --git a/src/include/smartdns/dns_conf.h b/src/include/smartdns/dns_conf.h
index 77365048a7..e8f5729741 100644
--- a/src/include/smartdns/dns_conf.h
+++ b/src/include/smartdns/dns_conf.h
@@ -35,7 +35,7 @@ extern "C" {
 #endif
 
 #define DNS_MAX_BIND_IP 32
-#define DNS_MAX_SERVERS 64
+#define DNS_MAX_SERVERS 65536
 #define DNS_MAX_SERVER_NAME_LEN 128
 #define DNS_MAX_PTR_LEN 128
 #define DNS_MAX_IPSET_NAMELEN 32
