diff --git a/src/dns_client/dns_client.h b/src/dns_client/dns_client.h
index 619438ecdf..16ba03409e 100644
--- a/src/dns_client/dns_client.h
+++ b/src/dns_client/dns_client.h
@@ -42,8 +42,8 @@ extern "C" {
 #define DNS_TCP_BUFFER (32 * 1024)
 #define DNS_TCP_IDLE_TIMEOUT (60 * 10)
 #define DNS_TCP_CONNECT_TIMEOUT (5)
-#define DNS_QUERY_TIMEOUT (500)
-#define DNS_QUERY_RETRY (4)
+#define DNS_QUERY_TIMEOUT (5000)
+#define DNS_QUERY_RETRY (5)
 #define DNS_PENDING_SERVER_RETRY 60
 #define SOCKET_PRIORITY (6)
 #define SOCKET_IP_TOS (IPTOS_LOWDELAY | IPTOS_RELIABILITY)
