From 6c58dc49c32aeab5d35e03d1d69ec109044028d2 Mon Sep 17 00:00:00 2001
From: Zhijie He <hezhijie0327@hotmail.com>
Date: Tue, 29 Aug 2023 20:56:36 +0800
Subject: [PATCH] options: change DNS_QUERY_TIMEOUT value

---
 src/dns_client.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/dns_client.c b/src/dns_client.c
index 0795abd..8b5e59d 100644
--- a/src/dns_client.c
+++ b/src/dns_client.c
@@ -61,7 +61,7 @@
 #define DNS_TCP_BUFFER (32 * 1024)
 #define DNS_TCP_IDLE_TIMEOUT (60 * 10)
 #define DNS_TCP_CONNECT_TIMEOUT (5)
-#define DNS_QUERY_TIMEOUT (500)
+#define DNS_QUERY_TIMEOUT (5000)
 #define DNS_QUERY_RETRY (4)
 #define DNS_PENDING_SERVER_RETRY 60
 #define SOCKET_PRIORITY (6)
--
2.42.0
