From 18d9d29fa7a489a417e2881d383f8b3f8c529579 Mon Sep 17 00:00:00 2001
From: Zhijie He <hezhijie0327@hotmail.com>
Date: Sun, 3 Sep 2023 10:23:59 +0800
Subject: [PATCH] change DNS_RC_REFUSED behavior

---
 src/dns_client.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/dns_client.c b/src/dns_client.c
index 0795abd..ebb46a2 100644
--- a/src/dns_client.c
+++ b/src/dns_client.c
@@ -3906,7 +3906,7 @@ static int _dns_client_pending_server_resolve(const struct dns_result *result, v
 	int ret = 0;
 	int has_soa = 0;

-	if (result->rtcode == DNS_RC_NXDOMAIN || result->has_soa == 1 || result->rtcode == DNS_RC_REFUSED ||
+	if (result->rtcode == DNS_RC_NXDOMAIN || result->has_soa == 1 ||
 		(result->rtcode == DNS_RC_NOERROR && result->ip_num == 0)) {
 		has_soa = 1;
 	}
--
2.42.0
