diff --git a/proxy/wireguard/client.go b/proxy/wireguard/client.go
index 4491886..d0067ff 100644
--- a/proxy/wireguard/client.go
+++ b/proxy/wireguard/client.go
@@ -151,7 +151,7 @@ func (h *Handler) Process(ctx context.Context, link *transport.Link, dialer inte
 
 	var addr netip.Addr
 	if ob.Target.Address.Family().IsDomain() {
-		ip, err := h.resolveRemote(ob.Target.Address.String())
+		ip, err := h.resolveLocal(ob.Target.Address.String())
 		if err != nil {
 			return errors.New("failed to resolve domain").Base(err)
 		}
@@ -212,7 +212,7 @@ func (h *Handler) Process(ctx context.Context, link *transport.Link, dialer inte
 		defer conn.Close()
 		c := &udpConnClient{
 			PacketConn:  conn.(*internet.PacketConnWrapper).PacketConn,
-			resolveFunc: h.resolveRemote,
+			resolveFunc: h.resolveLocal,
 			dest:        gonet.UDPAddrFromAddrPort(addrPort),
 		}
 		reader = c
