diff --git a/proxy/wireguard/client.go b/proxy/wireguard/client.go
index cae846077b93..59709b4bf915 100644
--- a/proxy/wireguard/client.go
+++ b/proxy/wireguard/client.go
@@ -138,6 +138,7 @@ func (h *Handler) Process(ctx context.Context, link *transport.Link, dialer inte
 	}
 	ob.Name = "wireguard"
 	ob.CanSpliceCopy = 3
+	dialer.SetOutboundGateway(ctx, ob)
 
 	if err := h.init(ctx); err != nil {
 		return err
