diff --git a/src/base/plugins/pluginsengine.cpp b/src/base/plugins/pluginsengine.cpp
index edfb276..3f41ca3 100644
--- a/src/base/plugins/pluginsengine.cpp
+++ b/src/base/plugins/pluginsengine.cpp
@@ -521,7 +521,7 @@ nonstd::expected<PluginsEngine::PluginEntry, QString> PluginsEngine::loadPlugin(
 {
     auto loadResult = Plugin::load(path);
     if (!loadResult)
-        return loadResult.get_unexpected();
+        return nonstd::make_unexpected(loadResult.error());
 
     PluginEntry pluginEntry {
         .plugin = loadResult.value(),
