diff --git a/src/app/(main)/settings/llm/ProviderList/Wenxin/index.tsx b/src/app/(main)/settings/llm/ProviderList/Wenxin/index.tsx
index afee7e97d219..ffde4872d07c 100644
--- a/src/app/(main)/settings/llm/ProviderList/Wenxin/index.tsx
+++ b/src/app/(main)/settings/llm/ProviderList/Wenxin/index.tsx
@@ -1,6 +1,5 @@
 'use client';
 
-import { Wenxin } from '@lobehub/icons';
 import { Input } from 'antd';
 import { useTranslation } from 'react-i18next';
 
@@ -41,6 +40,5 @@ export const useWenxinProvider = (): ProviderItem => {
         name: [KeyVaultsConfigKey, providerKey, 'secretKey'],
       },
     ],
-    title: <Wenxin.Combine size={32} type={'color'} />,
   };
 };
