diff --git a/src/app/(main)/settings/llm/ProviderList/Wenxin/index.tsx b/src/app/(main)/settings/llm/ProviderList/Wenxin/index.tsx
index afee7e97d219..a1557ccc040c 100644
--- a/src/app/(main)/settings/llm/ProviderList/Wenxin/index.tsx
+++ b/src/app/(main)/settings/llm/ProviderList/Wenxin/index.tsx
@@ -41,6 +41,6 @@ export const useWenxinProvider = (): ProviderItem => {
         name: [KeyVaultsConfigKey, providerKey, 'secretKey'],
       },
     ],
-    title: <Wenxin.Combine size={32} type={'color'} />,
+    title: <Wenxin.Combine extra={'一言'} size={24} type={'color'} />,
   };
 };
