diff --git a/Dockerfile b/Dockerfile
index 32ef8919c1cd..d3528c81e582 100644
--- a/Dockerfile
+++ b/Dockerfile
@@ -152,6 +152,8 @@ ENV \
     AZURE_API_KEY="" AZURE_API_VERSION="" AZURE_ENDPOINT="" AZURE_MODEL_LIST="" \
     # Baichuan
     BAICHUAN_API_KEY="" BAICHUAN_MODEL_LIST="" \
+    # Cloudflare
+    CLOUDFLARE_API_KEY="" CLOUDFLARE_BASE_URL_OR_ACCOUNT_ID="" CLOUDFLARE_MODEL_LIST="" \
     # DeepSeek
     DEEPSEEK_API_KEY="" DEEPSEEK_MODEL_LIST="" \
     # Fireworks AI
@@ -175,7 +177,7 @@ ENV \
     # Novita
     NOVITA_API_KEY="" NOVITA_MODEL_LIST="" \
     # Ollama
-    OLLAMA_MODEL_LIST="" OLLAMA_PROXY_URL="" \
+    ENABLED_OLLAMA="" OLLAMA_MODEL_LIST="" OLLAMA_PROXY_URL="" \
     # OpenAI
     OPENAI_API_KEY="" OPENAI_MODEL_LIST="" OPENAI_PROXY_URL="" \
     # OpenRouter
diff --git a/Dockerfile.database b/Dockerfile.database
index 4699983c0d27..cca4cc35ee78 100644
--- a/Dockerfile.database
+++ b/Dockerfile.database
@@ -187,6 +187,8 @@ ENV \
     AZURE_API_KEY="" AZURE_API_VERSION="" AZURE_ENDPOINT="" AZURE_MODEL_LIST="" \
     # Baichuan
     BAICHUAN_API_KEY="" BAICHUAN_MODEL_LIST="" \
+    # Cloudflare
+    CLOUDFLARE_API_KEY="" CLOUDFLARE_BASE_URL_OR_ACCOUNT_ID="" CLOUDFLARE_MODEL_LIST="" \
     # DeepSeek
     DEEPSEEK_API_KEY="" DEEPSEEK_MODEL_LIST="" \
     # Fireworks AI
@@ -210,7 +212,7 @@ ENV \
     # Novita
     NOVITA_API_KEY="" NOVITA_MODEL_LIST="" \
     # Ollama
-    OLLAMA_MODEL_LIST="" OLLAMA_PROXY_URL="" \
+    ENABLED_OLLAMA="" OLLAMA_MODEL_LIST="" OLLAMA_PROXY_URL="" \
     # OpenAI
     OPENAI_API_KEY="" OPENAI_MODEL_LIST="" OPENAI_PROXY_URL="" \
     # OpenRouter
diff --git a/src/config/llm.ts b/src/config/llm.ts
index 328f86b28e08..bc1db639be19 100644
--- a/src/config/llm.ts
+++ b/src/config/llm.ts
@@ -118,6 +118,7 @@ export const getLLMConfig = () => {
       ENABLED_CLOUDFLARE: z.boolean(),
       CLOUDFLARE_API_KEY: z.string().optional(),
       CLOUDFLARE_BASE_URL_OR_ACCOUNT_ID: z.string().optional(),
+      CLOUDFLARE_MODEL_LIST: z.string().optional(),
 
       ENABLED_AI360: z.boolean(),
       AI360_API_KEY: z.string().optional(),
@@ -273,6 +274,7 @@ export const getLLMConfig = () => {
         !!process.env.CLOUDFLARE_API_KEY && !!process.env.CLOUDFLARE_BASE_URL_OR_ACCOUNT_ID,
       CLOUDFLARE_API_KEY: process.env.CLOUDFLARE_API_KEY,
       CLOUDFLARE_BASE_URL_OR_ACCOUNT_ID: process.env.CLOUDFLARE_BASE_URL_OR_ACCOUNT_ID,
+      CLOUDFLARE_MODEL_LIST: process.env.CLOUDFLARE_MODEL_LIST,
 
       ENABLED_AI360: !!process.env.AI360_API_KEY,
       AI360_API_KEY: process.env.AI360_API_KEY,
