diff --git a/src/config/modelProviders/togetherai.ts b/src/config/modelProviders/togetherai.ts
index 1446f2c1a3e8..044c22496cab 100644
--- a/src/config/modelProviders/togetherai.ts
+++ b/src/config/modelProviders/togetherai.ts
@@ -1,14 +1,49 @@
 import { ModelProviderCard } from '@/types/llm';
 
-// ref :https://docs.together.ai/docs/chat-models
-// ref :https://www.together.ai/pricing
+// ref: https://docs.together.ai/docs/chat-models
+// ref: https://docs.together.ai/docs/vision-models
+// ref: https://docs.together.ai/docs/function-calling
+// ref: https://docs.together.ai/docs/language-and-code-models
+// ref: https://www.together.ai/pricing
 const TogetherAI: ModelProviderCard = {
   chatModels: [
+    {
+      description: '',
+      displayName: 'Llama 3.2 8B Instruct Turbo',
+      enabled: true,
+      id: 'meta-llama/Llama-3.2-3B-Instruct-Turbo',
+      tokens: 131_072,
+    },
+    {
+      description: '',
+      displayName: 'Llama 3.2 11B Vision Instruct Turbo (Free)',
+      enabled: true,
+      id: 'meta-llama/Llama-Vision-Free',
+      tokens: 131_072,
+      vision: true,
+    },
+    {
+      description: '',
+      displayName: 'Llama 3.2 11B Vision Instruct Turbo',
+      enabled: true,
+      id: 'meta-llama/Llama-3.2-11B-Vision-Instruct-Turbo',
+      tokens: 131_072,
+      vision: true,
+    },
+    {
+      description: '',
+      displayName: 'Llama 3.2 90B Vision Instruct Turbo',
+      enabled: true,
+      id: 'meta-llama/Llama-3.2-90B-Vision-Instruct-Turbo',
+      tokens: 131_072,
+      vision: true,
+    },
     {
       description:
         'Llama 3.1 8B 模型采用FP8量化，支持高达131,072个上下文标记，是开源模型中的佼佼者，适合复杂任务，表现优异于许多行业基准。',
       displayName: 'Llama 3.1 8B Instruct Turbo',
       enabled: true,
+      functionCall: true,
       id: 'meta-llama/Meta-Llama-3.1-8B-Instruct-Turbo',
       tokens: 131_072,
     },
@@ -17,6 +52,7 @@ const TogetherAI: ModelProviderCard = {
         'Llama 3.1 70B 模型经过精细调整，适用于高负载应用，量化至FP8提供更高效的计算能力和准确性，确保在复杂场景中的卓越表现。',
       displayName: 'Llama 3.1 70B Instruct Turbo',
       enabled: true,
+      functionCall: true,
       id: 'meta-llama/Meta-Llama-3.1-70B-Instruct-Turbo',
       tokens: 131_072,
     },
@@ -25,6 +61,7 @@ const TogetherAI: ModelProviderCard = {
         '405B 的 Llama 3.1 Turbo 模型，为大数据处理提供超大容量的上下文支持，在超大规模的人工智能应用中表现突出。',
       displayName: 'Llama 3.1 405B Instruct Turbo',
       enabled: true,
+      functionCall: true,
       id: 'meta-llama/Meta-Llama-3.1-405B-Instruct-Turbo',
       tokens: 8192,
     },
@@ -54,14 +91,14 @@ const TogetherAI: ModelProviderCard = {
       tokens: 8192,
     },
     {
-      description: 'LLaMA-3 Chat (8B) 提供多语言支持，涵盖丰富的领域知识。',
-      displayName: 'LLaMA-3 Chat (8B)',
+      description: 'Llama 3 8B Instruct Reference 提供多语言支持，涵盖丰富的领域知识。',
+      displayName: 'Llama 3 8B Instruct Reference',
       id: 'meta-llama/Llama-3-8b-chat-hf',
       tokens: 8192,
     },
     {
-      description: 'LLaMA-3 Chat (70B) 是功能强大的聊天模型，支持复杂的对话需求。',
-      displayName: 'LLaMA-3 Chat (70B)',
+      description: 'Llama 3 70B Instruct Reference 是功能强大的聊天模型，支持复杂的对话需求。',
+      displayName: 'Llama 3 70B Instruct Reference',
       id: 'meta-llama/Llama-3-70b-chat-hf',
       tokens: 8192,
     },
@@ -71,6 +108,12 @@ const TogetherAI: ModelProviderCard = {
       id: 'meta-llama/Llama-2-13b-chat-hf',
       tokens: 4096,
     },
+    {
+      description: '',
+      displayName: 'LLaMA-2 (70B)',
+      id: 'meta-llama/Llama-2-70b-hf',
+      tokens: 4096,
+    },
     {
       description: 'Gemma 2 9B 由Google开发，提供高效的指令响应和综合能力。',
       displayName: 'Gemma 2 9B',
@@ -94,7 +137,6 @@ const TogetherAI: ModelProviderCard = {
     {
       description: 'Mistral (7B) Instruct v0.3 提供高效的计算能力和自然语言理解，适合广泛的应用。',
       displayName: 'Mistral (7B) Instruct v0.3',
-      enabled: true,
       id: 'mistralai/Mistral-7B-Instruct-v0.3',
       tokens: 32_768,
     },
@@ -107,16 +149,31 @@ const TogetherAI: ModelProviderCard = {
     {
       description: 'Mistral (7B) Instruct 以高性能著称，适用于多种语言任务。',
       displayName: 'Mistral (7B) Instruct',
+      enabled: true,
+      functionCall: true,
       id: 'mistralai/Mistral-7B-Instruct-v0.1',
       tokens: 8192,
     },
+    {
+      description: '',
+      displayName: 'Mistral (7B)',
+      id: 'mistralai/Mistral-7B-v0.1',
+      tokens: 8192,
+    },
     {
       description: 'Mixtral-8x7B Instruct (46.7B) 提供高容量的计算框架，适合大规模数据处理。',
       displayName: 'Mixtral-8x7B Instruct (46.7B)',
       enabled: true,
+      functionCall: true,
       id: 'mistralai/Mixtral-8x7B-Instruct-v0.1',
       tokens: 32_768,
     },
+    {
+      description: '',
+      displayName: 'Mixtral-8x7B (46.7B)',
+      id: 'mistralai/Mixtral-8x7B-v0.1',
+      tokens: 32_768,
+    },
     {
       description: 'Mixtral-8x22B Instruct (141B) 是一款超级大语言模型，支持极高的处理需求。',
       displayName: 'Mixtral-8x22B Instruct (141B)',
@@ -131,10 +188,23 @@ const TogetherAI: ModelProviderCard = {
       id: 'deepseek-ai/deepseek-llm-67b-chat',
       tokens: 4096,
     },
+    {
+      description: '',
+      displayName: 'Qwen 2.5 7B Instruct Turbo',
+      enabled: true,
+      id: 'Qwen/Qwen2.5-7B-Instruct-Turbo',
+      tokens: 32_768,
+    },
+    {
+      description: '',
+      displayName: 'Qwen 2.5 72B Instruct Turbo',
+      enabled: true,
+      id: 'Qwen/Qwen2.5-72B-Instruct-Turbo',
+      tokens: 32_768,
+    },
     {
       description: 'Qwen 2 Instruct (72B) 为企业级应用提供精准的指令理解和响应。',
       displayName: 'Qwen 2 Instruct (72B)',
-      enabled: true,
       id: 'Qwen/Qwen2-72B-Instruct',
       tokens: 32_768,
     },
diff --git a/src/libs/agent-runtime/togetherai/__snapshots__/index.test.ts.snap b/src/libs/agent-runtime/togetherai/__snapshots__/index.test.ts.snap
index 4a969451ce8d..a684d98863b1 100644
--- a/src/libs/agent-runtime/togetherai/__snapshots__/index.test.ts.snap
+++ b/src/libs/agent-runtime/togetherai/__snapshots__/index.test.ts.snap
@@ -425,7 +425,7 @@ exports[`LobeTogetherAI > models > should get models 1`] = `
   {
     "description": "instruct fine-tuned version of Mistral-7B-v0.1",
     "displayName": "Mistral (7B) Instruct",
-    "enabled": false,
+    "enabled": true,
     "functionCall": false,
     "id": "mistralai/Mistral-7B-Instruct-v0.1",
     "maxOutput": 4096,
