diff --git a/src/libs/agent-runtime/sensenova/index.ts b/src/libs/agent-runtime/sensenova/index.ts
index 59e1f592b772..1b6390c4be16 100644
--- a/src/libs/agent-runtime/sensenova/index.ts
+++ b/src/libs/agent-runtime/sensenova/index.ts
@@ -7,9 +7,9 @@ import { AgentRuntimeError } from '../utils/createError';
 import { debugStream } from '../utils/debugStream';
 import { desensitizeUrl } from '../utils/desensitizeUrl';
 import { handleOpenAIError } from '../utils/handleOpenAIError';
-import { convertOpenAIMessages } from '../utils/openaiHelpers';
 import { StreamingResponse } from '../utils/response';
 import { OpenAIStream } from '../utils/streams';
+
 import { generateApiToken } from './authToken';
 
 const DEFAULT_BASE_URL = 'https://api.sensenova.cn/compatible-mode/v1';
@@ -82,10 +82,9 @@ export class LobeSenseNovaAI implements LobeRuntimeAI {
   }
 
   private async buildCompletionsParams(payload: ChatStreamPayload) {
-    const { frequency_penalty, messages, temperature, top_p, ...params } = payload;
+    const { frequency_penalty, temperature, top_p, ...params } = payload;
 
     return {
-      messages: await convertOpenAIMessages(messages as any),
       ...params,
       frequency_penalty: (frequency_penalty !== undefined && frequency_penalty > 0 && frequency_penalty <= 2) ? frequency_penalty : undefined,
       stream: true,
