diff --git a/src/app/(main)/settings/llm/ProviderList/providers.tsx b/src/app/(main)/settings/llm/ProviderList/providers.tsx
index f397d3afb626..0502219eb6dc 100644
--- a/src/app/(main)/settings/llm/ProviderList/providers.tsx
+++ b/src/app/(main)/settings/llm/ProviderList/providers.tsx
@@ -45,7 +45,6 @@ export const useProviderList = (): ProviderItem[] => {
       AnthropicProviderCard,
       BedrockProvider,
       GoogleProviderCard,
-      DeepSeekProviderCard,
       OpenRouterProviderCard,
       GroqProviderCard,
       NovitaProviderCard,
@@ -54,6 +53,7 @@ export const useProviderList = (): ProviderItem[] => {
       TogetherAIProviderCard,
       FireworksAIProviderCard,
       UpstageProviderCard,
+      DeepSeekProviderCard,
       QwenProviderCard,
       SparkProviderCard,
       ZhiPuProviderCard,
diff --git a/src/config/modelProviders/index.ts b/src/config/modelProviders/index.ts
index 3e1504b213e2..7e2ebd063c21 100644
--- a/src/config/modelProviders/index.ts
+++ b/src/config/modelProviders/index.ts
@@ -62,7 +62,6 @@ export const DEFAULT_MODEL_PROVIDER_LIST = [
   AnthropicProvider,
   BedrockProvider,
   GoogleProvider,
-  DeepSeekProvider,
   OpenRouterProvider,
   GroqProvider,
   NovitaProvider,
@@ -71,6 +70,7 @@ export const DEFAULT_MODEL_PROVIDER_LIST = [
   TogetherAIProvider,
   FireworksAIProvider,
   UpstageProvider,
+  DeepSeekProvider,
   QwenProvider,
   SparkProvider,
   ZhiPuProvider,
