diff --git a/src/config/modelProviders/sensenova.ts b/src/config/modelProviders/sensenova.ts
index fe4965a6897b..54125ae94bc1 100644
--- a/src/config/modelProviders/sensenova.ts
+++ b/src/config/modelProviders/sensenova.ts
@@ -20,7 +20,6 @@ const SenseNova: ModelProviderCard = {
     {
       description: '最新版本模型 (V5.5)，16K上下文长度，支持多图的输入，全面实现模型基础能力优化，在对象属性识别、空间关系、动作事件识别、场景理解、情感识别、逻辑常识推理和文本理解生成上都实现了较大提升。',
       displayName: 'SenseChat 5.5 Vision',
-      enabled: true,
       id: 'SenseChat-Vision',
       pricing: {
         currency: 'CNY',
diff --git a/src/libs/agent-runtime/sensenova/index.ts b/src/libs/agent-runtime/sensenova/index.ts
index 59e1f592b772..1b6390c4be16 100644
--- a/src/libs/agent-runtime/sensenova/index.ts
+++ b/src/libs/agent-runtime/sensenova/index.ts
@@ -7,9 +7,9 @@ import { AgentRuntimeError } from '../utils/createError';
 import { debugStream } from '../utils/debugStream';
 import { desensitizeUrl } from '../utils/desensitizeUrl';
 import { handleOpenAIError } from '../utils/handleOpenAIError';
-import { convertOpenAIMessages } from '../utils/openaiHelpers';
 import { StreamingResponse } from '../utils/response';
 import { OpenAIStream } from '../utils/streams';
+
 import { generateApiToken } from './authToken';
 
 const DEFAULT_BASE_URL = 'https://api.sensenova.cn/compatible-mode/v1';
@@ -82,10 +82,9 @@ export class LobeSenseNovaAI implements LobeRuntimeAI {
   }
 
   private async buildCompletionsParams(payload: ChatStreamPayload) {
-    const { frequency_penalty, messages, temperature, top_p, ...params } = payload;
+    const { frequency_penalty, temperature, top_p, ...params } = payload;
 
     return {
-      messages: await convertOpenAIMessages(messages as any),
       ...params,
       frequency_penalty: (frequency_penalty !== undefined && frequency_penalty > 0 && frequency_penalty <= 2) ? frequency_penalty : undefined,
       stream: true,
