diff --git a/packages/model-runtime/src/anthropic/index.ts b/packages/model-runtime/src/anthropic/index.ts
index 7e914e320b2a9..ae45280e66abe 100644
--- a/packages/model-runtime/src/anthropic/index.ts
+++ b/packages/model-runtime/src/anthropic/index.ts
@@ -61,7 +61,13 @@ export class LobeAnthropicAI implements LobeRuntimeAI {
     this.client = new Anthropic({
       apiKey,
       baseURL,
-      defaultHeaders: { ...defaultHeaders, 'anthropic-beta': betaHeaders },
+      defaultHeaders: {
+        ...defaultHeaders,
+        'anthropic-beta': betaHeaders,
+        ...(process.env.ENABLE_AIHUBMIX_APP_CODE === '1' && {
+          'APP-Code': 'LobeHub',
+        }),
+      },
       ...res,
     });
     this.baseURL = this.client.baseURL;
diff --git a/packages/model-runtime/src/google/index.ts b/packages/model-runtime/src/google/index.ts
index 1154a76ec6739..040b720aba14e 100644
--- a/packages/model-runtime/src/google/index.ts
+++ b/packages/model-runtime/src/google/index.ts
@@ -116,7 +116,12 @@ export class LobeGoogleAI implements LobeRuntimeAI {
     if (!apiKey) throw AgentRuntimeError.createError(AgentRuntimeErrorType.InvalidProviderAPIKey);
 
     const httpOptions = baseURL
-      ? ({ baseUrl: baseURL, headers: defaultHeaders } as HttpOptions)
+      ? ({ baseUrl: baseURL, headers: {
+        ...defaultHeaders,
+        ...(process.env.ENABLE_AIHUBMIX_APP_CODE === '1' && {
+          'APP-Code': 'LobeHub',
+        }),
+      } } as HttpOptions)
       : undefined;
 
     this.apiKey = apiKey;
diff --git a/packages/model-runtime/src/openai/index.ts b/packages/model-runtime/src/openai/index.ts
index 5d8001605ed89..d78e4c238dbc0 100644
--- a/packages/model-runtime/src/openai/index.ts
+++ b/packages/model-runtime/src/openai/index.ts
@@ -61,6 +61,13 @@ export const LobeOpenAI = createOpenAICompatibleRuntime({
       };
     },
   },
+  constructorOptions: {
+    defaultHeaders: {
+      ...(process.env.ENABLE_AIHUBMIX_APP_CODE === '1' && {
+        'APP-Code': 'LobeHub',
+      }),
+    },
+  },
   debug: {
     chatCompletion: () => process.env.DEBUG_OPENAI_CHAT_COMPLETION === '1',
     responses: () => process.env.DEBUG_OPENAI_RESPONSES === '1',
