diff --git a/src/config/modelProviders/index.ts b/src/config/modelProviders/index.ts
index b535d5c514dc..3fa1e590c1e5 100644
--- a/src/config/modelProviders/index.ts
+++ b/src/config/modelProviders/index.ts
@@ -81,6 +81,7 @@ export const DEFAULT_MODEL_PROVIDER_LIST = [
   DeepSeekProvider,
   HuggingFaceProvider,
   OpenRouterProvider,
+  CloudflareProvider,
   GithubProvider,
   NovitaProvider,
   TogetherAIProvider,
@@ -104,7 +105,6 @@ export const DEFAULT_MODEL_PROVIDER_LIST = [
   MinimaxProvider,
   Ai360Provider,
   TaichuProvider,
-  CloudflareProvider,
   SiliconCloudProvider,
 ];
 
diff --git a/src/server/modules/AgentRuntime/index.ts b/src/server/modules/AgentRuntime/index.ts
index 6d2ead94a01a..388c2d07ad1c 100644
--- a/src/server/modules/AgentRuntime/index.ts
+++ b/src/server/modules/AgentRuntime/index.ts
@@ -214,10 +214,7 @@ const getLlmOptionsFromPayload = (provider: string, payload: JWTPayload) => {
       const { CLOUDFLARE_API_KEY, CLOUDFLARE_BASE_URL_OR_ACCOUNT_ID } = getLLMConfig();
 
       const apiKey = apiKeyManager.pick(payload?.apiKey || CLOUDFLARE_API_KEY);
-      const baseURLOrAccountID =
-        payload.apiKey && payload.cloudflareBaseURLOrAccountID
-          ? payload.cloudflareBaseURLOrAccountID
-          : CLOUDFLARE_BASE_URL_OR_ACCOUNT_ID;
+      const baseURLOrAccountID = payload?.cloudflareBaseURLOrAccountID || CLOUDFLARE_BASE_URL_OR_ACCOUNT_ID;
 
       return { apiKey, baseURLOrAccountID };
     }
