diff --git a/src/config/modelProviders/stepfun.ts b/src/config/modelProviders/stepfun.ts
index b37da210f792..793a0c75b3cb 100644
--- a/src/config/modelProviders/stepfun.ts
+++ b/src/config/modelProviders/stepfun.ts
@@ -14,6 +14,7 @@ const Stepfun: ModelProviderCard = {
     {
       description: '具备超长上下文处理能力，尤其适合长文档分析。',
       displayName: 'Step 1 256K',
+      functionCall: true,
       id: 'step-1-256k',
       tokens: 256_000,
     },
@@ -21,6 +22,7 @@ const Stepfun: ModelProviderCard = {
       description: '平衡性能与成本，适合一般场景。',
       displayName: 'Step 1 128K',
       enabled: true,
+      functionCall: true,
       id: 'step-1-128k',
       tokens: 128_000,
     },
@@ -28,6 +30,7 @@ const Stepfun: ModelProviderCard = {
       description: '支持中等长度的对话，适用于多种应用场景。',
       displayName: 'Step 1 32K',
       enabled: true,
+      functionCall: true,
       id: 'step-1-32k',
       tokens: 32_000,
     },
@@ -35,6 +38,7 @@ const Stepfun: ModelProviderCard = {
       description: '小型模型，适合轻量级任务。',
       displayName: 'Step 1 8K',
       enabled: true,
+      functionCall: true,
       id: 'step-1-8k',
       tokens: 8000,
     },
@@ -49,6 +53,7 @@ const Stepfun: ModelProviderCard = {
       description: '支持视觉输入，增强多模态交互体验。',
       displayName: 'Step 1V 32K',
       enabled: true,
+      functionCall: true,
       id: 'step-1v-32k',
       tokens: 32_000,
       vision: true,
@@ -57,6 +62,7 @@ const Stepfun: ModelProviderCard = {
       description: '小型视觉模型，适合基本的图文任务。',
       displayName: 'Step 1V 8K',
       enabled: true,
+      functionCall: true,
       id: 'step-1v-8k',
       tokens: 8000,
       vision: true,
