diff --git a/src/config/modelProviders/stepfun.ts b/src/config/modelProviders/stepfun.ts
index b37da210f792..1e4a022484cb 100644
--- a/src/config/modelProviders/stepfun.ts
+++ b/src/config/modelProviders/stepfun.ts
@@ -8,12 +8,14 @@ const Stepfun: ModelProviderCard = {
       description: '支持大规模上下文交互，适合复杂对话场景。',
       displayName: 'Step 2 16K',
       enabled: true,
+      functionCall: true,
       id: 'step-2-16k',
       tokens: 16_000,
     },
     {
       description: '具备超长上下文处理能力，尤其适合长文档分析。',
       displayName: 'Step 1 256K',
+      functionCall: true,
       id: 'step-1-256k',
       tokens: 256_000,
     },
@@ -21,6 +23,7 @@ const Stepfun: ModelProviderCard = {
       description: '平衡性能与成本，适合一般场景。',
       displayName: 'Step 1 128K',
       enabled: true,
+      functionCall: true,
       id: 'step-1-128k',
       tokens: 128_000,
     },
@@ -28,6 +31,7 @@ const Stepfun: ModelProviderCard = {
       description: '支持中等长度的对话，适用于多种应用场景。',
       displayName: 'Step 1 32K',
       enabled: true,
+      functionCall: true,
       id: 'step-1-32k',
       tokens: 32_000,
     },
@@ -35,6 +39,7 @@ const Stepfun: ModelProviderCard = {
       description: '小型模型，适合轻量级任务。',
       displayName: 'Step 1 8K',
       enabled: true,
+      functionCall: true,
       id: 'step-1-8k',
       tokens: 8000,
     },
@@ -42,6 +47,7 @@ const Stepfun: ModelProviderCard = {
       description: '高速模型，适合实时对话。',
       displayName: 'Step 1 Flash',
       enabled: true,
+      functionCall: true,
       id: 'step-1-flash',
       tokens: 8000,
     },
@@ -49,6 +55,7 @@ const Stepfun: ModelProviderCard = {
       description: '支持视觉输入，增强多模态交互体验。',
       displayName: 'Step 1V 32K',
       enabled: true,
+      functionCall: true,
       id: 'step-1v-32k',
       tokens: 32_000,
       vision: true,
@@ -57,6 +64,7 @@ const Stepfun: ModelProviderCard = {
       description: '小型视觉模型，适合基本的图文任务。',
       displayName: 'Step 1V 8K',
       enabled: true,
+      functionCall: true,
       id: 'step-1v-8k',
       tokens: 8000,
       vision: true,
diff --git a/src/libs/agent-runtime/stepfun/index.ts b/src/libs/agent-runtime/stepfun/index.ts
index 02727361279e..4ae98b6fe3c7 100644
--- a/src/libs/agent-runtime/stepfun/index.ts
+++ b/src/libs/agent-runtime/stepfun/index.ts
@@ -3,6 +3,14 @@ import { LobeOpenAICompatibleFactory } from '../utils/openaiCompatibleFactory';
 
 export const LobeStepfunAI = LobeOpenAICompatibleFactory({
   baseURL: 'https://api.stepfun.com/v1',
+  chatCompletion: {
+    handlePayload: (payload) => {
+      return {
+        ...payload,
+        stream: !payload.tools,
+      } as any;
+    },
+  },
   debug: {
     chatCompletion: () => process.env.DEBUG_STEPFUN_CHAT_COMPLETION === '1',
   },
