diff --git a/src/config/aiModels/siliconcloud.ts b/src/config/aiModels/siliconcloud.ts
index 261f708f7d827..81f3eb5e63b0a 100644
--- a/src/config/aiModels/siliconcloud.ts
+++ b/src/config/aiModels/siliconcloud.ts
@@ -1,4 +1,4 @@
-import { AIChatModelCard } from '@/types/aiModel';
+import { AIChatModelCard, AIImageModelCard } from '@/types/aiModel';
 
 // https://siliconflow.cn/zh-cn/models
 const siliconcloudChatModels: AIChatModelCard[] = [
@@ -830,6 +830,27 @@ const siliconcloudChatModels: AIChatModelCard[] = [
   },
 ];
 
-export const allModels = [...siliconcloudChatModels];
+const siliconcloudImageModels: AIImageModelCard[] = [
+  {
+    description:
+      'Kolors 是由快手 Kolors 团队开发的基于潜在扩散的大规模文本到图像生成模型。该模型通过数十亿文本-图像对的训练，在视觉质量、复杂语义准确性以及中英文字符渲染方面展现出显著优势。它不仅支持中英文输入，在理解和生成中文特定内容方面也表现出色',
+    displayName: 'Kolors',
+    enabled: true,
+    id: 'Kwai-Kolors/Kolors',
+    parameters: {
+      prompt: {
+        default: '',
+      },
+      size: {
+        default: '1024x1024',
+        enum: ['1024x1024', '960x1280', '768x1024', '720x1440', '720x1280'],
+      },
+    },
+    releasedAt: '2024-07-06',
+    type: 'image',
+  },
+];
+
+export const allModels = [...siliconcloudChatModels, ...siliconcloudImageModels];
 
 export default allModels;
diff --git a/src/config/aiModels/stepfun.ts b/src/config/aiModels/stepfun.ts
index 117c3b5fc991e..a089098264710 100644
--- a/src/config/aiModels/stepfun.ts
+++ b/src/config/aiModels/stepfun.ts
@@ -1,4 +1,4 @@
-import { AIChatModelCard } from '@/types/aiModel';
+import { AIChatModelCard, AIImageModelCard } from '@/types/aiModel';
 
 // https://platform.stepfun.com/docs/pricing/details
 
@@ -275,6 +275,46 @@ const stepfunChatModels: AIChatModelCard[] = [
   },
 ];
 
-export const allModels = [...stepfunChatModels];
+const stepfunImageModels: AIImageModelCard[] = [
+  // https://platform.stepfun.com/docs/llm/image
+  {
+    description:
+      '阶跃星辰新一代生图模型,该模型专注于图像生成任务,能够根据用户提供的文本描述,生成高质量的图像。新模型生成图片质感更真实，中英文文字生成能力更强。',
+    displayName: 'Step 2X Large',
+    enabled: true,
+    id: 'step-2x-large',
+    parameters: {
+      prompt: {
+        default: '',
+      },
+      size: {
+        default: '1024x1024',
+        enum: ['256x256', '512x512', '768x768', '1024x1024', '1280x800', '800x1280'],
+      },
+    },
+    releasedAt: '2024-08-07',
+    type: 'image',
+  },
+  {
+    description:
+      '该模型拥有强大的图像生成能力，支持文本描述作为输入方式。具备原生的中文支持，能够更好的理解和处理中文文本描述，并且能够更准确地捕捉文本描述中的语义信息，并将其转化为图像特征，从而实现更精准的图像生成。模型能够根据输入生成高分辨率、高质量的图像，并具备一定的风格迁移能力。',
+    displayName: 'Step 1X Medium',
+    enabled: true,
+    id: 'step-1x-medium',
+    parameters: {
+      prompt: {
+        default: '',
+      },
+      size: {
+        default: '1024x1024',
+        enum: ['256x256', '512x512', '768x768', '1024x1024', '1280x800', '800x1280'],
+      },
+    },
+    releasedAt: '2025-07-15',
+    type: 'image',
+  },
+];
+
+export const allModels = [...stepfunChatModels, ...stepfunImageModels];
 
 export default allModels;
diff --git a/src/config/aiModels/wenxin.ts b/src/config/aiModels/wenxin.ts
index 4be4e6bbf736b..5c21ffa576936 100644
--- a/src/config/aiModels/wenxin.ts
+++ b/src/config/aiModels/wenxin.ts
@@ -1,4 +1,4 @@
-import { AIChatModelCard } from '@/types/aiModel';
+import { AIChatModelCard, AIImageModelCard } from '@/types/aiModel';
 
 const wenxinChatModels: AIChatModelCard[] = [
   {
@@ -564,6 +564,45 @@ const wenxinChatModels: AIChatModelCard[] = [
   },
 ];
 
-export const allModels = [...wenxinChatModels];
+const wenxinImageModels: AIImageModelCard[] = [
+  {
+    description:
+      '百度自研的iRAG（image based RAG），检索增强的文生图技术，将百度搜索的亿级图片资源跟强大的基础模型能力相结合，就可以生成各种超真实的图片，整体效果远远超过文生图原生系统，去掉了AI味儿，而且成本很低。iRAG具备无幻觉、超真实、立等可取等特点。',
+    displayName: 'ERINE iRAG',
+    enabled: true,
+    id: 'irag-1.0',
+    parameters: {
+      prompt: {
+        default: '',
+      },
+      size: {
+        default: '1024x1024',
+        enum: ['768x768', '1024x1024', '1536x1536', '2048x2048', '1024x768', '2048x1536', '768x1024', '1536x2048', '1024x576', '2048x1152', '576x1024', '1152x2048'],
+      },
+    },
+    releasedAt: '2025-02-05',
+    type: 'image',
+  },
+  {
+    description:
+      '具有120亿参数的修正流变换器，能够根据文本描述生成图像。',
+    displayName: 'FLUX.1-schnell',
+    enabled: true,
+    id: 'flux.1-schnell',
+    parameters: {
+      prompt: {
+        default: '',
+      },
+      size: {
+        default: '1024x1024',
+        enum: ['768x768', '1024x1024', '1536x1536', '2048x2048', '1024x768', '2048x1536', '768x1024', '1536x2048', '1024x576', '2048x1152', '576x1024', '1152x2048'],
+      },
+    },
+    releasedAt: '2025-03-27',
+    type: 'image',
+  },
+];
+
+export const allModels = [...wenxinChatModels, ...wenxinImageModels];
 
 export default allModels;
diff --git a/src/config/aiModels/xai.ts b/src/config/aiModels/xai.ts
index b65b18ec620ee..29752e50cacc3 100644
--- a/src/config/aiModels/xai.ts
+++ b/src/config/aiModels/xai.ts
@@ -1,4 +1,4 @@
-import { AIChatModelCard } from '@/types/aiModel';
+import { AIChatModelCard, AIImageModelCard } from '@/types/aiModel';
 
 // https://docs.x.ai/docs/models
 const xaiChatModels: AIChatModelCard[] = [
@@ -158,6 +158,29 @@ const xaiChatModels: AIChatModelCard[] = [
   },
 ];
 
-export const allModels = [...xaiChatModels];
+const xaiImageModels: AIImageModelCard[] = [
+  {
+    description:
+      '我们最新的图像生成模型可以根据文本提示生成生动逼真的图像。它在营销、社交媒体和娱乐等领域的图像生成方面表现出色。',
+    displayName: 'Grok 2 Image 1212',
+    enabled: true,
+    id: 'grok-2-image-1212',
+    parameters: {
+      prompt: {
+        default: '',
+      },
+      /* Note: https://docs.x.ai/docs/guides/image-generations#parameters
+      size: {
+        default: '1024x1024',
+        enum: ['1024x1024', '768x1344', '864x1152', '1344x768', '1152x864', '1440x720', '720x1440'],
+      },
+      */
+    },
+    releasedAt: '2024-12-12',
+    type: 'image',
+  },
+];
+
+export const allModels = [...xaiChatModels, ...xaiImageModels];
 
 export default allModels;
