diff --git a/packages/database/src/models/chunk.ts b/packages/database/src/models/chunk.ts
index f5b0bb3..0651570 100644
--- a/packages/database/src/models/chunk.ts
+++ b/packages/database/src/models/chunk.ts
@@ -209,9 +209,14 @@ export class ChunkModel {
       .leftJoin(embeddings, eq(chunks.id, embeddings.chunkId))
       .leftJoin(fileChunks, eq(chunks.id, fileChunks.chunkId))
       .leftJoin(files, eq(files.id, fileChunks.fileId))
-      .where(inArray(fileChunks.fileId, fileIds))
+      .where(
+        and(
+          inArray(fileChunks.fileId, fileIds),
+          // Ensure we only get chunks with embeddings
+          sql`embeddings.embeddings IS NOT NULL`,
+        ),
+      )
       .orderBy((t) => desc(t.similarity))
-      // Relaxed to 15 for now
       .limit(topK);
 
     return result.map((item) => {
