diff --git a/locales/ar/tool.json b/locales/ar/tool.json
index 5d90399e9f570..09053796557e0 100644
--- a/locales/ar/tool.json
+++ b/locales/ar/tool.json
@@ -21,8 +21,16 @@
     },
     "searchEngine": "محرك البحث:",
     "searchResult": "عدد النتائج:",
+    "searchTimeRange": "نطاق الوقت:",
     "summary": "ملخص",
     "summaryTooltip": "تلخيص المحتوى الحالي",
+    "timeRange": {
+      "anytime": "في أي وقت",
+      "day": "اليوم الماضي",
+      "month": "الشهر الماضي",
+      "week": "الأسبوع الماضي",
+      "year": "السنة الماضية"
+    },
     "viewMoreResults": "عرض المزيد من {{results}} نتيجة"
   }
 }
diff --git a/locales/bg-BG/tool.json b/locales/bg-BG/tool.json
index 5591223d2e9ea..bb856b1212ab8 100644
--- a/locales/bg-BG/tool.json
+++ b/locales/bg-BG/tool.json
@@ -21,8 +21,16 @@
     },
     "searchEngine": "Търсачка:",
     "searchResult": "Брой резултати:",
+    "searchTimeRange": "Времеви диапазон:",
     "summary": "Резюме",
     "summaryTooltip": "Резюме на текущото съдържание",
+    "timeRange": {
+      "anytime": "Без ограничения на времето",
+      "day": "Последен ден",
+      "month": "Последен месец",
+      "week": "Последна седмица",
+      "year": "Последна година"
+    },
     "viewMoreResults": "Вижте още {{results}} резултата"
   }
 }
diff --git a/locales/de-DE/tool.json b/locales/de-DE/tool.json
index 24f5bc750e18d..d887a42f3c2d3 100644
--- a/locales/de-DE/tool.json
+++ b/locales/de-DE/tool.json
@@ -21,8 +21,16 @@
     },
     "searchEngine": "Suchmaschine:",
     "searchResult": "Anzahl der Suchergebnisse:",
+    "searchTimeRange": "Zeitspanne:",
     "summary": "Zusammenfassung",
     "summaryTooltip": "Aktuellen Inhalt zusammenfassen",
+    "timeRange": {
+      "anytime": "Keine Zeitbeschränkung",
+      "day": "Letzte 24 Stunden",
+      "month": "Letzter Monat",
+      "week": "Letzte Woche",
+      "year": "Letztes Jahr"
+    },
     "viewMoreResults": "Weitere {{results}} Ergebnisse anzeigen"
   }
 }
diff --git a/locales/en-US/tool.json b/locales/en-US/tool.json
index 0798577aefb42..6103283e6540e 100644
--- a/locales/en-US/tool.json
+++ b/locales/en-US/tool.json
@@ -21,8 +21,16 @@
     },
     "searchEngine": "Search engine:",
     "searchResult": "Number of searches:",
+    "searchTimeRange": "Time Range:",
     "summary": "Summary",
     "summaryTooltip": "Summarize the current content",
+    "timeRange": {
+      "anytime": "Anytime",
+      "day": "Past Day",
+      "month": "Past Month",
+      "week": "Past Week",
+      "year": "Past Year"
+    },
     "viewMoreResults": "View {{results}} more results"
   }
 }
diff --git a/locales/es-ES/tool.json b/locales/es-ES/tool.json
index ba38af7e6b4c6..9f0d6eacc58a0 100644
--- a/locales/es-ES/tool.json
+++ b/locales/es-ES/tool.json
@@ -21,8 +21,16 @@
     },
     "searchEngine": "Motor de búsqueda:",
     "searchResult": "Número de búsquedas:",
+    "searchTimeRange": "Rango de tiempo:",
     "summary": "Resumen",
     "summaryTooltip": "Resumir el contenido actual",
+    "timeRange": {
+      "anytime": "Sin límite de tiempo",
+      "day": "Últimas 24 horas",
+      "month": "Último mes",
+      "week": "Última semana",
+      "year": "Último año"
+    },
     "viewMoreResults": "Ver más {{results}} resultados"
   }
 }
diff --git a/locales/fa-IR/tool.json b/locales/fa-IR/tool.json
index f728c1c6c04f4..c3ced50544c2a 100644
--- a/locales/fa-IR/tool.json
+++ b/locales/fa-IR/tool.json
@@ -21,8 +21,16 @@
     },
     "searchEngine": "موتور جستجو:",
     "searchResult": "تعداد جستجو:",
+    "searchTimeRange": "محدوده زمانی:",
     "summary": "خلاصه",
     "summaryTooltip": "خلاصه محتوای فعلی",
+    "timeRange": {
+      "anytime": "بدون محدودیت زمانی",
+      "day": "یک روز گذشته",
+      "month": "یک ماه گذشته",
+      "week": "یک هفته گذشته",
+      "year": "یک سال گذشته"
+    },
     "viewMoreResults": "مشاهده {{results}} نتیجه بیشتر"
   }
 }
diff --git a/locales/fr-FR/tool.json b/locales/fr-FR/tool.json
index 3aa6bf3ba763c..cd8fff47ac0e7 100644
--- a/locales/fr-FR/tool.json
+++ b/locales/fr-FR/tool.json
@@ -21,8 +21,16 @@
     },
     "searchEngine": "Moteur de recherche :",
     "searchResult": "Nombre de recherches :",
+    "searchTimeRange": "Plage horaire :",
     "summary": "Résumé",
     "summaryTooltip": "Résumer le contenu actuel",
+    "timeRange": {
+      "anytime": "Tout le temps",
+      "day": "Dernière journée",
+      "month": "Dernier mois",
+      "week": "Dernière semaine",
+      "year": "Dernière année"
+    },
     "viewMoreResults": "Voir {{results}} résultats supplémentaires"
   }
 }
diff --git a/locales/it-IT/tool.json b/locales/it-IT/tool.json
index 12339bbec0df7..d2acc45b7c972 100644
--- a/locales/it-IT/tool.json
+++ b/locales/it-IT/tool.json
@@ -21,8 +21,16 @@
     },
     "searchEngine": "Motore di ricerca:",
     "searchResult": "Numero di ricerche:",
+    "searchTimeRange": "Intervallo di tempo:",
     "summary": "Riepilogo",
     "summaryTooltip": "Riepiloga il contenuto attuale",
+    "timeRange": {
+      "anytime": "Nessun limite di tempo",
+      "day": "Ultime 24 ore",
+      "month": "Ultimo mese",
+      "week": "Ultima settimana",
+      "year": "Ultimo anno"
+    },
     "viewMoreResults": "Visualizza altri {{results}} risultati"
   }
 }
diff --git a/locales/ja-JP/tool.json b/locales/ja-JP/tool.json
index 30804a81b1602..b4fb004789a29 100644
--- a/locales/ja-JP/tool.json
+++ b/locales/ja-JP/tool.json
@@ -21,8 +21,16 @@
     },
     "searchEngine": "検索エンジン：",
     "searchResult": "検索結果の数：",
+    "searchTimeRange": "時間範囲：",
     "summary": "要約",
     "summaryTooltip": "現在の内容を要約",
+    "timeRange": {
+      "anytime": "制限なし",
+      "day": "過去1日",
+      "month": "過去1ヶ月",
+      "week": "過去1週間",
+      "year": "過去1年"
+    },
     "viewMoreResults": "さらに {{results}} 件の結果を見る"
   }
 }
diff --git a/locales/ko-KR/tool.json b/locales/ko-KR/tool.json
index 510c3ef871f38..3bb5d4a539d9e 100644
--- a/locales/ko-KR/tool.json
+++ b/locales/ko-KR/tool.json
@@ -21,8 +21,16 @@
     },
     "searchEngine": "검색 엔진：",
     "searchResult": "검색 수：",
+    "searchTimeRange": "시간 범위:",
     "summary": "요약",
     "summaryTooltip": "현재 내용 요약",
+    "timeRange": {
+      "anytime": "제한 없음",
+      "day": "지난 하루",
+      "month": "지난 한 달",
+      "week": "지난 일주일",
+      "year": "지난 일년"
+    },
     "viewMoreResults": "{{results}}개의 결과 더 보기"
   }
 }
diff --git a/locales/nl-NL/tool.json b/locales/nl-NL/tool.json
index 996bef4d730b9..882082fd66097 100644
--- a/locales/nl-NL/tool.json
+++ b/locales/nl-NL/tool.json
@@ -21,8 +21,16 @@
     },
     "searchEngine": "Zoekmachine:",
     "searchResult": "Aantal zoekresultaten:",
+    "searchTimeRange": "Tijdspanne:",
     "summary": "Samenvatting",
     "summaryTooltip": "Samenvatting van de huidige inhoud",
+    "timeRange": {
+      "anytime": "Geen tijdslimiet",
+      "day": "Afgelopen dag",
+      "month": "Afgelopen maand",
+      "week": "Afgelopen week",
+      "year": "Afgelopen jaar"
+    },
     "viewMoreResults": "Bekijk meer {{results}} resultaten"
   }
 }
diff --git a/locales/pl-PL/tool.json b/locales/pl-PL/tool.json
index 1a55136eb419e..244a551546ad8 100644
--- a/locales/pl-PL/tool.json
+++ b/locales/pl-PL/tool.json
@@ -21,8 +21,16 @@
     },
     "searchEngine": "Silnik wyszukiwania:",
     "searchResult": "Liczba wyników:",
+    "searchTimeRange": "Zakres czasu:",
     "summary": "Podsumowanie",
     "summaryTooltip": "Podsumuj bieżącą treść",
+    "timeRange": {
+      "anytime": "Bez ograniczeń czasowych",
+      "day": "Ostatni dzień",
+      "month": "Ostatni miesiąc",
+      "week": "Ostatni tydzień",
+      "year": "Ostatni rok"
+    },
     "viewMoreResults": "Zobacz więcej {{results}} wyników"
   }
 }
diff --git a/locales/pt-BR/tool.json b/locales/pt-BR/tool.json
index 90d0599209a3e..3e5ad5200a7cd 100644
--- a/locales/pt-BR/tool.json
+++ b/locales/pt-BR/tool.json
@@ -21,8 +21,16 @@
     },
     "searchEngine": "Motor de busca:",
     "searchResult": "Número de pesquisas:",
+    "searchTimeRange": "Intervalo de tempo:",
     "summary": "Resumo",
     "summaryTooltip": "Resumir o conteúdo atual",
+    "timeRange": {
+      "anytime": "Qualquer momento",
+      "day": "Últimas 24 horas",
+      "month": "Último mês",
+      "week": "Última semana",
+      "year": "Último ano"
+    },
     "viewMoreResults": "Ver mais {{results}} resultados"
   }
 }
diff --git a/locales/ru-RU/tool.json b/locales/ru-RU/tool.json
index aa187561e7714..97f61b632793e 100644
--- a/locales/ru-RU/tool.json
+++ b/locales/ru-RU/tool.json
@@ -21,8 +21,16 @@
     },
     "searchEngine": "Поисковая система:",
     "searchResult": "Количество результатов:",
+    "searchTimeRange": "Временной диапазон:",
     "summary": "Резюме",
     "summaryTooltip": "Суммировать текущее содержимое",
+    "timeRange": {
+      "anytime": "Без ограничения по времени",
+      "day": "За последний день",
+      "month": "За последний месяц",
+      "week": "За последнюю неделю",
+      "year": "За последний год"
+    },
     "viewMoreResults": "Посмотреть еще {{results}} результатов"
   }
 }
diff --git a/locales/tr-TR/tool.json b/locales/tr-TR/tool.json
index 43d1e4fbb2b1f..fdec2c0a14955 100644
--- a/locales/tr-TR/tool.json
+++ b/locales/tr-TR/tool.json
@@ -21,8 +21,16 @@
     },
     "searchEngine": "Arama motoru:",
     "searchResult": "Arama sayısı:",
+    "searchTimeRange": "Zaman Aralığı:",
     "summary": "Özet",
     "summaryTooltip": "Mevcut içeriği özetle",
+    "timeRange": {
+      "anytime": "Herhangi bir zaman",
+      "day": "Son bir gün",
+      "month": "Son bir ay",
+      "week": "Son bir hafta",
+      "year": "Son bir yıl"
+    },
     "viewMoreResults": "Daha fazla {{results}} sonuç görüntüle"
   }
 }
diff --git a/locales/vi-VN/tool.json b/locales/vi-VN/tool.json
index 88f63c51e81fd..10f55ffa6be59 100644
--- a/locales/vi-VN/tool.json
+++ b/locales/vi-VN/tool.json
@@ -21,8 +21,16 @@
     },
     "searchEngine": "Công cụ tìm kiếm:",
     "searchResult": "Số lượng tìm kiếm:",
+    "searchTimeRange": "Khoảng thời gian:",
     "summary": "Tóm tắt",
     "summaryTooltip": "Tóm tắt nội dung hiện tại",
+    "timeRange": {
+      "anytime": "Không giới hạn thời gian",
+      "day": "Ngày qua",
+      "month": "Tháng qua",
+      "week": "Tuần qua",
+      "year": "Năm qua"
+    },
     "viewMoreResults": "Xem thêm {{results}} kết quả"
   }
 }
diff --git a/locales/zh-CN/tool.json b/locales/zh-CN/tool.json
index ae58008fda2e8..85e9180153637 100644
--- a/locales/zh-CN/tool.json
+++ b/locales/zh-CN/tool.json
@@ -21,8 +21,16 @@
     },
     "searchEngine": "搜索引擎：",
     "searchResult": "搜索数量：",
+    "searchTimeRange": "时间范围：",
     "summary": "总结",
     "summaryTooltip": "总结当前内容",
+    "timeRange": {
+      "anytime": "不限时间",
+      "day": "过去一天",
+      "month": "过去一月",
+      "week": "过去一周",
+      "year": "过去一年"
+    },
     "viewMoreResults": "查看更多 {{results}} 个结果"
   }
 }
diff --git a/locales/zh-TW/tool.json b/locales/zh-TW/tool.json
index 8328746021e52..1975f26dd7079 100644
--- a/locales/zh-TW/tool.json
+++ b/locales/zh-TW/tool.json
@@ -21,8 +21,16 @@
     },
     "searchEngine": "搜尋引擎：",
     "searchResult": "搜尋數量：",
+    "searchTimeRange": "時間範圍：",
     "summary": "總結",
     "summaryTooltip": "總結當前內容",
+    "timeRange": {
+      "anytime": "不限時間",
+      "day": "過去一天",
+      "month": "過去一個月",
+      "week": "過去一週",
+      "year": "過去一年"
+    },
     "viewMoreResults": "查看更多 {{results}} 個結果"
   }
 }
diff --git a/src/locales/default/tool.ts b/src/locales/default/tool.ts
index 5bf24f48730c2..cbbe7424bf93d 100644
--- a/src/locales/default/tool.ts
+++ b/src/locales/default/tool.ts
@@ -21,8 +21,16 @@ export default {
     },
     searchEngine: '搜索引擎：',
     searchResult: '搜索数量：',
+    searchTimeRange: '时间范围：',
     summary: '总结',
     summaryTooltip: '总结当前内容',
+    timeRange: {
+      anytime: '不限时间',
+      day: '过去一天',
+      month: '过去一月',
+      week: '过去一周',
+      year: '过去一年',
+    },
     viewMoreResults: '查看更多 {{results}} 个结果',
   },
 };
diff --git a/src/server/modules/SearXNG.ts b/src/server/modules/SearXNG.ts
index d2f5872bbce07..b9266538c1c09 100644
--- a/src/server/modules/SearXNG.ts
+++ b/src/server/modules/SearXNG.ts
@@ -10,12 +10,13 @@ export class SearXNGClient {
     this.baseUrl = baseUrl;
   }
 
-  async search(query: string, engines?: string[]): Promise<SearchResponse> {
+  async search(query: string, engines?: string[], time_range?: string): Promise<SearchResponse> {
     try {
       const searchParams = qs.stringify({
         engines: engines?.join(','),
         format: 'json',
         q: query,
+        ...( time_range !== 'anytime' ? { time_range: time_range } : {} )
       });
 
       const response = await fetch(urlJoin(this.baseUrl, `/search?${searchParams}`));
diff --git a/src/server/routers/tools/search.ts b/src/server/routers/tools/search.ts
index 96bb14f205af6..a5912cd99c10b 100644
--- a/src/server/routers/tools/search.ts
+++ b/src/server/routers/tools/search.ts
@@ -38,6 +38,7 @@ export const searchRouter = router({
       z.object({
         query: z.string(),
         searchEngine: z.array(z.string()).optional(),
+        searchTimeRange: z.string().optional(),
       }),
     )
     .query(async ({ input }) => {
@@ -48,7 +49,7 @@ export const searchRouter = router({
       const client = new SearXNGClient(toolsEnv.SEARXNG_URL);
 
       try {
-        return await client.search(input.query, input.searchEngine);
+        return await client.search(input.query, input.searchEngine, input.searchTimeRange);
       } catch (e) {
         console.error(e);
 
diff --git a/src/services/search.ts b/src/services/search.ts
index 382979b452d31..50c404ca4f14d 100644
--- a/src/services/search.ts
+++ b/src/services/search.ts
@@ -1,8 +1,8 @@
 import { toolsClient } from '@/libs/trpc/client';
 
 class SearchService {
-  search(query: string, searchEngine?: string[]) {
-    return toolsClient.search.query.query({ query, searchEngine });
+  search(query: string, searchEngine?: string[], searchTimeRange?: string) {
+    return toolsClient.search.query.query({ query, searchEngine, searchTimeRange });
   }
 
   crawlPage(url: string) {
diff --git a/src/store/chat/slices/builtinTool/actions/searXNG.test.ts b/src/store/chat/slices/builtinTool/actions/searXNG.test.ts
index d95215d26dd4a..22ae559fdd6fd 100644
--- a/src/store/chat/slices/builtinTool/actions/searXNG.test.ts
+++ b/src/store/chat/slices/builtinTool/actions/searXNG.test.ts
@@ -88,7 +88,7 @@ describe('searXNG actions', () => {
         },
       ];
 
-      expect(searchService.search).toHaveBeenCalledWith('test query', ['google']);
+      expect(searchService.search).toHaveBeenCalledWith('test query', ['google'], undefined);
       expect(result.current.searchLoading[messageId]).toBe(false);
       expect(result.current.internal_updateMessageContent).toHaveBeenCalledWith(
         messageId,
@@ -143,6 +143,7 @@ describe('searXNG actions', () => {
       const query: SearchQuery = {
         query: 'test query',
         searchEngines: ['custom-engine'],
+        searchTimeRange: 'year',
       };
 
       await act(async () => {
@@ -150,7 +151,7 @@ describe('searXNG actions', () => {
       });
 
       expect(searchService.search).toHaveBeenCalledTimes(2);
-      expect(searchService.search).toHaveBeenNthCalledWith(1, 'test query', ['custom-engine']);
+      expect(searchService.search).toHaveBeenNthCalledWith(1, 'test query', ['custom-engine'], 'year');
       expect(searchService.search).toHaveBeenNthCalledWith(2, 'test query');
       expect(result.current.updatePluginArguments).toHaveBeenCalledWith(messageId, {
         query: 'test query',
diff --git a/src/store/chat/slices/builtinTool/actions/searXNG.ts b/src/store/chat/slices/builtinTool/actions/searXNG.ts
index 28b85d9d9573c..162ef6bfdf933 100644
--- a/src/store/chat/slices/builtinTool/actions/searXNG.ts
+++ b/src/store/chat/slices/builtinTool/actions/searXNG.ts
@@ -141,12 +141,12 @@ export const searchSlice: StateCreator<
     get().toggleSearchLoading(id, true);
     let data: SearchResponse | undefined;
     try {
-      data = await searchService.search(params.query, params.searchEngines);
+      data = await searchService.search(params.query, params.searchEngines, params.searchTimeRange);
 
       // 如果没有搜索到结果，那么尝试使用默认的搜索引擎再搜一次
-      if (data?.results.length === 0 && params.searchEngines && params.searchEngines?.length > 0) {
+      if (data?.results.length === 0 && (params.searchEngines && params.searchEngines?.length > 0 || params.searchTimeRange)) {
         data = await searchService.search(params.query);
-        get().updatePluginArguments(id, { ...params, searchEngines: undefined });
+        get().updatePluginArguments(id, { ...params, searchEngines: undefined, searchTimeRange: undefined });
       }
 
       await get().updatePluginState(id, data);
diff --git a/src/tools/web-browsing/components/SearchBar.tsx b/src/tools/web-browsing/components/SearchBar.tsx
index 2b630831d207f..0b893ed9f97ff 100644
--- a/src/tools/web-browsing/components/SearchBar.tsx
+++ b/src/tools/web-browsing/components/SearchBar.tsx
@@ -1,5 +1,5 @@
 import { Icon, Tooltip } from '@lobehub/ui';
-import { Button, Checkbox, Input, Select, Space, Typography } from 'antd';
+import { Button, Checkbox, Input, Radio, Select, Space, Typography } from 'antd';
 import { SearchIcon } from 'lucide-react';
 import { ReactNode, memo, useState } from 'react';
 import { useTranslation } from 'react-i18next';
@@ -17,6 +17,7 @@ interface SearchBarProps {
   aiSummary?: boolean;
   defaultEngines?: string[];
   defaultQuery: string;
+  defaultTimeRange?: string;
   messageId: string;
   onSearch?: (searchQuery: SearchQuery) => void;
   searchAddon?: ReactNode;
@@ -26,6 +27,7 @@ interface SearchBarProps {
 const SearchBar = memo<SearchBarProps>(
   ({
     defaultEngines = [],
+    defaultTimeRange,
     aiSummary = true,
     defaultQuery,
     tooltip = true,
@@ -37,11 +39,16 @@ const SearchBar = memo<SearchBarProps>(
     const loading = useChatStore(chatToolSelectors.isSearXNGSearching(messageId));
     const [query, setQuery] = useState(defaultQuery);
     const [engines, setEngines] = useState(defaultEngines);
+    const [time_range, setTimeRange] = useState(defaultTimeRange);
     const isMobile = useIsMobile();
     const [reSearchWithSearXNG] = useChatStore((s) => [s.reSearchWithSearXNG]);
 
     const updateAndSearch = async () => {
-      const data: SearchQuery = { query, searchEngines: engines };
+      const data: SearchQuery = {
+        query,
+        searchEngines: engines,
+        searchTimeRange: time_range,
+      };
       onSearch?.(data);
       await reSearchWithSearXNG(messageId, data, { aiSummary });
     };
@@ -127,6 +134,24 @@ const SearchBar = memo<SearchBarProps>(
             />
           </Flexbox>
         )}
+
+        <Flexbox align={'center'} gap={16} horizontal wrap={'wrap'}>
+          <Typography.Text type={'secondary'}>
+            {t('search.searchTimeRange')}
+          </Typography.Text>
+          <Radio.Group
+            onChange={(e) => setTimeRange(e.target.value)}
+            optionType="button"
+            options={[
+              { label: t('search.timeRange.anytime'), value: 'anytime' },
+              { label: t('search.timeRange.day'), value: 'day' },
+              { label: t('search.timeRange.week'), value: 'week' },
+              { label: t('search.timeRange.month'), value: 'month' },
+              { label: t('search.timeRange.year'), value: 'year' },
+            ]}
+            value={time_range}
+          />
+        </Flexbox>
       </Flexbox>
     );
   },
diff --git a/src/tools/web-browsing/index.ts b/src/tools/web-browsing/index.ts
index 7e32a75a70932..75b0affeb9883 100644
--- a/src/tools/web-browsing/index.ts
+++ b/src/tools/web-browsing/index.ts
@@ -19,13 +19,14 @@ export const WebBrowsingManifest: BuiltinToolManifest = {
       parameters: {
         properties: {
           query: {
-            description: 'The search query',
+            description: '(required) [string] The search query.',
             type: 'string',
           },
           searchEngines: {
-            description: 'The search engine you can use:',
+            description: '(optional) [array] Specifies the active search engines or categories.',
             items: {
               enum: [
+                // Engines
                 'google',
                 'bilibili',
                 'bing',
@@ -44,11 +45,33 @@ export const WebBrowsingManifest: BuiltinToolManifest = {
                 'unsplash',
                 'vimeo',
                 'youtube',
+                // Categories
+                'files',
+                'general',
+                'images',
+                'it',
+                'map',
+                'music',
+                'news',
+                'science',
+                'social_media',
+                'videos',
               ],
               type: 'string',
             },
             type: 'array',
           },
+          searchTimeRange: {
+            description: "(optional) [string] Specifies time range of search results.",
+            enum: [
+              'anytime',
+              'day',
+              'week',
+              'month',
+              'year',
+            ],
+            type: 'string'
+          },
         },
         required: ['query'],
         type: 'object',
diff --git a/src/tools/web-browsing/systemRole.ts b/src/tools/web-browsing/systemRole.ts
index 1a960d0476fcd..a7edf9ec35c9d 100644
--- a/src/tools/web-browsing/systemRole.ts
+++ b/src/tools/web-browsing/systemRole.ts
@@ -31,8 +31,27 @@ Choose search engines based on the query type:
 - Images: unsplash, pinterest
 - Entertainment: imdb, reddit
 - For region-specific information, prefer search engines popular in that region
+
+It's a good choice to use the following categories to search across all relevant engines if you need to broaden the range of search or are unsure about which specific engine to use:  
+- General information: general
+- News: news
+- Academic/scientific research: science
+- Technical/IT-related queries: it
+- Multimedia content: images, videos, music
+- Geographic/map-related searches: map
+- File searches: files
+- Social trends and discussions: social_media
 </search_engine_selection>
 
+<search_time_range_selection>
+Only specify a time range if the query is time-sensitive:
+- For the latest updates: day
+- For recent developments: week
+- For ongoing trends or updates: month
+- For long-term insights: year
+- For no time restriction: anytime
+</search_time_range_selection>
+
 <citation_requirements>
 - Always cite sources using markdown footnote format (e.g., [^1])
 - List all referenced URLs at the end of your response
diff --git a/src/types/tool/search.ts b/src/types/tool/search.ts
index 66e7314f2aa66..248d9f444310a 100644
--- a/src/types/tool/search.ts
+++ b/src/types/tool/search.ts
@@ -1,6 +1,7 @@
 export interface SearchQuery {
   query: string;
   searchEngines?: string[];
+  searchTimeRange?: string;
 }
 
 export const SEARCH_SEARXNG_NOT_CONFIG = 'SearXNG is not configured';
@@ -22,12 +23,14 @@ export interface SearchResult {
   engine: string;
   engines: string[];
   iframe_src?: string;
+  img_src?: string;
   parsed_url: string[];
   positions: number[];
   publishedDate?: string | null;
   score: number;
   template: string;
   thumbnail?: string | null;
+  thumbnail_src?: string;
   title: string;
   url: string;
 }
