diff --git a/src/app/(backend)/oidc/consent/route.ts b/src/app/(backend)/oidc/consent/route.ts
index 4166f27932c79..be8e3bfe84611 100644
--- a/src/app/(backend)/oidc/consent/route.ts
+++ b/src/app/(backend)/oidc/consent/route.ts
@@ -122,7 +122,6 @@ export async function POST(request: NextRequest) {
     }
 
     return NextResponse.redirect(finalRedirectUrl, {
-      headers: request.headers,
       status: 303,
     });
   } catch (error) {
