diff --git a/src/libs/agent-runtime/stepfun/index.ts b/src/libs/agent-runtime/stepfun/index.ts
index 945a5b53241fa..0fcfc68e888c4 100644
--- a/src/libs/agent-runtime/stepfun/index.ts
+++ b/src/libs/agent-runtime/stepfun/index.ts
@@ -46,9 +46,14 @@ export const LobeStepfunAI = LobeOpenAICompatibleFactory({
 
     const visionKeywords = [
       'step-1o-',
+      'step-r1-v-',
       'step-1v-',
     ];
 
+    const reasoningKeywords = [
+      'step-r1-',
+    ];
+
     const modelsPage = await client.models.list() as any;
     const modelList: StepfunModelCard[] = modelsPage.data;
 
@@ -66,7 +71,8 @@ export const LobeStepfunAI = LobeOpenAICompatibleFactory({
             || false,
           id: model.id,
           reasoning:
-            knownModel?.abilities?.reasoning
+            reasoningKeywords.some(keyword => model.id.toLowerCase().includes(keyword))
+            || knownModel?.abilities?.reasoning
             || false,
           vision:
             visionKeywords.some(keyword => model.id.toLowerCase().includes(keyword))
diff --git a/src/libs/agent-runtime/zhipu/index.ts b/src/libs/agent-runtime/zhipu/index.ts
index 0995d4f0d9c1c..a2ac8e57da6f5 100644
--- a/src/libs/agent-runtime/zhipu/index.ts
+++ b/src/libs/agent-runtime/zhipu/index.ts
@@ -50,23 +50,30 @@ export const LobeZhipuAI = LobeOpenAICompatibleFactory({
       } as any;
     },
   },
-  constructorOptions: {
-    defaultHeaders: {
-      'Bigmodel-Organization': 'lobehub',
-      'Bigmodel-project': 'lobechat',
-    },
-  },
   debug: {
     chatCompletion: () => process.env.DEBUG_ZHIPU_CHAT_COMPLETION === '1',
   },
   models: async ({ client }) => {
     const { LOBE_DEFAULT_MODEL_LIST } = await import('@/config/aiModels');
 
+    const reasoningKeywords = [
+      'glm-zero',
+      'glm-z1',
+    ];
+
     // ref: https://open.bigmodel.cn/console/modelcenter/square
-    client.baseURL = 'https://open.bigmodel.cn/api/fine-tuning/model_center/list?pageSize=100&pageNum=1';
+    const url = 'https://open.bigmodel.cn/api/fine-tuning/model_center/list?pageSize=100&pageNum=1';
+    const response = await fetch(url, {
+      headers: {
+        'Authorization': `Bearer ${client.apiKey}`,
+        'Bigmodel-Organization': 'lobehub',
+        'Bigmodel-Project': 'lobechat',
+      },
+      method: 'GET',
+    });
+    const json = await response.json();
 
-    const modelsPage = await client.models.list() as any;
-    const modelList: ZhipuModelCard[] = modelsPage.body.rows;
+    const modelList: ZhipuModelCard[] = json.rows;
 
     return modelList
       .map((model) => {
@@ -83,7 +90,7 @@ export const LobeZhipuAI = LobeOpenAICompatibleFactory({
             || false,
           id: model.modelCode,
           reasoning:
-            model.modelCode.toLowerCase().includes('glm-zero-preview')
+            reasoningKeywords.some(keyword => model.modelCode.toLowerCase().includes(keyword))
             || knownModel?.abilities?.reasoning
             || false,
           vision:
