diff --git a/src/config/aiModels/moonshot.ts b/src/config/aiModels/moonshot.ts
index 6fb1b4582caff..01d3f86bf3e33 100644
--- a/src/config/aiModels/moonshot.ts
+++ b/src/config/aiModels/moonshot.ts
@@ -2,6 +2,29 @@ import { AIChatModelCard } from '@/types/aiModel';
 
 // https://platform.moonshot.cn/docs/pricing/chat
 const moonshotChatModels: AIChatModelCard[] = [
+  {
+    abilities: {
+      functionCall: true,
+      search: true,
+    },
+    contextWindowTokens: 131_072,
+    description:
+      'kimi-k2 是一款具备超强代码和 Agent 能力的 MoE 架构基础模型，总参数 1T，激活参数 32B。在通用知识推理、编程、数学、Agent 等主要类别的基准性能测试中，K2 模型的性能超过其他主流开源模型。',
+    displayName: 'Kimi K2',
+    enabled: true,
+    id: 'kimi-k2-0711-preview',
+    pricing: {
+      cachedInput: 1,
+      currency: 'CNY',
+      input: 4,
+      output: 16,
+    },
+    releasedAt: '2025-07-11',
+    settings: {
+      searchImpl: 'params',
+    },
+    type: 'chat',
+  },
   {
     abilities: {
       functionCall: true,
@@ -17,7 +40,7 @@ const moonshotChatModels: AIChatModelCard[] = [
     pricing: {
       cachedInput: 1,
       currency: 'CNY',
-      input: 10,
+      input: 10, // 128k 上下文时
       output: 30,
     },
     releasedAt: '2025-02-17',
@@ -56,7 +79,7 @@ const moonshotChatModels: AIChatModelCard[] = [
     id: 'moonshot-v1-auto',
     pricing: {
       currency: 'CNY',
-      input: 10,
+      input: 10, // 128k 上下文时
       output: 30,
     },
     settings: {
