diff --git a/src/services/chat.ts b/src/services/chat.ts
index f57f05db2466..02ef60f45b76 100644
--- a/src/services/chat.ts
+++ b/src/services/chat.ts
@@ -106,18 +106,12 @@ export function initializeWithClientStore(provider: string, payload: any) {
       };
       break;
     }
-    case ModelProvider.ZhiPu: {
-      break;
-    }
     case ModelProvider.Google: {
       providerOptions = {
         baseURL: providerAuthPayload?.endpoint,
       };
       break;
     }
-    case ModelProvider.Moonshot: {
-      break;
-    }
     case ModelProvider.Bedrock: {
       if (providerAuthPayload?.apiKey) {
         providerOptions = {
@@ -142,20 +136,12 @@ export function initializeWithClientStore(provider: string, payload: any) {
       };
       break;
     }
-    case ModelProvider.Qwen: {
-      break;
-    }
-
     case ModelProvider.Anthropic: {
       providerOptions = {
         baseURL: providerAuthPayload?.endpoint,
       };
       break;
     }
-
-    case ModelProvider.Mistral: {
-      break;
-    }
     case ModelProvider.Groq: {
       providerOptions = {
         apikey: providerAuthPayload?.apiKey,
@@ -163,18 +149,6 @@ export function initializeWithClientStore(provider: string, payload: any) {
       };
       break;
     }
-    case ModelProvider.DeepSeek: {
-      break;
-    }
-    case ModelProvider.OpenRouter: {
-      break;
-    }
-    case ModelProvider.TogetherAI: {
-      break;
-    }
-    case ModelProvider.ZeroOne: {
-      break;
-    }
     case ModelProvider.Cloudflare: {
       providerOptions = {
         apikey: providerAuthPayload?.apiKey,
