diff --git a/src/config/aiModels/volcengine.ts b/src/config/aiModels/volcengine.ts
index e3934804d885d..13224f87ac30e 100644
--- a/src/config/aiModels/volcengine.ts
+++ b/src/config/aiModels/volcengine.ts
@@ -33,7 +33,7 @@ const doubaoChatModels: AIChatModelCard[] = [
       vision: true,
     },
     config: {
-      deploymentName: 'doubao-1-5-thinking-pro-vision-250415',
+      deploymentName: 'doubao-1-5-thinking-pro-m-250415',
     },
     contextWindowTokens: 128_000,
     description:
