diff --git a/src/store/chat/slices/builtinTool/actions/searXNG.ts b/src/store/chat/slices/builtinTool/actions/searXNG.ts
index f77a0a4828d88..5e5795dc1e243 100644
--- a/src/store/chat/slices/builtinTool/actions/searXNG.ts
+++ b/src/store/chat/slices/builtinTool/actions/searXNG.ts
@@ -191,6 +191,7 @@ export const searchSlice: StateCreator<
       content: item.content,
       title: item.title,
       url: item.url,
+      ...(item.img_src && { img_src: item.img_src }),
     }));
 
     await get().internal_updateMessageContent(id, JSON.stringify(searchContent));
diff --git a/src/types/tool/search.ts b/src/types/tool/search.ts
index ddcff3b461d8e..2e14238890e9f 100644
--- a/src/types/tool/search.ts
+++ b/src/types/tool/search.ts
@@ -26,18 +26,21 @@ export interface SearchResult {
   engine: string;
   engines: string[];
   iframe_src?: string;
+  img_src?: string;
   parsed_url: string[];
   positions: number[];
   publishedDate?: string | null;
   score: number;
   template: string;
   thumbnail?: string | null;
+  thumbnail_src?: string | null;
   title: string;
   url: string;
 }
 
 export interface SearchContent {
   content?: string;
+  img_src?: string;
   title: string;
   url: string;
 }
