diff --git a/src/types/tool/search.ts b/src/types/tool/search.ts
index ddcff3b461d8e..3310f1adc60fd 100644
--- a/src/types/tool/search.ts
+++ b/src/types/tool/search.ts
@@ -26,11 +26,13 @@ export interface SearchResult {
   engine: string;
   engines: string[];
   iframe_src?: string;
+  img_src?: string;
   parsed_url: string[];
   positions: number[];
   publishedDate?: string | null;
   score: number;
   template: string;
+  thumbnail_src?: string | null;
   thumbnail?: string | null;
   title: string;
   url: string;
@@ -38,6 +40,7 @@ export interface SearchResult {
 
 export interface SearchContent {
   content?: string;
+  img_src?: string;
   title: string;
   url: string;
 }
