diff --git a/src/components/ModelIcon/index.tsx b/src/components/ModelIcon/index.tsx
index 15be9c148d91..2f4e4b75c631 100644
--- a/src/components/ModelIcon/index.tsx
+++ b/src/components/ModelIcon/index.tsx
@@ -4,6 +4,7 @@ import {
   Ai21,
   Ai360,
   Aws,
+  // Aya, // TODO: wait for Aya ready
   Azure,
   Baichuan,
   ByteDance,
@@ -62,11 +63,12 @@ const ModelIcon = memo<ModelProviderIconProps>(({ model: originModel, size = 12
   if (model.includes('moonshot')) return <Moonshot.Avatar size={size} />;
   if (model.includes('qwen')) return <Tongyi.Avatar background={Tongyi.colorPrimary} size={size} />;
   if (model.includes('minmax') || model.includes('abab')) return <Minimax.Avatar size={size} />;
-  if (model.includes('mistral') || model.includes('mixtral') || model.includes('codestral')) return <Mistral.Avatar size={size} />;
+  if (model.includes('mistral') || model.includes('mixtral') || model.includes('codestral') || model.includes('mathstral')) return <Mistral.Avatar size={size} />;
   if (model.includes('pplx') || model.includes('sonar')) return <Perplexity.Avatar size={size} />;
   if (model.includes('yi-')) return <Yi.Avatar size={size} />;
   if (model.startsWith('openrouter')) return <OpenRouter.Avatar size={size} />; // only for Cinematika and Auto
   if (model.startsWith('openchat')) return <OpenChat.Avatar size={size} />;
+  // if (model.includes('aya')) return <Aya.Avatar size={size} />; // TODO: wait for Aya ready
   if (model.includes('command')) return <Cohere.Avatar size={size} />;
   if (model.includes('dbrx')) return <Dbrx.Avatar size={size} />;
   if (model.includes('step')) return <Stepfun.Avatar size={size} />;
@@ -96,8 +98,7 @@ const ModelIcon = memo<ModelProviderIconProps>(({ model: originModel, size = 12
   )
     return <Stability.Avatar size={size} />;
 
-  if (model.includes('wizardlm')) return <Azure.Avatar size={size} />;
-  if (model.includes('phi3') || model.includes('phi-3')) return <Azure.Avatar size={size} />;
+  if (model.includes('phi3') || model.includes('phi-3') || model.includes('wizardlm')) return <Azure.Avatar size={size} />;
   if (model.includes('firefly')) return <Adobe.Avatar size={size} />;
   if (model.includes('jamba') || model.includes('j2-')) return <Ai21.Avatar size={size} />;
 });
diff --git a/src/components/ModelTag/ModelIcon.tsx b/src/components/ModelTag/ModelIcon.tsx
index f6bc0e949799..8d713ca2e71b 100644
--- a/src/components/ModelTag/ModelIcon.tsx
+++ b/src/components/ModelTag/ModelIcon.tsx
@@ -4,6 +4,7 @@ import {
   Ai360,
   AiMass,
   Aws,
+  // Aya, // TODO: wait for Aya ready
   Azure,
   Baichuan,
   ByteDance,
@@ -62,11 +63,12 @@ const ModelIcon = memo<ModelIconProps>(({ model: originModel, size = 12 }) => {
   if (model.includes('qwen')) return <Tongyi size={size} />;
   if (model.includes('minmax')) return <Minimax size={size} />;
   if (model.includes('abab')) return <Minimax size={size} />;
-  if (model.includes('mistral') || model.includes('mixtral') || model.includes('codestral')) return <Mistral size={size} />;
+  if (model.includes('mistral') || model.includes('mixtral') || model.includes('codestral') || model.includes('mathstral')) return <Mistral size={size} />;
   if (model.includes('pplx') || model.includes('sonar')) return <Perplexity size={size} />;
   if (model.includes('yi-')) return <Yi size={size} />;
   if (model.startsWith('openrouter')) return <OpenRouter size={size} />; // only for Cinematika and Auto
   if (model.startsWith('openchat')) return <OpenChat size={size} />;
+  // if (model.includes('aya')) return <Aya.Avatar size={size} />; // TODO: wait for Aya ready
   if (model.includes('command')) return <Cohere size={size} />;
   if (model.includes('dbrx')) return <Dbrx size={size} />;
   if (model.includes('step')) return <Stepfun size={size} />;
@@ -95,8 +97,7 @@ const ModelIcon = memo<ModelIconProps>(({ model: originModel, size = 12 }) => {
   )
     return <Stability size={size} />;
 
-  if (model.includes('wizardlm')) return <Azure size={size} />;
-  if (model.includes('phi3') || model.includes('phi-3')) return <Azure size={size} />;
+  if (model.includes('phi3') || model.includes('phi-3') || model.includes('wizardlm')) return <Azure.Avatar size={size} />;
   if (model.includes('firefly')) return <AdobeFirefly size={size} />;
   if (model.includes('jamba') || model.includes('j2-')) return <Ai21 size={size} />;
 });
diff --git a/src/config/modelProviders/ollama.ts b/src/config/modelProviders/ollama.ts
index c19d23f4919b..c1da02570d71 100644
--- a/src/config/modelProviders/ollama.ts
+++ b/src/config/modelProviders/ollama.ts
@@ -4,200 +4,204 @@ import { ModelProviderCard } from '@/types/llm';
 const Ollama: ModelProviderCard = {
   chatModels: [
     {
-      displayName: 'Llama3 8B',
+      displayName: 'Llama3.1 8B',
       enabled: true,
-      id: 'llama3',
-      tokens: 8000, // https://huggingface.co/blog/zh/llama3#llama-3-的新进展
-    },
-    {
-      displayName: 'Llama3 70B',
-      id: 'llama3:70b',
-      tokens: 8000,
+      id: 'llama3.1',
+      tokens: 128_000,
     },
     {
-      displayName: 'Phi-3 3.8B',
-      enabled: true,
-      id: 'phi3',
+      displayName: 'Llama3.1 70B',
+      id: 'llama3.1:70b',
       tokens: 128_000,
     },
     {
-      displayName: 'Phi-3 14B',
-      id: 'phi3:14b',
+      displayName: 'Llama3.1 405B',
+      id: 'llama3.1:405b',
       tokens: 128_000,
     },
     {
-      displayName: 'Aya 23 8B',
+      displayName: 'Code Llama 7B',
       enabled: true,
-      id: 'aya',
-      tokens: 8192, // https://cohere.com/research/papers/aya-command-23-8b-and-35b-technical-report-2024-05-23
-    },
-    {
-      displayName: 'Aya 23 35B',
-      id: 'aya:35b',
-      tokens: 8192,
+      id: 'codellama',
+      tokens: 16_384,
     },
     {
-      displayName: 'Qwen2 7B',
-      enabled: true,
-      id: 'qwen2',
-      tokens: 128_000,
+      displayName: 'Code Llama 13B',
+      id: 'codellama:13b',
+      tokens: 16_384,
     },
     {
-      displayName: 'Qwen2 72B',
-      id: 'qwen2:72b',
-      tokens: 128_000,
+      displayName: 'Code Llama 34B',
+      id: 'codellama:34b',
+      tokens: 16_384,
     },
     {
-      displayName: 'Command R 35B',
-      enabled: true,
-      id: 'command-r',
-      tokens: 131_072, // https://huggingface.co/CohereForAI/c4ai-command-r-v01/blob/main/config.json
+      displayName: 'Code Llama 70B',
+      id: 'codellama:70b',
+      tokens: 16_384,
     },
     {
-      displayName: 'Command R+ 104B (Q2_K)',
-      id: 'command-r-plus:104b-q2_K',
-      tokens: 131_072, // https://huggingface.co/CohereForAI/c4ai-command-r-plus/blob/main/config.json
+      displayName: 'Gemma2 2B',
+      id: 'gemma2:2b',
+      tokens: 8192,
     },
     {
       displayName: 'Gemma2 9B',
       enabled: true,
       id: 'gemma2',
-      tokens: 8192, // https://huggingface.co/blog/zh/gemma2
+      tokens: 8192,
     },
     {
-      displayName: 'Gemma2 27B',
-      id: 'gemma2:27b',
+      displayName: 'CodeGemma 2B',
+      enabled: true,
+      id: 'codegemma:2b',
       tokens: 8192,
     },
     {
-      displayName: 'Gemma 7B',
-      id: 'gemma',
-      tokens: 8192, // https://huggingface.co/google/gemma-7b-it/discussions/73#65e9678c0cda621164a95bad
+      displayName: 'CodeGemma 7B',
+      enabled: true,
+      id: 'codegemma',
+      tokens: 8192,
     },
     {
-      displayName: 'Gemma 2B',
-      id: 'gemma:2b',
+      displayName: 'Gemma2 27B',
+      id: 'gemma2:27b',
       tokens: 8192,
     },
     {
-      displayName: 'Deepseek V2 16B',
+      displayName: 'Phi-3 3.8B',
       enabled: true,
-      id: 'deepseek-v2',
-      tokens: 32_000,
+      id: 'phi3',
+      tokens: 128_000,
     },
     {
-      displayName: 'Deepseek V2 236B',
-      id: 'deepseek-v2:236b',
+      displayName: 'Phi-3 14B',
+      id: 'phi3:14b',
       tokens: 128_000,
     },
     {
-      displayName: 'Deepseek Coder V2 16B', // https://huggingface.co/deepseek-ai/DeepSeek-Coder-V2-Lite-Instruct
+      displayName: 'WizardLM 2 7B',
       enabled: true,
-      id: 'deepseek-coder-v2',
-      tokens: 128_000,
+      id: 'wizardlm2',
+      tokens: 65_536,
     },
     {
-      displayName: 'Deepseek Coder V2 236B', // https://huggingface.co/deepseek-ai/DeepSeek-Coder-V2-Lite-Instruct
-      id: 'deepseek-coder-v2:236b',
-      tokens: 128_000,
+      displayName: 'WizardLM 2 8x22B',
+      id: 'wizardlm2:8x22b',
+      tokens: 65_536,
     },
     {
-      displayName: 'Llama2 Chat 13B',
-      id: 'llama2:13b',
-      tokens: 4096, // https://llama.meta.com/llama2/
+      displayName: 'MathΣtral 7B',
+      enabled: true,
+      id: 'mathstral',
+      tokens: 32_000,
     },
     {
-      displayName: 'Llama2 Chat 7B',
-      id: 'llama2',
-      tokens: 4096,
+      displayName: 'Mistral 7B',
+      enabled: true,
+      id: 'mistral',
+      tokens: 32_768,
     },
     {
-      displayName: 'Llama2 Chat 70B',
-      id: 'llama2:70b',
-      tokens: 4096,
+      displayName: 'Mixtral 8x7B',
+      enabled: true,
+      id: 'mixtral',
+      tokens: 32_768,
     },
     {
-      displayName: 'Llama2 CN 13B',
-      id: 'llama2-chinese:13b',
-      tokens: 4096,
+      displayName: 'Mixtral 8x22B',
+      id: 'mixtral:8x22b',
+      tokens: 65_536,
     },
     {
-      displayName: 'Llama2 CN 7B',
-      id: 'llama2-chinese',
-      tokens: 4096,
+      displayName: 'Mixtral Nemo 12B',
+      enabled: true,
+      id: 'mistral-nemo',
+      tokens: 128_000,
     },
     {
-      displayName: 'WizardLM 2 7B',
+      displayName: 'Codestral 22B',
       enabled: true,
-      id: 'wizardlm2',
-      tokens: 65_536,
+      id: 'codestral',
+      tokens: 32_000,
     },
     {
-      displayName: 'WizardLM 2 8x22B',
-      id: 'wizardlm2:8x22b',
-      tokens: 65_536,
+      displayName: 'Aya 23 8B',
+      enabled: true,
+      id: 'aya',
+      tokens: 8192,
     },
     {
-      displayName: 'Code Llama 7B',
-      id: 'codellama',
-      tokens: 16_384, // https://huggingface.co/codellama/CodeLlama-7b-hf/blob/main/config.json
+      displayName: 'Aya 23 35B',
+      id: 'aya:35b',
+      tokens: 8192,
     },
     {
-      displayName: 'Code Llama 34B',
-      id: 'codellama:34b',
-      tokens: 16_384,
+      displayName: 'Command R 35B',
+      enabled: true,
+      id: 'command-r',
+      tokens: 131_072,
     },
     {
-      displayName: 'Code Llama 70B',
-      id: 'codellama:70b',
-      tokens: 16_384,
+      displayName: 'Command R+ 104B',
+      enabled: true,
+      id: 'command-r-plus',
+      tokens: 131_072,
     },
     {
-      displayName: 'Code Llama 7B (Python)',
-      id: 'codellama:python',
-      tokens: 16_384,
+      displayName: 'DeepSeek V2 16B',
+      enabled: true,
+      id: 'deepseek-v2',
+      tokens: 32_000,
     },
     {
-      displayName: 'MathΣtral',
-      enabled: true,
-      id: 'mathstral',
-      tokens: 32_000, // https://huggingface.co/mistralai/mathstral-7B-v0.1
+      displayName: 'DeepSeek V2 236B',
+      id: 'deepseek-v2:236b',
+      tokens: 128_000,
     },
     {
-      displayName: 'Mixtral 8x7B',
+      displayName: 'DeepSeek Coder V2 16B',
       enabled: true,
-      id: 'mixtral',
-      tokens: 32_768,
+      id: 'deepseek-coder-v2',
+      tokens: 128_000,
     },
     {
-      displayName: 'Mixtral 8x22B',
-      id: 'mixtral:8x22b',
-      tokens: 65_536, // https://huggingface.co/mistralai/Mixtral-8x22B-v0.1/blob/main/config.json
+      displayName: 'DeepSeek Coder V2 236B',
+      id: 'deepseek-coder-v2:236b',
+      tokens: 128_000,
     },
     {
-      displayName: 'Qwen Chat 4B',
-      id: 'qwen',
-      tokens: 32_768,
+      displayName: 'Qwen2 0.5B',
+      id: 'qwen2:0.5b',
+      tokens: 128_000,
     },
     {
-      displayName: 'Qwen Chat 7B',
-      id: 'qwen:7b',
-      tokens: 32_768,
+      displayName: 'Qwen2 1.5B',
+      id: 'qwen2:1.5b',
+      tokens: 128_000,
     },
     {
-      displayName: 'Qwen Chat 14B',
-      id: 'qwen:14b',
-      tokens: 32_768,
+      displayName: 'Qwen2 7B',
+      enabled: true,
+      id: 'qwen2',
+      tokens: 128_000,
     },
     {
-      displayName: 'Qwen Chat 72B',
-      id: 'qwen:72b',
+      displayName: 'Qwen2 72B',
+      id: 'qwen2:72b',
+      tokens: 128_000,
+    },
+    {
+      displayName: 'CodeQwen1.5 7B',
+      enabled: true,
+      id: 'codeqwen',
       tokens: 32_768,
     },
     {
       displayName: 'LLaVA 7B',
+      enabled: true,
       id: 'llava',
-      tokens: 4096, // https://huggingface.co/llava-hf/llava-1.5-7b-hf/blob/main/config.json
+      tokens: 4096,
       vision: true,
     },
     {
