diff --git a/src/tools/web-browsing/components/CategoryAvatar.tsx b/src/tools/web-browsing/components/CategoryAvatar.tsx
index 7db845e9e9b91..b99a40d47a327 100644
--- a/src/tools/web-browsing/components/CategoryAvatar.tsx
+++ b/src/tools/web-browsing/components/CategoryAvatar.tsx
@@ -6,29 +6,21 @@ import { CATEGORY_ICON_MAP } from '../const';
 
 interface CategoryAvatarProps {
   category: string;
-  size?: number;
 }
 
-export const CategoryAvatar = memo<CategoryAvatarProps>(
-  ({ category, size = 26 }) => {
-    const Icon = CATEGORY_ICON_MAP[category];
-    const theme = useTheme();
+export const CategoryAvatar = memo<CategoryAvatarProps>(({ category }) => {
+  const theme = useTheme();
+  const IconComponent = CATEGORY_ICON_MAP[category];
 
-    return (
-      <Avatar
-        alt={category}
-        style={{
-          backgroundColor: 'transparent',
-          color: theme.colorTextSecondary,
-          display: 'flex',
-          height: size,
-          width: size,
-        }}
-      >
-        <Icon 
-          size={size} 
-        />
-      </Avatar>
-    );
-  },
-);
+  return (
+    <Avatar
+      alt={category}
+      icon={<IconComponent size={16} />}
+      style={{
+        backgroundColor: 'transparent',
+        color: theme.colorTextSecondary,
+        fontSize: 16,
+      }}
+    />
+  );
+});
diff --git a/src/tools/web-browsing/components/SearchBar.tsx b/src/tools/web-browsing/components/SearchBar.tsx
index fbc442312a6ce..ea69a4121cd47 100644
--- a/src/tools/web-browsing/components/SearchBar.tsx
+++ b/src/tools/web-browsing/components/SearchBar.tsx
@@ -157,7 +157,7 @@ const SearchBar = memo<SearchBarProps>(
             options={Object.keys(CATEGORY_ICON_MAP).map((item) => ({
               label: (
                 <Flexbox align={'center'} gap={8} horizontal>
-                  <CategoryAvatar category={item as any} size={22} />
+                  <CategoryAvatar category={item as any} />
                   {t(`search.searchCategory.value.${item}` as any)}
                 </Flexbox>
               ),
