diff --git a/src/components/ModelIcon/index.tsx b/src/components/ModelIcon/index.tsx
index 5135ee8eda6a..15be9c148d91 100644
--- a/src/components/ModelIcon/index.tsx
+++ b/src/components/ModelIcon/index.tsx
@@ -62,7 +62,7 @@ const ModelIcon = memo<ModelProviderIconProps>(({ model: originModel, size = 12
   if (model.includes('moonshot')) return <Moonshot.Avatar size={size} />;
   if (model.includes('qwen')) return <Tongyi.Avatar background={Tongyi.colorPrimary} size={size} />;
   if (model.includes('minmax') || model.includes('abab')) return <Minimax.Avatar size={size} />;
-  if (model.includes('mistral') || model.includes('mixtral')) return <Mistral.Avatar size={size} />;
+  if (model.includes('mistral') || model.includes('mixtral') || model.includes('codestral')) return <Mistral.Avatar size={size} />;
   if (model.includes('pplx') || model.includes('sonar')) return <Perplexity.Avatar size={size} />;
   if (model.includes('yi-')) return <Yi.Avatar size={size} />;
   if (model.startsWith('openrouter')) return <OpenRouter.Avatar size={size} />; // only for Cinematika and Auto
diff --git a/src/components/ModelTag/ModelIcon.tsx b/src/components/ModelTag/ModelIcon.tsx
index c5d01fcd432a..a57a157ed01d 100644
--- a/src/components/ModelTag/ModelIcon.tsx
+++ b/src/components/ModelTag/ModelIcon.tsx
@@ -58,7 +58,7 @@ const ModelIcon = memo<ModelIconProps>(({ model, size = 12 }) => {
   if (model.includes('qwen')) return <Tongyi size={size} />;
   if (model.includes('minmax')) return <Minimax size={size} />;
   if (model.includes('abab')) return <Minimax size={size} />;
-  if (model.includes('mistral') || model.includes('mixtral')) return <Mistral size={size} />;
+  if (model.includes('mistral') || model.includes('mixtral') || model.includes('codestral')) return <Mistral size={size} />;
   if (model.includes('pplx') || model.includes('sonar')) return <Perplexity size={size} />;
   if (model.includes('yi-')) return <Yi size={size} />;
   if (model.startsWith('openrouter')) return <OpenRouter size={size} />; // only for Cinematika and Auto
diff --git a/src/config/modelProviders/mistral.ts b/src/config/modelProviders/mistral.ts
index 24a45f380fda..883a758584c2 100644
--- a/src/config/modelProviders/mistral.ts
+++ b/src/config/modelProviders/mistral.ts
@@ -1,6 +1,7 @@
 import { ModelProviderCard } from '@/types/llm';
 
 // ref https://docs.mistral.ai/getting-started/models/
+// ref https://docs.mistral.ai/capabilities/function_calling/
 const Mistral: ModelProviderCard = {
   chatModels: [
     {
@@ -24,22 +25,43 @@ const Mistral: ModelProviderCard = {
     },
     {
       displayName: 'Mistral Small',
-      enabled: true,
+      enabled: false,
+      functionCall: true,
       id: 'mistral-small-latest',
       tokens: 32_768,
     },
     {
       displayName: 'Mistral Medium',
-      enabled: true,
+      enabled: false,
       id: 'mistral-medium-latest',
       tokens: 32_768,
     },
+    {
+      displayName: 'Mistral Nemo',
+      enabled: true,
+      functionCall: true,
+      id: 'open-mistral-nemo',
+      tokens: 128_000,
+    },
     {
       displayName: 'Mistral Large',
       enabled: true,
+      functionCall: true,
       id: 'mistral-large-latest',
+      tokens: 128_000,
+    },
+    {
+      displayName: 'Codestral',
+      enabled: true,
+      id: 'codestral-latest',
       tokens: 32_768,
     },
+    {
+      displayName: 'Codestral Mamba',
+      enabled: true,
+      id: 'open-codestral-mamba',
+      tokens: 256_000,
+    },
   ],
   checkModel: 'open-mistral-7b',
   id: 'mistral',
