diff --git a/src/libs/next-auth/sso-providers/cloudflare-zero-trust.ts b/src/libs/next-auth/sso-providers/cloudflare-zero-trust.ts
index 2f0911f65c5a..0c03c770d385 100644
--- a/src/libs/next-auth/sso-providers/cloudflare-zero-trust.ts
+++ b/src/libs/next-auth/sso-providers/cloudflare-zero-trust.ts
@@ -24,7 +24,7 @@ const provider = {
     profile(profile) {
       return {
         email: profile.email,
-        name: profile.name,
+        name: profile.name ?? profile.email,
         providerAccountId: profile.sub,
       };
     },
