diff --git a/src/libs/mcp/client.ts b/src/libs/mcp/client.ts
index 8b140ed1e84d0..a81657e63e39a 100644
--- a/src/libs/mcp/client.ts
+++ b/src/libs/mcp/client.ts
@@ -1,5 +1,6 @@
 import { Client } from '@modelcontextprotocol/sdk/client/index.js';
 import { StdioClientTransport } from '@modelcontextprotocol/sdk/client/stdio.js';
+import { SSEClientTransport } from '@modelcontextprotocol/sdk/client/sse.js'
 import { StreamableHTTPClientTransport } from '@modelcontextprotocol/sdk/client/streamableHttp.js';
 import type { Transport } from '@modelcontextprotocol/sdk/shared/transport.d.ts';
 import debug from 'debug';
@@ -18,8 +19,13 @@ export class MCPClient {
 
     switch (params.type) {
       case 'http': {
-        log('Using HTTP transport with url: %s', params.url);
-        this.transport = new StreamableHTTPClientTransport(new URL(params.url));
+        if (params.url.includes('sse')) {
+          log('Using SSE transport with url: %s', params.url);
+          this.transport = new SSEClientTransport(new URL(params.url));
+        } else {
+          log('Using HTTP transport with url: %s', params.url);
+          this.transport = new StreamableHTTPClientTransport(new URL(params.url));
+        }
         break;
       }
       case 'stdio': {
