diff --git a/src/app/(main)/chat/(workspace)/_layout/Mobile/TopicModal.tsx b/src/app/(main)/chat/(workspace)/_layout/Mobile/TopicModal.tsx
index 3bbfd67d3e05..65fcf5a708f9 100644
--- a/src/app/(main)/chat/(workspace)/_layout/Mobile/TopicModal.tsx
+++ b/src/app/(main)/chat/(workspace)/_layout/Mobile/TopicModal.tsx
@@ -23,6 +23,7 @@ const Topics = memo(({ children }: PropsWithChildren) => {
   return (
     <Modal
       allowFullscreen
+      footer={null}
       onCancel={() => setOpen(false)}
       open={open}
       styles={{
