diff --git a/src/config/aiModels/openai.ts b/src/config/aiModels/openai.ts
index e6e229b8d8389..5f4e15cda054d 100644
--- a/src/config/aiModels/openai.ts
+++ b/src/config/aiModels/openai.ts
@@ -21,6 +21,7 @@ export const openaiChatModels: AIChatModelCard[] = [
     id: 'o3-mini',
     maxOutput: 100_000,
     pricing: {
+      cachedInput: 0.55,
       input: 1.1,
       output: 4.4,
     },
@@ -39,6 +40,7 @@ export const openaiChatModels: AIChatModelCard[] = [
     id: 'o1-mini',
     maxOutput: 65_536,
     pricing: {
+      cachedInput: 0.55,
       input: 1.1,
       output: 4.4,
     },
@@ -58,6 +60,7 @@ export const openaiChatModels: AIChatModelCard[] = [
     id: 'o1',
     maxOutput: 100_000,
     pricing: {
+      cachedInput: 7.5,
       input: 15,
       output: 60,
     },
@@ -82,6 +85,26 @@ export const openaiChatModels: AIChatModelCard[] = [
     releasedAt: '2024-09-12',
     type: 'chat',
   },
+  {
+    abilities: {
+      functionCall: true,
+      vision: true,
+    },
+    contextWindowTokens: 128_000,
+    description:
+      'GPT-4.5 的研究预览版，它是我们迄今为止最大、最强大的 GPT 模型。它拥有广泛的世界知识，并能更好地理解用户意图，使其在创造性任务和自主规划方面表现出色。GPT-4.5 可接受文本和图像输入，并生成文本输出（包括结构化输出）。支持关键的开发者功能，如函数调用、批量 API 和流式输出。在需要创造性、开放式思考和对话的任务（如写作、学习或探索新想法）中，GPT-4.5 表现尤为出色。知识截止日期为 2023 年 10 月。',
+    displayName: 'GPT-4.5 Preview',
+    enabled: true,
+    id: 'gpt-4.5-preview',
+    maxOutput: 16_384,
+    pricing: {
+      cachedInput: 37.5,
+      input: 75,
+      output: 150,
+    },
+    releasedAt: '2025-02-27',
+    type: 'chat',
+  },
   {
     abilities: {
       functionCall: true,
@@ -93,8 +116,9 @@ export const openaiChatModels: AIChatModelCard[] = [
     displayName: 'GPT-4o mini',
     enabled: true,
     id: 'gpt-4o-mini',
-    maxOutput: 16_385,
+    maxOutput: 16_384,
     pricing: {
+      cachedInput: 0.075,
       input: 0.15,
       output: 0.6,
     },
@@ -110,7 +134,6 @@ export const openaiChatModels: AIChatModelCard[] = [
     description:
       'ChatGPT-4o 是一款动态模型，实时更新以保持当前最新版本。它结合了强大的语言理解与生成能力，适合于大规模应用场景，包括客户服务、教育和技术支持。',
     displayName: 'GPT-4o 1120',
-    enabled: true,
     id: 'gpt-4o-2024-11-20',
     pricing: {
       input: 2.5,
@@ -131,6 +154,7 @@ export const openaiChatModels: AIChatModelCard[] = [
     enabled: true,
     id: 'gpt-4o',
     pricing: {
+      cachedInput: 1.25,
       input: 2.5,
       output: 10,
     },
@@ -350,7 +374,7 @@ export const openaiChatModels: AIChatModelCard[] = [
     abilities: {
       functionCall: true,
     },
-    contextWindowTokens: 16_385,
+    contextWindowTokens: 16_384,
     description:
       'GPT 3.5 Turbo，适用于各种文本生成和理解任务，Currently points to gpt-3.5-turbo-0125',
     displayName: 'GPT-3.5 Turbo',
@@ -365,7 +389,7 @@ export const openaiChatModels: AIChatModelCard[] = [
     abilities: {
       functionCall: true,
     },
-    contextWindowTokens: 16_385,
+    contextWindowTokens: 16_384,
     description:
       'GPT 3.5 Turbo，适用于各种文本生成和理解任务，Currently points to gpt-3.5-turbo-0125',
     displayName: 'GPT-3.5 Turbo 0125',
@@ -381,7 +405,7 @@ export const openaiChatModels: AIChatModelCard[] = [
     abilities: {
       functionCall: true,
     },
-    contextWindowTokens: 16_385,
+    contextWindowTokens: 16_384,
     description:
       'GPT 3.5 Turbo，适用于各种文本生成和理解任务，Currently points to gpt-3.5-turbo-0125',
     displayName: 'GPT-3.5 Turbo 1106',
