diff --git a/src/config/modelProviders/mistral.ts b/src/config/modelProviders/mistral.ts
index ca9e28bf00df..ef09e7c8bf5e 100644
--- a/src/config/modelProviders/mistral.ts
+++ b/src/config/modelProviders/mistral.ts
@@ -53,6 +53,20 @@ const Mistral: ModelProviderCard = {
       },
       tokens: 32_768,
     },
+    {
+      description:
+        'Pixtral Large 是一款拥有 1240 亿参数的开源多模态模型，基于 Mistral Large 2 构建。这是我们多模态家族中的第二款模型，展现了前沿水平的图像理解能力。',
+      displayName: 'Pixtral Large',
+      enabled: true,
+      functionCall: true,
+      id: 'pixtral-large-latest',
+      pricing: {
+        input: 2,
+        output: 6,
+      },
+      tokens: 128_000,
+      vision: true,
+    },
     {
       description:
         'Pixtral 模型在图表和图理解、文档问答、多模态推理和指令遵循等任务上表现出强大的能力，能够以自然分辨率和宽高比摄入图像，还能够在长达 128K 令牌的长上下文窗口中处理任意数量的图像。',
