diff --git a/src/libs/agent-runtime/qwen/index.ts b/src/libs/agent-runtime/qwen/index.ts
index b0cc566f5b0b..702980cdad30 100644
--- a/src/libs/agent-runtime/qwen/index.ts
+++ b/src/libs/agent-runtime/qwen/index.ts
@@ -36,7 +36,8 @@ export const LobeQwenAI = LobeOpenAICompatibleFactory({
         ...(model.startsWith('qwen-vl') ? {
           top_p: (top_p !== undefined && top_p > 0 && top_p <= 1) ? top_p : undefined,
         } : {
-          enable_search: true,
+          // Note: The Qwen-VL, Qwen open-source, and Qwen-Long currently do not support configuring this parameter
+          enable_search: model.startsWith('qwen-max') || model.startsWith('qwen-plus') || model.startsWith('qwen-turbo'),
           top_p: (top_p !== undefined && top_p > 0 && top_p < 1) ? top_p : undefined,
         }),
       } as any;
