diff --git a/src/config/modelProviders/ollama.ts b/src/config/modelProviders/ollama.ts
index f21091ab8cd8..6d7373b8f839 100644
--- a/src/config/modelProviders/ollama.ts
+++ b/src/config/modelProviders/ollama.ts
@@ -334,7 +334,7 @@ const Ollama: ModelProviderCard = {
   modelsUrl: 'https://ollama.com/library',
   name: 'Ollama',
   settings: {
-    disableBrowserRequest: true,
+    defaultShowBrowserRequest: true,
     sdkType: 'ollama',
     showApiKey: false,
     showModelFetcher: true,
