diff --git a/src/app/(main)/discover/(detail)/model/[...slugs]/features/ParameterList/index.tsx b/src/app/(main)/discover/(detail)/model/[...slugs]/features/ParameterList/index.tsx
index 4d11a87a6e7a..14f1e6708ad0 100644
--- a/src/app/(main)/discover/(detail)/model/[...slugs]/features/ParameterList/index.tsx
+++ b/src/app/(main)/discover/(detail)/model/[...slugs]/features/ParameterList/index.tsx
@@ -5,6 +5,7 @@ import {
   ChartColumnBig,
   Delete,
   FileMinus,
+  Pickaxe,
   LucideIcon,
   MessageSquareText,
   Thermometer,
@@ -83,6 +84,15 @@ const ParameterList = memo<ParameterListProps>(({ data }) => {
       range: data?.meta?.maxOutput ? [0, formatTokenNumber(data.meta.maxOutput)] : undefined,
       type: 'int',
     },
+    {
+      defaultValue: '--',
+      desc: t('models.parameterList.reasoning_effort.desc'),
+      icon: Pickaxe,
+      key: 'reasoning_effort',
+      label: t('models.parameterList.reasoning_effort.title'),
+      range: ['low', 'high'],
+      type: 'string',
+    },
   ];
 
   return (
