diff --git a/src/config/modelProviders/qwen.ts b/src/config/modelProviders/qwen.ts
index 2246235c6b6e..425c8837fa9a 100644
--- a/src/config/modelProviders/qwen.ts
+++ b/src/config/modelProviders/qwen.ts
@@ -254,6 +254,9 @@ const Qwen: ModelProviderCard = {
   modelList: { showModelFetcher: true },
   modelsUrl: 'https://help.aliyun.com/zh/dashscope/developer-reference/api-details',
   name: 'Qwen',
+  proxyUrl: {
+    placeholder: 'https://dashscope.aliyuncs.com/compatible-mode/v1',
+  },
   smoothing: {
     speed: 2,
     text: true,
