diff --git a/src/app/(backend)/webapi/chat/models/[provider]/route.ts b/src/app/(backend)/webapi/chat/models/[provider]/route.ts
index 1f6c4e5b24987..b99a05f5efa0f 100644
--- a/src/app/(backend)/webapi/chat/models/[provider]/route.ts
+++ b/src/app/(backend)/webapi/chat/models/[provider]/route.ts
@@ -9,7 +9,7 @@ import { createErrorResponse } from '@/utils/errorResponse';
 export const runtime = 'edge';
 
 const noNeedAPIKey = (provider: string) =>
-  [ModelProvider.OpenRouter, ModelProvider.TogetherAI].includes(provider as any);
+  [ModelProvider.OpenRouter].includes(provider as any);
 
 export const GET = checkAuth(async (req, { params, jwtPayload }) => {
   const { provider } = await params;
