diff --git a/src/config/aiModels/perplexity.ts b/src/config/aiModels/perplexity.ts
index 3c46da5dfdaad..00e7c5340b783 100644
--- a/src/config/aiModels/perplexity.ts
+++ b/src/config/aiModels/perplexity.ts
@@ -2,62 +2,55 @@ import { AIChatModelCard } from '@/types/aiModel';
 
 const perplexityChatModels: AIChatModelCard[] = [
   {
-    contextWindowTokens: 128_000,
+    contextWindowTokens: 127_072,
     description:
-      'Llama 3.1 Sonar Small Online 模型，具备8B参数，支持约127,000个标记的上下文长度，专为在线聊天设计，能高效处理各种文本交互。',
-    displayName: 'Llama 3.1 Sonar Small Online',
-    enabled: true,
-    id: 'llama-3.1-sonar-small-128k-online',
-    type: 'chat',
-  },
-  {
-    contextWindowTokens: 128_000,
-    description:
-      'Llama 3.1 Sonar Large Online 模型，具备70B参数，支持约127,000个标记的上下文长度，适用于高容量和多样化聊天任务。',
-    displayName: 'Llama 3.1 Sonar Large Online',
+      '由 DeepSeek 推理模型提供支持的新 API 产品。',
+    displayName: 'Sonar Reasoning',
     enabled: true,
-    id: 'llama-3.1-sonar-large-128k-online',
+    id: 'sonar-reasoning',
     type: 'chat',
   },
   {
-    contextWindowTokens: 128_000,
+    contextWindowTokens: 200_000,
     description:
-      'Llama 3.1 Sonar Huge Online 模型，具备405B参数，支持约127,000个标记的上下文长度，设计用于复杂的在线聊天应用。',
-    displayName: 'Llama 3.1 Sonar Huge Online',
+      '支持搜索上下文的高级搜索产品，支持高级查询和跟进。',
+    displayName: 'Sonar Pro',
     enabled: true,
-    id: 'llama-3.1-sonar-huge-128k-online',
+    id: 'sonar-pro',
     type: 'chat',
   },
   {
-    contextWindowTokens: 128_000,
+    contextWindowTokens: 127_072,
     description:
-      'Llama 3.1 Sonar Small Chat 模型，具备8B参数，专为离线聊天设计，支持约127,000个标记的上下文长度。',
-    displayName: 'Llama 3.1 Sonar Small Chat',
+      '基于搜索上下文的轻量级搜索产品，比 Sonar Pro 更快、更便宜。',
+    displayName: 'Sonar',
     enabled: true,
-    id: 'llama-3.1-sonar-small-128k-chat',
+    id: 'sonar',
     type: 'chat',
   },
+  // The following will be deprecated on 02-22
   {
-    contextWindowTokens: 128_000,
+    contextWindowTokens: 127_072,
     description:
-      'Llama 3.1 Sonar Large Chat 模型，具备70B参数，支持约127,000个标记的上下文长度，适合于复杂的离线聊天任务。',
-    displayName: 'Llama 3.1 Sonar Large Chat',
-    enabled: true,
-    id: 'llama-3.1-sonar-large-128k-chat',
+      'Llama 3.1 Sonar Small Online 模型，具备8B参数，支持约127,000个标记的上下文长度，专为在线聊天设计，能高效处理各种文本交互。',
+    displayName: 'Llama 3.1 Sonar Small Online',
+    id: 'llama-3.1-sonar-small-128k-online',
     type: 'chat',
   },
   {
-    contextWindowTokens: 128_000,
+    contextWindowTokens: 127_072,
     description:
-      'Llama 3.1 8B Instruct 模型，具备8B参数，支持画面指示任务的高效执行，提供优质的文本生成能力。',
-    id: 'llama-3.1-8b-instruct',
+      'Llama 3.1 Sonar Large Online 模型，具备70B参数，支持约127,000个标记的上下文长度，适用于高容量和多样化聊天任务。',
+    displayName: 'Llama 3.1 Sonar Large Online',
+    id: 'llama-3.1-sonar-large-128k-online',
     type: 'chat',
   },
   {
-    contextWindowTokens: 128_000,
+    contextWindowTokens: 127_072,
     description:
-      'Llama 3.1 70B Instruct 模型，具备70B参数，能在大型文本生成和指示任务中提供卓越性能。',
-    id: 'llama-3.1-70b-instruct',
+      'Llama 3.1 Sonar Huge Online 模型，具备405B参数，支持约127,000个标记的上下文长度，设计用于复杂的在线聊天应用。',
+    displayName: 'Llama 3.1 Sonar Huge Online',
+    id: 'llama-3.1-sonar-huge-128k-online',
     type: 'chat',
   },
 ];
diff --git a/src/config/modelProviders/perplexity.ts b/src/config/modelProviders/perplexity.ts
index 6475b75d8262e..bf99f7fb7fc40 100644
--- a/src/config/modelProviders/perplexity.ts
+++ b/src/config/modelProviders/perplexity.ts
@@ -4,59 +4,52 @@ import { ModelProviderCard } from '@/types/llm';
 const Perplexity: ModelProviderCard = {
   chatModels: [
     {
-      contextWindowTokens: 128_000,
-      description:
-        'Llama 3.1 Sonar Small Online 模型，具备8B参数，支持约127,000个标记的上下文长度，专为在线聊天设计，能高效处理各种文本交互。',
-      displayName: 'Llama 3.1 Sonar Small Online',
-      enabled: true,
-      id: 'llama-3.1-sonar-small-128k-online',
-    },
-    {
-      contextWindowTokens: 128_000,
-      description:
-        'Llama 3.1 Sonar Large Online 模型，具备70B参数，支持约127,000个标记的上下文长度，适用于高容量和多样化聊天任务。',
-      displayName: 'Llama 3.1 Sonar Large Online',
-      enabled: true,
-      id: 'llama-3.1-sonar-large-128k-online',
-    },
-    {
-      contextWindowTokens: 128_000,
-      description:
-        'Llama 3.1 Sonar Huge Online 模型，具备405B参数，支持约127,000个标记的上下文长度，设计用于复杂的在线聊天应用。',
-      displayName: 'Llama 3.1 Sonar Huge Online',
-      enabled: true,
-      id: 'llama-3.1-sonar-huge-128k-online',
-    },
-    {
-      contextWindowTokens: 128_000,
-      description:
-        'Llama 3.1 Sonar Small Chat 模型，具备8B参数，专为离线聊天设计，支持约127,000个标记的上下文长度。',
-      displayName: 'Llama 3.1 Sonar Small Chat',
-      enabled: true,
-      id: 'llama-3.1-sonar-small-128k-chat',
-    },
-    {
-      contextWindowTokens: 128_000,
-      description:
-        'Llama 3.1 Sonar Large Chat 模型，具备70B参数，支持约127,000个标记的上下文长度，适合于复杂的离线聊天任务。',
-      displayName: 'Llama 3.1 Sonar Large Chat',
-      enabled: true,
-      id: 'llama-3.1-sonar-large-128k-chat',
-    },
-    {
-      contextWindowTokens: 128_000,
-      description:
-        'Llama 3.1 8B Instruct 模型，具备8B参数，支持画面指示任务的高效执行，提供优质的文本生成能力。',
-      id: 'llama-3.1-8b-instruct',
-    },
-    {
-      contextWindowTokens: 128_000,
-      description:
-        'Llama 3.1 70B Instruct 模型，具备70B参数，能在大型文本生成和指示任务中提供卓越性能。',
-      id: 'llama-3.1-70b-instruct',
-    },
+      contextWindowTokens: 127_072,
+    description:
+      '由 DeepSeek 推理模型提供支持的新 API 产品。',
+    displayName: 'Sonar Reasoning',
+    enabled: true,
+    id: 'sonar-reasoning',
+  },
+  {
+    contextWindowTokens: 200_000,
+    description:
+      '支持搜索上下文的高级搜索产品，支持高级查询和跟进。',
+    displayName: 'Sonar Pro',
+    enabled: true,
+    id: 'sonar-pro',
+  },
+  {
+    contextWindowTokens: 127_072,
+    description:
+      '基于搜索上下文的轻量级搜索产品，比 Sonar Pro 更快、更便宜。',
+    displayName: 'Sonar',
+    enabled: true,
+    id: 'sonar',
+  },
+  // The following will be deprecated on 02-22
+  {
+    contextWindowTokens: 127_072,
+    description:
+      'Llama 3.1 Sonar Small Online 模型，具备8B参数，支持约127,000个标记的上下文长度，专为在线聊天设计，能高效处理各种文本交互。',
+    displayName: 'Llama 3.1 Sonar Small Online',
+    id: 'llama-3.1-sonar-small-128k-online',
+  },
+  {
+    contextWindowTokens: 127_072,
+    description:
+      'Llama 3.1 Sonar Large Online 模型，具备70B参数，支持约127,000个标记的上下文长度，适用于高容量和多样化聊天任务。',
+    displayName: 'Llama 3.1 Sonar Large Online',
+    id: 'llama-3.1-sonar-large-128k-online',
+  },
+  {
+    contextWindowTokens: 127_072,
+    description:
+      'Llama 3.1 Sonar Huge Online 模型，具备405B参数，支持约127,000个标记的上下文长度，设计用于复杂的在线聊天应用。',
+    displayName: 'Llama 3.1 Sonar Huge Online',
+    id: 'llama-3.1-sonar-huge-128k-online',
   ],
-  checkModel: 'llama-3.1-8b-instruct',
+  checkModel: 'sonar',
   description:
     'Perplexity 是一家领先的对话生成模型提供商，提供多种先进的Llama 3.1模型，支持在线和离线应用，特别适用于复杂的自然语言处理任务。',
   id: 'perplexity',
