diff --git a/src/app/(main)/settings/llm/components/ProviderModelList/ModelConfigModal/Form.tsx b/src/app/(main)/settings/llm/components/ProviderModelList/ModelConfigModal/Form.tsx
index 0cb2c3ad3a04..6f3b7a1f7772 100644
--- a/src/app/(main)/settings/llm/components/ProviderModelList/ModelConfigModal/Form.tsx
+++ b/src/app/(main)/settings/llm/components/ProviderModelList/ModelConfigModal/Form.tsx
@@ -66,7 +66,7 @@ const ModelConfigForm = memo<ModelConfigFormProps>(
           >
             <Input placeholder={t('llm.customModelCards.modelConfig.displayName.placeholder')} />
           </Form.Item>
-          <Form.Item label={t('llm.customModelCards.modelConfig.tokens.title')} name={'tokens'}>
+          <Form.Item label={t('llm.customModelCards.modelConfig.tokens.title')} name={'contextWindowTokens'}>
             <MaxTokenSlider />
           </Form.Item>
           <Form.Item
