diff --git a/src/config/aiModels/baichuan.ts b/src/config/aiModels/baichuan.ts
index 5139377289bc0..b62f2081e9155 100644
--- a/src/config/aiModels/baichuan.ts
+++ b/src/config/aiModels/baichuan.ts
@@ -4,6 +4,7 @@ const baichuanChatModels: AIChatModelCard[] = [
   {
     abilities: {
       functionCall: true,
+      search: true,
     },
     contextWindowTokens: 32_768,
     description:
@@ -17,11 +18,15 @@ const baichuanChatModels: AIChatModelCard[] = [
       input: 100,
       output: 100,
     },
+    settings: {
+      searchImpl: 'params',
+    },
     type: 'chat',
   },
   {
     abilities: {
       functionCall: true,
+      search: true,
     },
     contextWindowTokens: 32_768,
     description:
@@ -35,11 +40,15 @@ const baichuanChatModels: AIChatModelCard[] = [
       input: 15,
       output: 15,
     },
+    settings: {
+      searchImpl: 'params',
+    },
     type: 'chat',
   },
   {
     abilities: {
       functionCall: true,
+      search: true,
     },
     contextWindowTokens: 32_768,
     description:
@@ -53,11 +62,15 @@ const baichuanChatModels: AIChatModelCard[] = [
       input: 0.98,
       output: 0.98,
     },
+    settings: {
+      searchImpl: 'params',
+    },
     type: 'chat',
   },
   {
     abilities: {
       functionCall: true,
+      search: true,
     },
     contextWindowTokens: 32_768,
     description:
@@ -70,6 +83,9 @@ const baichuanChatModels: AIChatModelCard[] = [
       input: 12,
       output: 12,
     },
+    settings: {
+      searchImpl: 'params',
+    },
     type: 'chat',
   },
   {
diff --git a/src/libs/agent-runtime/baichuan/index.ts b/src/libs/agent-runtime/baichuan/index.ts
index cd36499769f4f..15c46143518a6 100644
--- a/src/libs/agent-runtime/baichuan/index.ts
+++ b/src/libs/agent-runtime/baichuan/index.ts
@@ -1,5 +1,3 @@
-import OpenAI from 'openai';
-
 import { ChatStreamPayload, ModelProvider } from '../types';
 import { LobeOpenAICompatibleFactory } from '../utils/openaiCompatibleFactory';
 
@@ -17,14 +15,26 @@ export const LobeBaichuanAI = LobeOpenAICompatibleFactory({
   baseURL: 'https://api.baichuan-ai.com/v1',
   chatCompletion: {
     handlePayload: (payload: ChatStreamPayload) => {
-      const { temperature, ...rest } = payload;
+      const { enabledSearch, temperature, tools, ...rest } = payload;
 
       return {
         ...rest,
         // [baichuan] frequency_penalty must be between 1 and 2.
         frequency_penalty: undefined,
         temperature: temperature !== undefined ? temperature / 2 : undefined,
-      } as OpenAI.ChatCompletionCreateParamsStreaming;
+        ...(enabledSearch && {
+          tools: [
+            ...(tools || []),
+            {
+              type: "web_search",
+              web_search: {
+                enable: true,
+                search_mode: process.env.BAICHUAN_SEARCH_MODE || "performance_first", // performance_first or quality_first
+              },
+            }
+          ]
+        }),
+      } as any;
     },
   },
   debug: {
