diff --git a/src/features/ChatInput/Topic/index.tsx b/src/features/ChatInput/Topic/index.tsx
index 4f5d690e92d34..609e816729d6f 100644
--- a/src/features/ChatInput/Topic/index.tsx
+++ b/src/features/ChatInput/Topic/index.tsx
@@ -49,14 +49,12 @@ const SaveTopic = memo<{ mobile?: boolean }>(({ mobile }) => {
           </div>
         }
       >
-        <Tooltip>
-          <ActionIcon
-            aria-label={desc}
-            icon={icon}
-            loading={isValidating}
-            onClick={() => setConfirmOpened(true)}
-          />
-        </Tooltip>
+        <ActionIcon
+          aria-label={desc}
+          icon={icon}
+          loading={isValidating}
+          onClick={() => setConfirmOpened(true)}
+        />
       </Popconfirm>
     );
   } else {
