diff --git a/next.config.mjs b/next.config.mjs
index 4678774e..337b60e9 100644
--- a/next.config.mjs
+++ b/next.config.mjs
@@ -1,4 +1,10 @@
 /** @type {import('next').NextConfig} */
-const nextConfig = {};
+const isProd = process.env.NODE_ENV === 'production';
+const buildWithDocker = process.env.DOCKER === 'true';
+
+const nextConfig = {
+  compress: isProd,
+  output: buildWithDocker ? 'standalone' : undefined,
+};
 
 export default nextConfig;
