From 6f88bd474f1e2e75b4c6ec36dc51e82073d7de75 Mon Sep 17 00:00:00 2001
From: Zhijie He <hezhijie0327@hotmail.com>
Date: Tue, 24 Oct 2023 19:58:29 +0800
Subject: [PATCH] Change default EDNS UDP buffer size to 1232 (2020 DNS Flag Day)

---
 proxy/helpers.go | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/proxy/helpers.go b/proxy/helpers.go
index 273e178..1d4959c 100644
--- a/proxy/helpers.go
+++ b/proxy/helpers.go
@@ -150,7 +150,7 @@ func setECS(m *dns.Msg, ip net.IP, scope uint8) (subnet *net.IPNet) {
 		},
 		Option: []dns.EDNS0{e},
 	}
-	o.SetUDPSize(4096)
+	o.SetUDPSize(1232)
 	m.Extra = append(m.Extra, o)
 
 	return subnet
-- 
2.42.0.windows.2

